{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2fad2d26-d5e6-426a-94e4-63f06b62004f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from find_path import find_file\n",
    "import sqlalchemy\n",
    "import db_utils\n",
    "import warnings \n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71136ed3-77fd-4317-9d09-b8a90560ab86",
   "metadata": {},
   "source": [
    "# Matching with fragments predicted by the DMI predictor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "babb2992-676e-4bf1-8147-bbe068bf0b8a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# load mutations\n",
    "eng = sqlalchemy.create_engine('mysql://', creator= db_utils.get_connection)\n",
    "query = '''select * from milena_db.DDD_and_clinvar_mutations_2023_release as df'''\n",
    "mut_data = pd.read_sql_query(query, con=eng)\n",
    "mut_data_sub1 = mut_data[mut_data['allele_id'].isna()] #the data types were floats for some reason and mixed with NaN values\n",
    "mut_data_sub2 = mut_data[~(mut_data['allele_id'].isna())]\n",
    "mut_data_sub2['allele_id'] = mut_data_sub2['allele_id'].astype('int')\n",
    "mut_data_sub2['variation_id'] = mut_data_sub2['variation_id'].astype('int')\n",
    "mut_data_sub1['allele_id'] = None\n",
    "mut_data_sub1['variation_id'] = None\n",
    "mut_data = pd.concat([mut_data_sub2, mut_data_sub1], axis=0)\n",
    "\n",
    "# load John's table of predicted DMI interfaces\n",
    "query = pd.read_sql_query(\"select * from chopyan_db.HuRI_BioPlex_lit17_union as df where DMIMatchScore >= 0.7\", con = eng)\n",
    "dmi = pd.DataFrame(query)\n",
    "\n",
    "# I saved all the columns I think I will need later:\n",
    "dmi = dmi[['intx_ID', 'Accession', 'Elm', 'NDDSLiMProtein', 'SLiMProtein', 'SLiMMatch', 'NDDDomainProtein', 'DomainProtein', 'DomainID1', 'DomainName1', 'DomainMatch1', 'DomainMatchSequence1', 'DomainMatchFound1', 'DomainMatchRequired1', 'DomainID2', 'DomainName2', 'DomainMatch2', 'DomainMatchSequence2', 'DomainMatchFound2', 'DomainMatchRequired2', 'DMIMatchScore', 'Source_lit17', 'Source_HuRI', 'Source_BioPlex']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3dc942ac-d644-4258-9cd2-7c9b0340f835",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fragment_protein</th>\n",
       "      <th>fragment_id</th>\n",
       "      <th>fragment_pos</th>\n",
       "      <th>fragment_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A087WUL8</td>\n",
       "      <td>ELME000443</td>\n",
       "      <td>258-264</td>\n",
       "      <td>slim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A087WUL8</td>\n",
       "      <td>ELME000084</td>\n",
       "      <td>211-214</td>\n",
       "      <td>slim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A087WUL8</td>\n",
       "      <td>ELME000045</td>\n",
       "      <td>126-132</td>\n",
       "      <td>slim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A096LP49</td>\n",
       "      <td>ELME000271</td>\n",
       "      <td>246-252</td>\n",
       "      <td>slim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A096LP49</td>\n",
       "      <td>ELME000388</td>\n",
       "      <td>250-254</td>\n",
       "      <td>slim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25032</th>\n",
       "      <td>Q05655</td>\n",
       "      <td>SM00133</td>\n",
       "      <td>604-667</td>\n",
       "      <td>domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25033</th>\n",
       "      <td>Q9NRM7</td>\n",
       "      <td>SM00133</td>\n",
       "      <td>974-1044</td>\n",
       "      <td>domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25034</th>\n",
       "      <td>Q08999</td>\n",
       "      <td>PF01858</td>\n",
       "      <td>417-609</td>\n",
       "      <td>domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25035</th>\n",
       "      <td>Q9Y5S2</td>\n",
       "      <td>SM00133</td>\n",
       "      <td>343-405</td>\n",
       "      <td>domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25036</th>\n",
       "      <td>Q5VT25</td>\n",
       "      <td>SM00133</td>\n",
       "      <td>344-406</td>\n",
       "      <td>domain</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25037 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      fragment_protein fragment_id fragment_pos fragment_type\n",
       "0           A0A087WUL8  ELME000443      258-264          slim\n",
       "1           A0A087WUL8  ELME000084      211-214          slim\n",
       "2           A0A087WUL8  ELME000045      126-132          slim\n",
       "3           A0A096LP49  ELME000271      246-252          slim\n",
       "4           A0A096LP49  ELME000388      250-254          slim\n",
       "...                ...         ...          ...           ...\n",
       "25032           Q05655     SM00133      604-667        domain\n",
       "25033           Q9NRM7     SM00133     974-1044        domain\n",
       "25034           Q08999     PF01858      417-609        domain\n",
       "25035           Q9Y5S2     SM00133      343-405        domain\n",
       "25036           Q5VT25     SM00133      344-406        domain\n",
       "\n",
       "[25037 rows x 4 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# making a list of slims:\n",
    "slim = dmi[['SLiMProtein','Accession','SLiMMatch']]\n",
    "slim.rename(columns = {'SLiMProtein':'fragment_protein','Accession':'fragment_id', 'SLiMMatch':'fragment_pos'}, inplace = True)\n",
    "slim['fragment_type'] = 'slim'\n",
    "slim.drop_duplicates(inplace = True)\n",
    "slim.reset_index(inplace=True, drop=True)\n",
    "\n",
    "# making a list of domains 1; like slims, they are always provided (in each row)\n",
    "dom_1 = dmi[['DomainProtein','DomainID1','DomainMatch1']]\n",
    "dom_1.rename(columns = {'DomainProtein':'fragment_protein','DomainID1':'fragment_id', 'DomainMatch1':'fragment_pos'}, inplace = True)\n",
    "dom_1['fragment_type'] = 'domain'\n",
    "dom_1.drop_duplicates(inplace=True)\n",
    "dom_1.reset_index(inplace=True, drop=True)\n",
    "\n",
    "# second domain\n",
    "dom_2 = dmi[~(dmi['DomainID2'].isna()) & ~(dmi['DomainMatch2'].isna())][['DomainProtein','DomainID2','DomainMatch2']]\n",
    "dom_2.rename(columns={'DomainProtein':'fragment_protein','DomainID2':'fragment_id','DomainMatch2':'fragment_pos'}, inplace = True)\n",
    "dom_2['fragment_type'] = 'domain'\n",
    "dom_2.drop_duplicates(inplace=True)\n",
    "dom_2.reset_index(inplace=True, drop=True)\n",
    "\n",
    "# concatenate them:\n",
    "frag = pd.concat([slim,dom_1,dom_2], axis=0, ignore_index=True)\n",
    "frag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d924653d-3008-40aa-a116-bd9b967594f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fragment_type</th>\n",
       "      <th>fragment_protein</th>\n",
       "      <th>fragment_id</th>\n",
       "      <th>fragment_pos</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>composite_id</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>pathogenicity</th>\n",
       "      <th>ndd_phe</th>\n",
       "      <th>ndd_protein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>domain</td>\n",
       "      <td>Q92625</td>\n",
       "      <td>SM00248</td>\n",
       "      <td>79-108|112-141|148-177|181-210|214-243|246-275</td>\n",
       "      <td>2219276</td>\n",
       "      <td>2228534</td>\n",
       "      <td>6</td>\n",
       "      <td>6_34981954_C_A</td>\n",
       "      <td>Leu234Ile</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>domain</td>\n",
       "      <td>Q92625</td>\n",
       "      <td>SM00248</td>\n",
       "      <td>79-108|112-141|148-177|181-210|214-243|246-275</td>\n",
       "      <td>2704296</td>\n",
       "      <td>2537954</td>\n",
       "      <td>6</td>\n",
       "      <td>6_34981799_T_G</td>\n",
       "      <td>Phe182Cys</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>domain</td>\n",
       "      <td>Q92625</td>\n",
       "      <td>SM00248</td>\n",
       "      <td>79-108|112-141|148-177|181-210|214-243|246-275</td>\n",
       "      <td>2762399</td>\n",
       "      <td>2593541</td>\n",
       "      <td>6</td>\n",
       "      <td>6_34970074_T_C</td>\n",
       "      <td>Tyr115His</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>domain</td>\n",
       "      <td>A1X283</td>\n",
       "      <td>SM00326</td>\n",
       "      <td>155-210|224-279|371-426|853-911</td>\n",
       "      <td>303282</td>\n",
       "      <td>352815</td>\n",
       "      <td>5</td>\n",
       "      <td>5_172346198_T_C</td>\n",
       "      <td>Ile376Val</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>domain</td>\n",
       "      <td>A1X283</td>\n",
       "      <td>SM00326</td>\n",
       "      <td>155-210|224-279|371-426|853-911</td>\n",
       "      <td>443751</td>\n",
       "      <td>451250</td>\n",
       "      <td>5</td>\n",
       "      <td>5_172338487_C_G</td>\n",
       "      <td>Gly873Ala</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49921</th>\n",
       "      <td>domain</td>\n",
       "      <td>Q9Y587</td>\n",
       "      <td>PF01217</td>\n",
       "      <td>1-141</td>\n",
       "      <td>2383119</td>\n",
       "      <td>2384873</td>\n",
       "      <td>14</td>\n",
       "      <td>14_31069903_A_G</td>\n",
       "      <td>Ile67Val</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49922</th>\n",
       "      <td>domain</td>\n",
       "      <td>Q9Y587</td>\n",
       "      <td>PF01217</td>\n",
       "      <td>1-141</td>\n",
       "      <td>2404999</td>\n",
       "      <td>2431495</td>\n",
       "      <td>14</td>\n",
       "      <td>14_31066240_G_A</td>\n",
       "      <td>Arg15Gln</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49923</th>\n",
       "      <td>domain</td>\n",
       "      <td>Q9Y587</td>\n",
       "      <td>PF01217</td>\n",
       "      <td>1-141</td>\n",
       "      <td>2411751</td>\n",
       "      <td>2439139</td>\n",
       "      <td>14</td>\n",
       "      <td>14_31069856_A_G</td>\n",
       "      <td>Glu51Gly</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49924</th>\n",
       "      <td>domain</td>\n",
       "      <td>Q9Y587</td>\n",
       "      <td>PF01217</td>\n",
       "      <td>1-141</td>\n",
       "      <td>2448610</td>\n",
       "      <td>2469842</td>\n",
       "      <td>14</td>\n",
       "      <td>14_31069883_G_A</td>\n",
       "      <td>Arg60Gln</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49925</th>\n",
       "      <td>domain</td>\n",
       "      <td>Q9Y587</td>\n",
       "      <td>PF01217</td>\n",
       "      <td>1-141</td>\n",
       "      <td>2465196</td>\n",
       "      <td>2495208</td>\n",
       "      <td>14</td>\n",
       "      <td>14_31072908_G_A</td>\n",
       "      <td>Glu77Lys</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49926 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      fragment_type fragment_protein fragment_id  \\\n",
       "0            domain           Q92625     SM00248   \n",
       "1            domain           Q92625     SM00248   \n",
       "2            domain           Q92625     SM00248   \n",
       "3            domain           A1X283     SM00326   \n",
       "4            domain           A1X283     SM00326   \n",
       "...             ...              ...         ...   \n",
       "49921        domain           Q9Y587     PF01217   \n",
       "49922        domain           Q9Y587     PF01217   \n",
       "49923        domain           Q9Y587     PF01217   \n",
       "49924        domain           Q9Y587     PF01217   \n",
       "49925        domain           Q9Y587     PF01217   \n",
       "\n",
       "                                         fragment_pos allele_id variation_id  \\\n",
       "0      79-108|112-141|148-177|181-210|214-243|246-275   2219276      2228534   \n",
       "1      79-108|112-141|148-177|181-210|214-243|246-275   2704296      2537954   \n",
       "2      79-108|112-141|148-177|181-210|214-243|246-275   2762399      2593541   \n",
       "3                     155-210|224-279|371-426|853-911    303282       352815   \n",
       "4                     155-210|224-279|371-426|853-911    443751       451250   \n",
       "...                                               ...       ...          ...   \n",
       "49921                                           1-141   2383119      2384873   \n",
       "49922                                           1-141   2404999      2431495   \n",
       "49923                                           1-141   2411751      2439139   \n",
       "49924                                           1-141   2448610      2469842   \n",
       "49925                                           1-141   2465196      2495208   \n",
       "\n",
       "      chromosome     composite_id  aa_change pathogenicity  ndd_phe  \\\n",
       "0              6   6_34981954_C_A  Leu234Ile     uncertain        0   \n",
       "1              6   6_34981799_T_G  Phe182Cys     uncertain        0   \n",
       "2              6   6_34970074_T_C  Tyr115His     uncertain        0   \n",
       "3              5  5_172346198_T_C  Ile376Val     uncertain        0   \n",
       "4              5  5_172338487_C_G  Gly873Ala     uncertain        0   \n",
       "...          ...              ...        ...           ...      ...   \n",
       "49921         14  14_31069903_A_G   Ile67Val     uncertain        0   \n",
       "49922         14  14_31066240_G_A   Arg15Gln     uncertain        1   \n",
       "49923         14  14_31069856_A_G   Glu51Gly     uncertain        1   \n",
       "49924         14  14_31069883_G_A   Arg60Gln     uncertain        0   \n",
       "49925         14  14_31072908_G_A   Glu77Lys     uncertain        0   \n",
       "\n",
       "       ndd_protein  \n",
       "0                0  \n",
       "1                0  \n",
       "2                0  \n",
       "3                0  \n",
       "4                0  \n",
       "...            ...  \n",
       "49921            0  \n",
       "49922            0  \n",
       "49923            0  \n",
       "49924            0  \n",
       "49925            0  \n",
       "\n",
       "[49926 rows x 12 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merging frag table with the mut_data on uniprot_id\n",
    "\n",
    "# # to check if a mutation falls within a certain fragment in a certain protein, we need to find fragments of the protein where the mutation lies\n",
    "overlap = frag.merge(mut_data, how='inner', left_on='fragment_protein', right_on='uniprot_id')\n",
    "# I filtered clinvar additionally, so I might not have the same numbers as Kristina\n",
    "\n",
    "# now let's look for overlap:\n",
    "overlap['fragment_pos_for_overlap'] = [''.join([a for a in i.split('|')]) if len([a for a in i.split('|')]) == 1 else [a for a in i.split('|')] for i in overlap['fragment_pos'].values]\n",
    "\n",
    "overlap_list = overlap[overlap['fragment_pos_for_overlap'].apply(lambda x: type(x) == list)] # dataset where there is only 1 predicted position per fragment\n",
    "overlap_not_list = overlap[~(overlap['fragment_pos_for_overlap'].apply(lambda x: type(x) == list))]\n",
    "overlap_list.reset_index(inplace=True, drop=True)\n",
    "overlap_not_list.reset_index(inplace=True, drop=True)\n",
    "\n",
    "overlap_not_list['overlap'] = [1 if (a >= int(b.split('-')[0])) and (a <= int(b.split('-')[1])) else 0 for a, b in zip(overlap_not_list['aa_pos'], overlap_not_list['fragment_pos_for_overlap'])]\n",
    "overlap_list['overlap'] = [1 if 1 in [1 if (a >= int(x.split('-')[0])) and (a <= int(x.split('-')[1])) else 0 for x in b] else 0 for a, b in zip(overlap_list['aa_pos'], overlap_list['fragment_pos_for_overlap'])]\n",
    "\n",
    "overlap = pd.concat([overlap_list, overlap_not_list], axis=0)\n",
    "overlap.reset_index(inplace=True, drop=True)\n",
    "\n",
    "# annotate proteins for NDDs\n",
    "query = '''select * from milena_db.NDD_genes_list_v3 as df'''\n",
    "ndd = pd.read_sql_query(query, con=eng)\n",
    "ndd = ndd['uniprot_id'].unique().tolist()\n",
    "overlap['ndd_protein'] = [1 if i in ndd else 0 for i in overlap['uniprot_id']]\n",
    "\n",
    "# fix dtype in chromosome column:\n",
    "overlap['chromosome'] = [int(i) if i.isdigit() else i for i in overlap['chromosome']]\n",
    "\n",
    "# select columns and filter for those rows where overlap == 1\n",
    "overlap = overlap[['fragment_type','fragment_protein','fragment_id','fragment_pos','allele_id','variation_id','chromosome', 'composite_id', 'aa_change', 'pathogenicity', 'ndd_phe', 'ndd_protein', 'overlap']]\n",
    "overlap = overlap[overlap['overlap'] == 1]\n",
    "overlap.drop(columns='overlap', inplace=True)\n",
    "overlap.drop_duplicates(inplace=True)\n",
    "overlap.reset_index(inplace=True, drop=True)\n",
    "\n",
    "overlap"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5bd05209-b5a4-43bc-b1d0-f53295327d99",
   "metadata": {},
   "source": [
    "# Push to mysql"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "afd88d04-35bd-404a-90ae-a4b5afbff057",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "allele_id\n",
      "variation_id\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "11"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#check for NaN values:\n",
    "col_list = overlap.columns.tolist()\n",
    "for col in col_list:\n",
    "    if ((overlap[col].isna()).any()).any():\n",
    "        print(col) \n",
    "#none of the columns contain NaN values\n",
    "\n",
    "max([len((str(i))) for i in overlap['pathogenicity']])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "22dbd30d-ed3a-487f-aa5e-8aa59a5eb728",
   "metadata": {},
   "source": [
    "A note about the primary key: protein, fragment_id and fragment_pos is what uniquely identifies a fragment, and on the other hand composite id and aa_change is what uniquely defines a mutation (aa_change also needs to be there because a gene can code for 2 proteins and then aa_change can be different, even though the genomic location of the mutation is the same), so the primary key combines all these columns.\n",
    "Theoretically, composite_id and aa_change could be replaced by allele_id, variation_id and chromosome, but the problem is that DDD mutations don't have this, and there are None values there, so technically there are duplicates for these columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c3507d20-25f2-4736-886a-37418070e0dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#  CREATE TABLE `DMI_overlap_ddd_and_clinvar_2023_release` (\n",
    "#    `fragment_type` text(7) NOT NULL,\n",
    "#    `fragment_protein` varchar(12) NOT NULL,\n",
    "#    `fragment_id` varchar(12) NOT NULL,\n",
    "#    `fragment_pos` varchar(130) NOT NULL,\n",
    "#    `allele_id` int(8) DEFAULT NULL,\n",
    "#    `variation_id` int(8) DEFAULT NULL,\n",
    "#    `chromosome` varchar(2) NOT NULL,\n",
    "#    `composite_id` varchar(18) NOT NULL,\n",
    "#    `aa_change` varchar(12) NOT NULL,\n",
    "#    `pathogenicity` text(12) NOT NULL,\n",
    "#    `ndd_phe` int(1) NOT NULL,\n",
    "#    `ndd_protein` int(1) NOT NULL,\n",
    "#    PRIMARY KEY (`fragment_protein`,`fragment_id`,`fragment_pos`,`composite_id`,`aa_change`)\n",
    "#  ) ENGINE=MyISAM DEFAULT CHARSET=utf8mb4;"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "f9885a53-d246-4e79-a31a-f2904650f6e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "connect = db_utils.get_connection()\n",
    "cursor = connect.cursor()\n",
    "\n",
    "db_table = 'milena_db.DMI_overlap_ddd_and_clinvar_2023_release'\n",
    "query = \"\"\"insert into %s \n",
    "        (fragment_type, fragment_protein, fragment_id, fragment_pos, allele_id, variation_id, chromosome, composite_id, aa_change, pathogenicity, ndd_phe, ndd_protein)\n",
    "        values (%%s, %%s, %%s, %%s, %%s, %%s, %%s, %%s, %%s, %%s, %%s, %%s)\"\"\" % (db_table)\n",
    "\n",
    "for i in range(overlap.shape[0]):\n",
    "    cursor.execute(query, tuple(overlap.loc[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff1f5650-1e07-44bb-a2b5-f9b215ac0c45",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
