---
title: "DMI_ClinVar_AF"
author: "JesusAV"
date: "2025-01-16"
output: html_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T, message = F, warning = F)
```

```{r Packages, echo=F}
library(tidyverse)
library(gridExtra)
library(knitr) 
library(dplyr)
library(ggstats)
library(ggplot2)
library(ggvenn) 
library(ggbeeswarm)
library(ggVennDiagram)
library(corrplot)
library(colorspace)
library(igraph)
library(networkD3)
library(tools)
library(ggh4x)
library(scales)
```


```{r Palettes}
colors_Var <-c("#56B4E9","#E69F00","#d52221","#999999")
colors_BIN <- c("darkgrey","#006AB4")
colors_DMI <- c("darkgrey","#18974C")
colors_blues <- c("#CADCE8","#66A9D6","#4E93C1","#006AB4","#00548C")
colors_ELMtypes <- c("#734595","#A1BE1F","#F49E17","#3B6FB6","#D41645","#F4C61F")
colors_BRET <- c("grey40", "green", "magenta")
# https://www.datanovia.com/en/blog/awesome-list-of-657-r-color-names/
```

```{r Data_maps}
amino_acid_map <- c(
  Ala = "A", Arg = "R", Asn = "N", Asp = "D", 
  Cys = "C", Gln = "Q", Glu = "E", Gly = "G",
  His = "H", Ile = "I",  Leu = "L", Lys = "K", 
  Met = "M", Phe = "F", Pro = "P",  Ser = "S", 
  Thr = "T", Trp = "W", Tyr = "Y", Val = "V"
  )

elm_categories <- c("Cleavage", "Degrons","Subst. recog.", "Scaffolding", "PTM sites", "Localization")
strc_categories <- c("D. interface", "D. buried","D. surface", "M. interface", "M. not int.")
```


```{r Figure1_Data_Tables}
# Load all ClinVar variant
ClinVar_mutations <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/All_DMIs-All_mutations/DDD_and_clinvar_mutations_2023_release.csv", col_names = T)

# Load per residue protein disorder scores
Disorder_pLDDT <- read_tsv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/AFDB_HumanReferenceProteome/proteins_pLDDT_values_byResidue.txt", col_names = T)
Disorder_IUPred <- read_tsv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/AFDB_HumanReferenceProteome/CANO_ISO_SEQUENCES_IUPRed_LONG_disorder_byResidue.txt", col_names = T) %>% rename(UniProt_ID=ID, Residue_n=Residue_N)

# Edit ClinVar table
ClinVar_mutations <- ClinVar_mutations %>% select(uniprot_id, gene_symbol, aa_change, aa_pos, pathogenicity) %>% unique() # Keep columns with relevant information (excluding nucleotide information)
ClinVar_mutations$pathogenicity <- toTitleCase(ClinVar_mutations$pathogenicity) # Capitalize pathogenicity categories
ClinVar_mut_dis <-  ClinVar_mutations %>% rename(UniProt_ID = uniprot_id, Residue_n = aa_pos) %>% # Rename columns
  mutate(Residue= amino_acid_map[gsub("[0-9].*?$","",aa_change)] ) # Change variant format

# Combine ClinVar and disorder tables 
ClinVar_mut_dis <- left_join(ClinVar_mut_dis,Disorder_pLDDT,by=c("UniProt_ID","Residue","Residue_n"))
ClinVar_mut_dis <- left_join(ClinVar_mut_dis,Disorder_IUPred,by=c("UniProt_ID","Residue","Residue_n"))
rm(Disorder_pLDDT,Disorder_IUPred)

# Summarize disorder values for long proteins, which have values in different AFDB fragments
ClinVar_mut_dis <- ClinVar_mut_dis %>% group_by(UniProt_ID, gene_symbol, aa_change, Residue_n,pathogenicity,Residue ) %>% 
   summarise(avg_pLDDT = mean(pLDDT_value, na.rm=TRUE), avg_IUPred = mean(IUPred_long, na.rm=TRUE)) %>% ungroup()

#Disorder Thresholds
pLDDT_thrs <- 70 # as in AFDB includes Low and very Low # also by further reference listed at manuscript
IUPred_thrs <- 0.4 # as in Davey et al. 

# Use thresholds to categorise variants into Order and Disorder 
ClinVar_mut_dis <- ClinVar_mut_dis %>% mutate(dis_pLDDT=case_when(avg_pLDDT < pLDDT_thrs ~ 1, avg_pLDDT >= pLDDT_thrs ~ 0  )) %>% # using pLDDT values
  mutate(dis_IUPred=case_when(avg_IUPred >= IUPred_thrs ~ 1, avg_IUPred < IUPred_thrs ~ 0 )) # using IUPred scores

```

```{r Figure1_Data_ChiSqr}
# Contingency tables for Order - Disorder enrichments across pathogenicities

# pLDDT contingency table
Pathogenicity_pLDDT_table <- ClinVar_mut_dis %>% filter(!is.na(dis_pLDDT)) %>% count(pathogenicity, dis_pLDDT) %>% 
  mutate(Category=case_when(dis_pLDDT==1 ~ "Disorder", dis_pLDDT==0 ~ "Order")) %>% select(-dis_pLDDT)  %>% pivot_wider(names_from = Category, values_from = n, values_fill=0 ) 

Pathogenicity_pLDDT_dt <- as.table(as.matrix(Pathogenicity_pLDDT_table[,2:3]))
rownames(Pathogenicity_pLDDT_dt) <-Pathogenicity_pLDDT_table$pathogenicity
Pathogenicity_pLDDT_chisq <- chisq.test(Pathogenicity_pLDDT_dt)

Pathogenicity_pLDDT_PU_dt <- Pathogenicity_pLDDT_dt[3:4,] # Only VUS and Pathogenic
Pathogenicity_pLDDT_PU_chisq <- chisq.test(Pathogenicity_pLDDT_PU_dt)

# UIPred contingency table
Pathogenicity_IUPred_table <- ClinVar_mut_dis %>% filter(!is.na(dis_IUPred)) %>% count(pathogenicity, dis_IUPred) %>% 
  mutate(Category=case_when(dis_IUPred==1 ~ "Disorder", dis_IUPred==0 ~ "Order")) %>% select(-dis_IUPred)  %>% pivot_wider(names_from = Category, values_from = n, values_fill=0 ) 

Pathogenicity_IUPred_dt <- as.table(as.matrix(Pathogenicity_IUPred_table[,2:3]))
rownames(Pathogenicity_IUPred_dt) <-Pathogenicity_IUPred_table$pathogenicity
Pathogenicity_IUPred_chisq <- chisq.test(Pathogenicity_IUPred_dt)

Pathogenicity_IUPred_PU_dt <- Pathogenicity_IUPred_dt[3:4,] # Only VUS and Pathogenic
Pathogenicity_IUPred_PU_chisq <- chisq.test(Pathogenicity_IUPred_PU_dt)
```

```{r Figure1_SupData_AlphaMissense}
#The reference AM table should be extended to include all ClinVar proteins and not only the one from the DMI predictions
AM_ClinVar_proteins <- read_tsv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/AFDB_HumanReferenceProteome/AlphaMissense/Chunks/DMI_AM_Complete.tsv", col_names = F)

# Edit table 
colnames(AM_ClinVar_proteins) <- c("UniProt_ID", "AA_change","AM_value", "AM_pathogenicity")
AM_ClinVar_proteins <- AM_ClinVar_proteins %>% mutate(AM_pathogenicity=case_when(AM_pathogenicity=="benign" ~ "Benign",
                                                                                 AM_pathogenicity=="ambiguous" ~ "Ambiguous",
                                                                                 AM_pathogenicity=="pathogenic" ~ "Pathogenic" ))
#AM_ClinVar_proteins$AM_pathogenicity <- toTitleCase(AM_ClinVar_proteins$AM_pathogenicity) # Capitalize pathogenicity categories

# Combine AlphaMissense table with ClinVar one
ClinVar_mut_dis <- ClinVar_mut_dis %>% mutate(R_variant=amino_acid_map[gsub("^.*?_","",gsub("[0-9]{1,5}","_",aa_change))] ) %>% 
  mutate(AA_change=paste(Residue,Residue_n,R_variant, sep=""))
ClinVar_mut_dis <- left_join(ClinVar_mut_dis, AM_ClinVar_proteins, by=c("UniProt_ID", "AA_change"))
ClinVar_mut_dis$AM_pathogenicity <- ordered(ClinVar_mut_dis$AM_pathogenicity, levels = c("Benign","Ambiguous","Pathogenic"))



```

```{r Figure1_Data_DMIs}
# Load all DMI predictions
All_DMIs <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/All_DMIs-All_mutations/HuRI_BioPlex_lit17_union.csv")
All_ELM_classes <- All_DMIs %>% select(Elm) %>% unique() %>% mutate(ELM_type=gsub("_.*?$","",Elm))
All_ELM_types <-  All_ELM_classes %>% count(ELM_type)

# Keep only HuRI ones
HuRI_DMIs <- All_DMIs %>% filter(Source_HuRI=="1") %>% mutate(ELM_type=gsub("_.*?$","", Elm)) 
HuRI_DMIs <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_DMIs-ClinVar_mutations/HuRI_human.csv")

# Filter HuRI DMI table to get scores
HuRI_DMI_scores <- HuRI_DMIs %>% select(intx_ID, SLiMProtein, Accession, SLiMMatch, DomainProtein, DomainID1, DomainMatch1, DMIMatchScore)

HuRI_DMI_scores %>% filter(DMIMatchScore >= 0.7) %>% nrow()
HuRI_DMI_scores %>% filter(DMIMatchScore >= 0.7) %>% select(intx_ID) %>% unique() %>% nrow()
(HuRI_DMI_scores %>% filter(DMIMatchScore >= 0.7) %>% nrow()) /( HuRI_DMI_scores %>% nrow())
```

```{r Figure1_Data_RRS4}
# Load all RRS4 data
RRS1_ROC <- read_tsv(file="/Volumes/imb-luckgr/projects/dmi_predictor/ML_items/RRS/RRSv1/RRSv1_avg_ROC_scores.txt", col_names = T, skip = 1)
RRS2_ROC <- read_tsv(file="/Volumes/imb-luckgr/projects/dmi_predictor/ML_items/RRS/RRSv2/RRSv2_avg_ROC_scores.txt", col_names = T, skip = 1)
RRS3_ROC <- read_tsv(file="/Volumes/imb-luckgr/projects/dmi_predictor/ML_items/RRS/RRSv3/RRSv3_avg_ROC_scores.txt", col_names = T, skip = 1)
RRS4_ROC <- read_tsv(file="/Volumes/imb-luckgr/projects/dmi_predictor/ML_items/RRS/RRSv4/RRSv4_avg_ROC_scores.txt", col_names = T, skip = 1)
RRS1_ROC$RRS <- "RRS1"
RRS2_ROC$RRS <- "RRS2"
RRS3_ROC$RRS <- "RRS3"
RRS4_ROC$RRS <- "RRS4"
RRS_ROC <- rbind(RRS4_ROC,rbind(RRS3_ROC,rbind(RRS1_ROC,RRS2_ROC)))
rm(RRS1_ROC,RRS2_ROC,RRS3_ROC,RRS4_ROC)

RRS1_PRC <- read_tsv(file="/Volumes/imb-luckgr/projects/dmi_predictor/ML_items/RRS/RRSv1/RRSv1_avg_PR_scores.txt", col_names = T, skip = 1)
RRS2_PRC <- read_tsv(file="/Volumes/imb-luckgr/projects/dmi_predictor/ML_items/RRS/RRSv2/RRSv2_avg_PR_scores.txt", col_names = T, skip = 1)
RRS3_PRC <- read_tsv(file="/Volumes/imb-luckgr/projects/dmi_predictor/ML_items/RRS/RRSv3/RRSv3_avg_PR_scores.txt", col_names = T, skip = 1)
RRS4_PRC <- read_tsv(file="/Volumes/imb-luckgr/projects/dmi_predictor/ML_items/RRS/RRSv4/RRSv4_avg_PR_scores.txt", col_names = T, skip = 1)
RRS1_PRC$RRS <- "RRS1"
RRS2_PRC$RRS <- "RRS2"
RRS3_PRC$RRS <- "RRS3"
RRS4_PRC$RRS <- "RRS4"
RRS_PRC <- rbind(RRS4_PRC,rbind(RRS3_PRC,rbind(RRS1_PRC,RRS2_PRC)))
rm(RRS1_PRC,RRS2_PRC,RRS3_PRC,RRS4_PRC)
```

```{r Figure1_Data_DMIscores}
# Count number of DMIs based on an increasing trheshold filter
DMIscores_count <- tibble(bin=seq(from = 0, to = 1, by = 0.05), count_dmi=0, count_ppi=0)

cc_v <- c() ## DMI count
pp_v <- c() ## PPI count

for(i in seq(from = 0, to = 1, by = 0.05)){
  cc <- HuRI_DMI_scores %>% filter(DMIMatchScore >= i) %>% nrow()
  pp <- HuRI_DMI_scores %>% filter(DMIMatchScore >= i) %>% select(intx_ID) %>% unique() %>% nrow()
  cc_v <- c(cc_v, cc)
  pp_v <- c(pp_v, pp)
}

DMIscores_count$count_dmi <-cc_v 
DMIscores_count$count_ppi <-pp_v 


```

```{r Figure1_SuppData_ELMclasses}
# Load ELM instances and classes
elm_instances <- read_tsv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/ELM_Interacting_Regions_instances.tsv", col_names = T)

elm_instances <- elm_instances %>% filter(motif_protein_species=="Homo sapiens" & curators_instance_assessment=="true positive") %>% select(ELM_Class, motif_protein_uniprot_accession, motif_start_residue_number, motif_end_residue_number, domain_protein_uniprot_accession)
elm_instances$SLiMMatch <- paste(elm_instances$motif_start_residue_number, elm_instances$motif_end_residue_number, sep="-")
elm_instances <- elm_instances %>% select(-motif_start_residue_number, -motif_end_residue_number)
elm_instances$ELM_instance <- "1"
colnames(elm_instances)[c(1:3)] <- c("ELM_class", "SLiMProtein", "DomainProtein")
elm_instances 

elm_instances %>% filter(ELM_class %in% HuRI_DMIs$ELM_class) %>% select(ELM_class) %>% unique() # 87 ELM classes

DMI_ELM_instances <- HuRI_DMIs %>% filter(DMIMatchScore >= 0.7) %>% 
  select(intx_ID, SLiMProtein, Accession, ELM_class, SLiMMatch, DomainProtein, DomainID1, DomainName1, DomainMatch1, DMIMatchScore, ELM_type)

DMI_ELM_instances <- left_join(DMI_ELM_instances, elm_instances) 
DMI_ELM_instances <- DMI_ELM_instances %>% mutate(ELM_instance=case_when(is.na(ELM_instance) ~ "0", TRUE ~  ELM_instance))

elm_motifs <- elm_instances %>% select(ELM_class, SLiMProtein, SLiMMatch) %>% mutate(ELM_motif="1")
DMI_ELM_instances <- left_join(DMI_ELM_instances, elm_motifs)
DMI_ELM_instances <- DMI_ELM_instances %>% mutate(ELM_motif=case_when(is.na(ELM_motif) ~ "0", TRUE ~  ELM_motif))
DMI_ELM_instances <- DMI_ELM_instances %>% unique()

DMI_ELM_instances %>% count(ELM_instance)
DMI_ELM_instances %>% count(ELM_motif)
DMI_ELM_instances %>% filter(ELM_instance=="1")

# 
# left_join(All_DMIs, elm_instances) %>% count(ELM_instance)
# left_join(HuRI_DMIs, elm_instances) %>% count(ELM_instance)
# left_join(DMI_ELM_instances, elm_instances) %>% count(ELM_instance)
# Only 31 high confidence HuRI predicted DMIs correspond to motif-domain ELM instaces 
# Reason might include any/some of the following: 
### DMI pair is not on HuRI
### DMI doesn't have a high score  
### The whole predictions only have around 7k human proteins, high-scored ones only 1.4 K
### ELM has many instances, and classes, that are not human
# 
# left_join(DMI_ELM_instances, elm_instances, by = c("Elm", "SLiMProtein", "SLiMMatch"))  %>% count(ELM_instance)
# elm_instances %>% filter(Elm=="DOC_ANK_TNKS_1")
# DMI_ELM_instances %>% filter(Elm=="DOC_ANK_TNKS_1" & SLiMProtein=="Q7Z6K5") #456-465
# All_DMIs %>% filter(Elm=="DOC_ANK_TNKS_1" & SLiMProtein=="Q7Z6K5")




HURI_ELM_DMI_PPI_list <- list(CLV = HuRI_DMIs %>% filter(ELM_type=="CLV" & DMIMatchScore >=0.7 ) %>% select(intx_ID) %>% unique() %>% unlist(),
                              DEG = HuRI_DMIs %>% filter(ELM_type=="DEG" & DMIMatchScore >=0.7) %>% select(intx_ID) %>% unique() %>% unlist(), 
                              DOC = HuRI_DMIs %>% filter(ELM_type=="DOC" & DMIMatchScore >=0.7) %>% select(intx_ID) %>% unique() %>% unlist(), 
                              LIG = HuRI_DMIs %>% filter(ELM_type=="LIG" & DMIMatchScore >=0.7) %>% select(intx_ID) %>% unique() %>% unlist(), 
                              MOD = HuRI_DMIs %>% filter(ELM_type=="MOD" & DMIMatchScore >=0.7) %>% select(intx_ID) %>% unique() %>% unlist(), 
                              TRG = HuRI_DMIs %>% filter(ELM_type=="TRG" & DMIMatchScore >=0.7) %>% select(intx_ID) %>% unique() %>% unlist() ) 

DMI_ELM_classes <- HuRI_DMIs %>% filter(DMIMatchScore >= 0.7) %>% select(Accession, ELM_class, ELM_type) %>% unique()
```


```{r Figure2_Data}
HURI_ClinVar_mutation <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_DMIs-ClinVar_mutations/DMI_human_HuRI_overlap_clinvar_2023_release.csv")
HURI_ClinVar_mutation$fragment_type <- toTitleCase(HURI_ClinVar_mutation$fragment_type)
HURI_ClinVar_mutation$pathogenicity <- toTitleCase(HURI_ClinVar_mutation$pathogenicity)
HURI_ClinVar_mutation <-  HURI_ClinVar_mutation %>% mutate(fragment_type=case_when(fragment_type=="Slim" ~ "Motif", TRUE ~ fragment_type))

HURI_ClinVar_mut_motif <- HURI_ClinVar_mutation %>% filter(fragment_type=="Motif") %>% select(fragment_protein, fragment_id, fragment_pos, pathogenicity) %>% rename(Accession=fragment_id)
HURI_ClinVar_mut_motif <- left_join(HURI_ClinVar_mut_motif, DMI_ELM_classes, by="Accession")
```

```{r Figure2_Data_ELM_pathogenic}
HURI_ClinVar_mut_motif %>% filter(ELM_type=="DEG" & pathogenicity=="Pathogenic")
HURI_ClinVar_mutation %>% filter(fragment_type=="Motif" & pathogenicity=="Pathogenic" & fragment_id=="ELME000241")
# AFF4 (Q9UHB7) 252-260 DEG_SIAH_1 (Pro253Leu, Thr254Ala, Thr254Ser, Arg258Trp, Met260Val)
HURI_ClinVar_mut_motif %>% filter(ELM_type=="DEG" & pathogenicity=="Pathogenic")

HURI_ClinVar_mut_motif %>% filter(ELM_type=="DOC" & pathogenicity=="Pathogenic") 
DOC_prot_path <- HURI_ClinVar_mut_motif %>% filter(ELM_type=="DOC" & pathogenicity=="Pathogenic") %>% select(fragment_protein,Accession) %>% unique() 
HURI_ClinVar_mutation %>% filter(fragment_type=="Motif" & pathogenicity=="Pathogenic" & fragment_id %in% DOC_prot_path$Accession)
# 6 motifs - 6 proteins
HuRI_DMIs %>% filter(SLiMProtein=="P84022" & ELM_class=="DOC_WW_Pin1_4" & SLiMMatch=="5-10") %>% select(DomainProtein) 
HuRI_DMIs %>% filter(SLiMProtein=="P84022" & ELM_class=="DOC_WW_Pin1_4" ) %>% select(SLiMMatch, DMIMatchScore) %>% unique()
HURI_ClinVar_mut_motif %>% filter(fragment_protein=="P84022" )

HuRI_DMIs %>% filter(SLiMProtein=="O95817" & ELM_class=="DOC_MAPK_MEF2A_6") %>% select(DomainProtein) 

HURI_ClinVar_mut_motif %>% filter(ELM_type=="LIG" & pathogenicity=="Pathogenic") 
LIG_prot_path <- HURI_ClinVar_mut_motif %>% filter(ELM_type=="LIG" & pathogenicity=="Pathogenic") %>% select(fragment_protein,Accession) %>% unique() 
HURI_ClinVar_mutation %>% filter(fragment_type=="Motif" & pathogenicity=="Pathogenic" & fragment_id %in% LIG_prot_path$Accession) %>% group_by(fragment_protein)
# 6 motifs - 6 proteins


HURI_ClinVar_mut_motif %>% filter(ELM_type=="MOD" & pathogenicity=="Pathogenic") 
MOD_prot_path <- HURI_ClinVar_mut_motif %>% filter(ELM_type=="MOD" & pathogenicity=="Pathogenic") %>% select(fragment_protein,Accession) %>% unique() 
HURI_ClinVar_mutation %>% filter(fragment_type=="Motif" & pathogenicity=="Pathogenic" & fragment_id %in% MOD_prot_path$Accession) %>% group_by(fragment_protein) 
# 2 motifs - 2 proteins


HURI_ClinVar_mut_motif %>% filter(ELM_type=="TRG" & pathogenicity=="Pathogenic") 
HURI_ClinVar_mutation %>% filter(fragment_type=="Motif" & pathogenicity=="Pathogenic" & fragment_id=="ELME000120") %>% group_by(fragment_protein) 
#HuRi_DMI %>% filter(Accession=="ELME000512" & SLiMProtein=="P30279")

HuRI_DMIs %>% filter(DMIMatchScore>=0.7 & SLiMProtein=="Q9UHB7" & ELM_class=="DEG_SIAH_1" & SLiMMatch=="252-260") %>% select(DomainName1) %>% unique()
HuRI_DMIs %>% filter(DMIMatchScore>=0.7 & SLiMProtein=="P40763" & ELM_class=="LIG_SH2_STAT5" & SLiMMatch=="705-708") %>% select(DomainName1) %>% unique()
HuRI_DMIs %>% filter(DMIMatchScore>=0.7 & SLiMProtein=="P98174" & ELM_class=="LIG_SH3_1" & SLiMMatch=="186-192") %>% select(DomainName1) %>% unique()
HuRI_DMIs %>% filter(DMIMatchScore>=0.7 & SLiMProtein=="Q8WWZ3" & ELM_class=="LIG_TRAF6" & SLiMMatch=="25-33") %>% select(DomainName1) %>% unique()
```

```{r Figure2_Data2}
HuRi_DMI <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_DMIs-ClinVar_mutations/HuRI_human_highScores.csv")
HuRi_DMI <- HuRi_DMI %>% filter(DMIMatchScore>=0.7)
HURI_ClinVar_DMI <- HuRi_DMI %>% filter(SLiMProtein %in% HURI_ClinVar_mutation$fragment_protein | DomainProtein %in% HURI_ClinVar_mutation$fragment_protein)
```

```{r Figure2_Data2_DMI_VariantCount}
HURI_ClinVar_mutation_count_DMI <- HURI_ClinVar_mutation %>% count(fragment_protein,fragment_id, fragment_pos, pathogenicity) %>% pivot_wider(names_from = pathogenicity, values_from = n, values_fill = 0)

HURI_ClinVar_DMI_mut_count_DMI <- HURI_ClinVar_DMI %>% select(intx_ID, DomainProtein, DomainID1, DomainMatch1, SLiMProtein, Accession, SLiMMatch) %>% rename(fragment_protein=SLiMProtein, fragment_id=Accession, fragment_pos=SLiMMatch)

HURI_ClinVar_DMI_mut_count_DMI <- left_join(HURI_ClinVar_DMI_mut_count_DMI, HURI_ClinVar_mutation_count_DMI, by=c("fragment_protein", "fragment_id", "fragment_pos")) %>% mutate_if(is.numeric,coalesce,0) %>% rename(SLiMProtein=fragment_protein, Accession=fragment_id, SLiMMatch=fragment_pos) %>% rename(fragment_protein=DomainProtein, fragment_id=DomainID1, fragment_pos=DomainMatch1)

HURI_ClinVar_DMI_mut_count_DMI <- left_join(HURI_ClinVar_DMI_mut_count_DMI, HURI_ClinVar_mutation_count_DMI, by=c("fragment_protein", "fragment_id", "fragment_pos")) %>% rename(DomainProtein=fragment_protein, DomainID1=fragment_id, DomainMatch1=fragment_pos)

## Avoid counting twice when there are homotypic interactions
HURI_ClinVar_DMI_mut_count_DMI <- HURI_ClinVar_DMI_mut_count_DMI %>% mutate(Benign = Benign.x + Benign.y) %>%
  mutate(Uncertain = Uncertain.x + Uncertain.y) %>% mutate(Conflicting = Conflicting.x + Conflicting.y) %>%
  mutate(Pathogenic = Pathogenic.x + Pathogenic.y)


HURI_ClinVar_DMI_mut_count_DMI_long <- HURI_ClinVar_DMI_mut_count_DMI %>% select(intx_ID, DomainProtein, DomainID1, DomainMatch1, SLiMProtein, Accession, SLiMMatch, Benign, Conflicting, Pathogenic, Uncertain) %>% pivot_longer(names_to = "pathogenicity", cols = Benign:Uncertain) %>% filter(value>0) %>% unique()

```

```{r Figure2_Data2_PPI_VariantCount}
HURI_ClinVar_mutation_count <- HURI_ClinVar_mutation %>% count(fragment_protein,pathogenicity) %>% pivot_wider(names_from = pathogenicity, values_from = n, values_fill = 0)

HURI_ClinVar_DMI_mut_count <- HURI_ClinVar_DMI %>% select(intx_ID, SLiMProtein,DomainProtein) %>% rename(fragment_protein=SLiMProtein)

HURI_ClinVar_DMI_mut_count <- left_join(HURI_ClinVar_DMI_mut_count,HURI_ClinVar_mutation_count, by="fragment_protein") %>% mutate_if(is.numeric,coalesce,0) %>% rename(SLiMProtein=fragment_protein) %>% rename(fragment_protein=DomainProtein)

HURI_ClinVar_DMI_mut_count <- left_join(HURI_ClinVar_DMI_mut_count,HURI_ClinVar_mutation_count, by="fragment_protein") %>% rename(DomainProtein=fragment_protein)

## Avoid counting twice when there are homotypic interactions
HURI_ClinVar_DMI_mut_count <- HURI_ClinVar_DMI_mut_count %>% 
  mutate(benign=case_when(SLiMProtein==DomainProtein ~  Benign.x, TRUE ~ Benign.x + Benign.y))  %>% 
  mutate(uncertain=case_when(SLiMProtein==DomainProtein ~  Uncertain.x, TRUE ~ Uncertain.x + Uncertain.y)) %>% 
  mutate(conflicting=case_when(SLiMProtein==DomainProtein ~ Conflicting.x, TRUE ~ Conflicting.x + Conflicting.y)) %>% 
  mutate(pathogenic=case_when(SLiMProtein==DomainProtein ~  Pathogenic.x, TRUE ~ Pathogenic.x + Pathogenic.y))


HURI_ClinVar_DMI_mut_count_long <- HURI_ClinVar_DMI_mut_count %>% select(intx_ID, benign,conflicting , pathogenic, uncertain) %>% pivot_longer(names_to = "pathogenicity", cols = benign:uncertain) %>% filter(value>0) %>% unique()
HURI_ClinVar_DMI_mut_count_long$pathogenicity <- toTitleCase(HURI_ClinVar_DMI_mut_count_long$pathogenicity)

HURI_ClinVar_DMI_mut_count_long_B <- HURI_ClinVar_DMI_mut_count_long %>% filter(pathogenicity=="Benign") %>% select(intx_ID) %>% unique() %>% unlist()
HURI_ClinVar_DMI_mut_count_long_C <- HURI_ClinVar_DMI_mut_count_long %>% filter(pathogenicity=="Conflicting") %>% select(intx_ID) %>% unique() %>% unlist()
HURI_ClinVar_DMI_mut_count_long_P <- HURI_ClinVar_DMI_mut_count_long %>% filter(pathogenicity=="Pathogenic") %>% select(intx_ID) %>% unique() %>% unlist()
HURI_ClinVar_DMI_mut_count_long_U <- HURI_ClinVar_DMI_mut_count_long %>% filter(pathogenicity=="Uncertain") %>% select(intx_ID) %>% unique() %>% unlist()

HURI_ClinVar_DMI_PPI_list <- list(Benign = HURI_ClinVar_DMI_mut_count_long_B, 
                                  Conflicting = HURI_ClinVar_DMI_mut_count_long_C, 
                                  Pathogenic = HURI_ClinVar_DMI_mut_count_long_P, 
                                  Uncertain = HURI_ClinVar_DMI_mut_count_long_U) 
#length(HURI_ClinVar_DMI_mut_count_long_P)
```

```{r Figure2_Data_Mutations_fragment_tables}
Fragment_contingency_tb <- data.frame(motif=c(0,0,0,0),
                                  domain=c(0,0,0,0),
                                  other=c(0,0,0,0),
                                  total=c(0,0,0,0))
rownames(Fragment_contingency_tb) <- c(" benign", "conflicting", "pathogenic", "uncertain")

Fragment_contingency_tb$motif <- HURI_ClinVar_mutation %>% filter(fragment_type=="Motif") %>% select(pathogenicity) %>% table() %>% as.vector()
Fragment_contingency_tb$domain <- HURI_ClinVar_mutation %>% filter(fragment_type=="Domain") %>% select(pathogenicity) %>% table() %>% as.vector()

DMI_proteins <- HURI_ClinVar_mutation %>% select(fragment_protein) %>% unique() %>% unlist()
Fragment_contingency_tb$total <- ClinVar_mutations %>% filter(uniprot_id %in% DMI_proteins) %>% select(pathogenicity) %>% table() %>% as.vector()
Fragment_contingency_tb$other <- Fragment_contingency_tb$total - Fragment_contingency_tb$motif - Fragment_contingency_tb$domain
Fragment_contingency_tb


Motif_simple_tb <- as.data.frame(Fragment_contingency_tb[c(1,3),1])
colnames(Motif_simple_tb) <- "motif"
Motif_simple_tb$non_motif <- unlist(Fragment_contingency_tb[c(1,3),2] + Fragment_contingency_tb[c(1,3),3])
row.names(Motif_simple_tb) <- c("benign", "pathogenic")
Motif_simple_tb

Domain_simple_tb <- as.data.frame(Fragment_contingency_tb[c(1,3),2])
colnames(Domain_simple_tb) <- "domain"
Domain_simple_tb$non_domain <- unlist(Fragment_contingency_tb[c(1,3),1] + Fragment_contingency_tb[c(1,3),3])
row.names(Domain_simple_tb) <- c("benign", "pathogenic")
Domain_simple_tb

Fragment_contingency_dt <- as.table(as.matrix(Fragment_contingency_tb[,1:3]))
Fragment_contingency_chisq <- chisq.test(Fragment_contingency_dt)
```


```{r Figure3_data_BRET}
BRET_results <- read_tsv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/BRET_experiments/titration_cat_scoring_v6_with_include.tsv")
BRET_results <- BRET_results %>% mutate(mut_type=case_when(mCit_plasmid=="pcDNA3.1 mCit-His3C-MYD88_134_138del" ~ "Motif validation", TRUE ~ mut_type))
BRET_results <- BRET_results %>% mutate(include=case_when(NL_plasmid=="pcDNA3.1 cmyc-NL-IQCB1_N273Y" &  mCit_symbol=="MNS1" & project_id=="Lu153r01" ~ "1",
                                                          NL_plasmid=="pcDNA3.1 cmyc-NL-SPOP_S119L" &  mCit_symbol=="MYD88" & project_id=="Lu171r01" ~ "0",
                                                          NL_plasmid=="pcDNA3.1 cmyc-NL-SPOP_R70T" &  mCit_symbol=="RXRB" & project_id=="Lu169r01" ~ "0" , TRUE ~ include))
BRET_results <- BRET_results %>% filter(include=="1")
BRET_results <- BRET_results %>% mutate(exp_category=case_when(mut_type=="Domain validation" | mut_type=="Motif validation" ~ "interface", TRUE ~ "variant"))

#Significance labels
BRET_results <- BRET_results %>%  
  mutate(tstat_bret50 = as.double(gsub("\\,","\\.",tstat_bret50)) ) %>% mutate(tstat_cBRET = as.double(gsub("\\,","\\.",tstat_cBRET)) ) %>% 
  mutate(pval_bret50 = as.double(gsub("\\,","\\.",pval_bret50)) ) %>% mutate(pval_cBRET = as.double(gsub("\\,","\\.",pval_cBRET)) ) %>% 
  mutate(sig_BRET50=case_when(pval_bret50>0.05 ~ "",  pval_bret50 <= 0.05 & pval_bret50>0.01 ~ "*", pval_bret50 <= 0.01 & pval_bret50>0.001 ~ "**",pval_bret50 <=  0.001 ~ "***")) %>%
  mutate(sig_maxBRET=case_when(pval_cBRET>0.05 ~ "",  pval_cBRET <= 0.05 & pval_cBRET>0.01 ~ "*", pval_cBRET <= 0.01 & pval_cBRET>0.001 ~ "**",pval_cBRET <=  0.001 ~ "***"))

BRET_results <- BRET_results %>% mutate(BRET_category=case_when(num_use_fit_WT==3 & num_use_fit_MUT==3 ~ "BRET50", 
                                                                num_repl_include_MUT==3 & num_repl_include_MUT==3 ~ "maxBRET",
                                                                TRUE ~ "other"))

domain_BRET_categories <- BRET_results %>% select(mCit_symbol) %>% unique()
domain_BRET_categories$category <- as.factor(c("candidate", "positive", "negative", 
                                     "positive", "candidate", "negative",
                                     "candidate","negative",
                                     "positive", "candidate",
                                     "candidate", "candidate","positive","negative","negative"))

domain_BRET <- BRET_results %>% filter(mut_type=="Domain validation") %>% 
  mutate(mutant=gsub("^pc.*?NL-", "", NL_plasmid)) %>% # Mutant labels
  mutate(residue=as.numeric(gsub("[A-Z]", "", gsub("^.*?_", "", mutant)))) %>% # Mutant index
  mutate(tstat_bret50=case_when(tstat_bret50 < -20 ~ -20, tstat_bret50 > 20 ~ 20, TRUE ~ tstat_bret50), 
         tstat_cBRET=case_when(tstat_cBRET < -20 ~ -20, tstat_cBRET > 20 ~ 20, TRUE ~ tstat_cBRET)) # Capping values
domain_BRET <- domain_BRET %>% select(mCit_plasmid,NL_symbol, mCit_symbol, pval_bret50, tstat_bret50, pval_cBRET, tstat_cBRET, sig_BRET50, sig_maxBRET, BRET_category, mutant, residue)
domain_BRET <- left_join(domain_BRET, domain_BRET_categories)
domain_BRET$category <- factor(domain_BRET$category, levels=c("negative", "positive", "candidate"))



motif_BRET <- BRET_results %>% filter(mut_type=="Motif validation") %>% 
  mutate(mutant=gsub("^pc.*?His3C-", "", mCit_plasmid)) %>%  # Mutant labels
  mutate(residue=as.numeric(gsub("[A-Z]", "", gsub("_.*?$", "",gsub("^.*?_", "", mutant))))) %>% # Mutant index
  mutate(tstat_bret50=case_when(tstat_bret50 < -20 ~ -20, tstat_bret50 > 20 ~ 20, TRUE ~ tstat_bret50), 
         tstat_cBRET=case_when(tstat_cBRET < -20 ~ -20, tstat_cBRET > 20 ~ 20, TRUE ~ tstat_cBRET)) # Capping values
motif_BRET <- motif_BRET %>% select(NL_plasmid, NL_symbol, mCit_symbol, pval_bret50, tstat_bret50, pval_cBRET, tstat_cBRET, sig_BRET50, sig_maxBRET, BRET_category, mutant, residue)
motif_BRET <- left_join(motif_BRET, domain_BRET_categories)
motif_BRET$category <- factor(motif_BRET$category, levels=c("negative", "positive", "candidate"))



mutants_BRET <- BRET_results %>% filter(exp_category=="variant") %>% 
  mutate(domain_variant=str_detect(NL_plasmid, "_[A-Z][0-9]{1,4}[A-Z]"), motif_variant=str_detect(mCit_plasmid, "_[A-Z][0-9]{1,4}[A-Z]")) %>% # mutant labels
  mutate(mutant=case_when(domain_variant==TRUE ~ gsub("^pc.*?NL-", "", NL_plasmid), 
                          motif_variant==TRUE ~ gsub("^pc.*?His3C-", "", mCit_plasmid) )) %>% 
  mutate(mutant=case_when(domain_variant==TRUE ~ gsub("_", " ",mutant), motif_variant==TRUE ~ gsub("_", " ",mutant))) %>% 
  mutate(residue=as.numeric(gsub("[A-Z]", "", gsub("^.*? ", "", mutant)))) %>% # Mutant index
  mutate(tstat_bret50=case_when(tstat_bret50 < -20 ~ -20, tstat_bret50 > 20 ~ 20, TRUE ~ tstat_bret50), 
         tstat_cBRET=case_when(tstat_cBRET < -20 ~ -20, tstat_cBRET > 20 ~ 20, TRUE ~ tstat_cBRET)) # Capping values
mutants_BRET <- mutants_BRET %>% select(NL_plasmid, mCit_plasmid,NL_symbol, mCit_symbol, pval_bret50, tstat_bret50, pval_cBRET, tstat_cBRET, sig_BRET50, sig_maxBRET, BRET_category, motif_variant, domain_variant, mutant, residue)

mutants_BRET_categories <- mutants_BRET %>% select(mutant) %>% unique()

BRET_mutants <- read_tsv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/BRET_experiments/tested_disease_mutations_structure_computations_extended.txt")
Clinvar_mutants <- BRET_mutants %>% 
  mutate(mutant=case_when(mutation_location=="domain" ~ paste(domain_partner,aa_change, sep=" "), 
                                                            TRUE ~ paste(motif_partner,aa_change, sep=" "))) %>%
  select(mutant, ClinVar) %>% unique()
mutants_BRET_categories <- left_join(mutants_BRET_categories, Clinvar_mutants)
mutants_BRET <- left_join(mutants_BRET, mutants_BRET_categories)
mutants_BRET <- left_join(mutants_BRET, domain_BRET_categories)
mutants_BRET$category <- factor(mutants_BRET$category, levels=c("negative", "positive", "candidate"))
mutants_BRET <- mutants_BRET %>% mutate(ClinVar=case_when(mutant=="IKZF1 S41L" ~ "Uncertain", 
                                          mutant=="SPOP Y87C" ~ "Uncertain", TRUE ~ ClinVar))
#CALM1 D94H
BRET_results %>% filter(NL_symbol=="CTBP1" & mCit_symbol=="DMRTB1") %>% arrange(project_id) %>% select(NL_plasmid, mCit_plasmid, tstat_bret50, tstat_cBRET, sig_BRET50, sig_maxBRET)

BRET_results %>% filter(NL_symbol=="SPOP" &  mCit_symbol=="MYD88" ) %>% select(NL_plasmid, mCit_plasmid, project_id, include)
```

```{r Figure3_data_titration}
BRET_selection <- read_csv(file="/Users/jesusav/Desktop/titrations_filtering.csv")
BRET_selection <- BRET_selection %>% filter(include==1 & use_fit==1) %>% select(-include, -use_fit, -comment)
BRET_selection$repl_id <- as.character(BRET_selection$repl_id)

BRET_plasmid_layout <- read_csv(file="/Users/jesusav/Desktop/plate_layout.csv")
BRET_plasmid_info <- BRET_plasmid_layout %>% 
  select(project_id, repl_id, NL_plasmid, mCit_plasmid, NL_plasmid_id, mCit_plasmid_id) %>% unique() %>%
  filter(NL_plasmid!="empty" & mCit_plasmid!="empty")
BRET_plasmid_info <- BRET_plasmid_info %>% filter(project_id %in% BRET_selection$project_id)


BRET_plasmid_info %>% filter(project_id=="Lu143r01")
BRET_values <- read_csv(file="/Users/jesusav/Desktop/titration_values.csv")
BRET_values$ratio <- BRET_values$fluo / BRET_values$lumi
BRET_values$repl_id <- as.character(BRET_values$repl_id)

BRET_values_info <- left_join(BRET_values,BRET_plasmid_info)
BRET_values_info <- BRET_values_info %>% mutate(NL_construct= gsub("pcDNA3.1 cmyc-NL-", "", NL_plasmid),
                                                          mCit_construct = gsub("pcDNA3.1 mCit-His3C-", "", mCit_plasmid))
BRET_values_info <- BRET_values_info %>% separate(NL_construct, c("NL_protein","NL_mutation"), sep="_", remove=F) %>% 
  mutate(NL_mutation=case_when(is.na(NL_mutation) ~ NA, !is.na(NL_mutation) ~ gsub("^.*?_", "", NL_construct) )) %>% 
  separate(mCit_construct, c("mCit_protein","mCit_mutation"), sep="_", remove=F) %>% 
  mutate(mCit_mutation=case_when(is.na(mCit_mutation) ~ NA, !is.na(mCit_mutation) ~ gsub("^.*?_", "", mCit_construct) )) %>% 
  mutate(test=case_when(is.na(NL_mutation) & is.na(mCit_mutation) ~ "WT", TRUE ~ "mutant"))

BRET_fits <- read_csv(file="/Users/jesusav/Desktop/titration_fit.csv")

BRET_selection_values <- left_join(BRET_selection, BRET_values_info)
BRET_mutant_pairs <- BRET_selection_values %>% filter(test=="mutant") %>% select(project_id, NL_plasmid_id, mCit_plasmid_id) %>% unique()

BRET_selection_values <- left_join(BRET_selection, BRET_values_info)

BRET_mutant_pairs <- BRET_results %>% select(project_id, NL_plasmid_id, mCit_plasmid_id,BRET_category) %>% unique()
BRET_mutant_pairs <- left_join(BRET_mutant_pairs, BRET_values_info %>% select(project_id, NL_plasmid_id, mCit_plasmid_id,NL_construct, mCit_construct, NL_protein, mCit_protein, test)) %>% unique()
```


```{r Figure4_data1}
BRET_mutants <- read_tsv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/BRET_experiments/tested_disease_mutations_structure_computations_extended.txt")

BRET_mutants <- BRET_mutants %>% mutate(AlphaMissense=case_when(AlphaMissense=="likely pathogenic"  ~ "Pathogenic",
                                                                AlphaMissense=="uncertain"  ~ "Ambiguos",
                                                                AlphaMissense=="likely benign"  ~ "Benign", TRUE ~ AlphaMissense))

BRET_mutants <- BRET_mutants %>% mutate(Fragment=case_when(mutation_location=="domain"  ~ "Domain",
                                                                mutation_location=="outside motif" |  mutation_location=="core motif" ~ "Motif"))

BRET_mutants <- BRET_mutants %>% mutate(interface = case_when(pLDDT >= 80 & deltaRSA<(-0.05)  ~ "interface",
                                                                TRUE ~ "not_interface"))  %>% 
  mutate(structural_region = case_when(Fragment=="Domain" & interface == "interface"  ~ "d_interface",
                                       Fragment=="Domain" & interface == "not_interface" & RSA>0.36 ~ "d_surface",
                                       Fragment=="Domain" & interface == "not_interface" & RSA<=0.36 ~ "d_buried",
                                       Fragment=="Motif" & interface == "interface" ~ "m_interface",
                                       Fragment=="Motif" & interface == "not_interface" ~ "m_n_interface",
                                       TRUE ~ "other" ) )

BRET_mutants$structural_region <- ordered(BRET_mutants$structural_region, levels = c("d_interface", "d_buried","d_surface", "m_interface","m_n_interface"))  
BRET_mutants$AlphaMissense <- ordered(BRET_mutants$AlphaMissense, levels = c("Benign", "Ambiguos","Pathogenic")) 

BRET_mutants <- BRET_mutants %>% filter(!(motif_partner=="LITAF" & (aa_change=="P58L" |aa_change=="P59R" |aa_change=="Y61C"  )))

BRET_mutants_motif <-  BRET_mutants %>% filter(mutation_location!="domain") %>% mutate(mutation_location=case_when(mutation_location=="core motif" ~ "Core",
                                                                                                                   mutation_location=="outside motif" ~ "Flank"))
BRET_mutants %>% filter(structural_region=="d_interface")

BRET_mutants %>% filter(structural_region=="d_surface")

#BRET_mutants_update <- mutants_BRET %>% mutate(exp_conclusion=case_when(pval_bret50 <= 0.05 | pval_cBRET <= 0.05 ~ "disrupting", TRUE ~ "not_disrupting")) %>%
#  mutate(fragment_variant=case_when(motif_variant==TRUE ~ "motif", domain_variant==TRUE ~ "domain")) %>%
#  select(-NL_plasmid, -mCit_plasmid, -motif_variant, -domain_variant) %>% separate(mutant, c("mut_protein", "aa_change"), sep=" ")
  
#write_tsv(BRET_mutants_update, file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/BRET_experiments/tested_disease_mutations_structure_computations_extended_2.tsv")

BRET_mutants_update <- read_tsv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/BRET_experiments/tested_disease_mutations_structure_computations_extended_2.tsv")

BRET_mutants_update <- BRET_mutants_update %>% mutate(AlphaMissense=case_when(AlphaMissense=="likely pathogenic"  ~ "Pathogenic",
                                                                AlphaMissense=="uncertain"  ~ "Ambiguos",
                                                                AlphaMissense=="likely benign"  ~ "Benign", TRUE ~ AlphaMissense))

BRET_mutants_update <- BRET_mutants_update %>% mutate(Fragment=case_when(mutation_location=="domain"  ~ "Domain",
                                                                mutation_location=="outside motif" |  mutation_location=="core motif" ~ "Motif"))

BRET_mutants_update <- BRET_mutants_update %>% mutate(interface = case_when(pLDDT >= 80 & deltaRSA<(-0.05)  ~ "interface", TRUE ~ "not_interface"))  %>% 
  mutate(interface = case_when( mutation_location=="outside motif" ~ "not_interface", TRUE ~ interface)) %>% 
  mutate(structural_region = case_when(Fragment=="Domain" & interface == "interface"  ~ "d_interface",
                                       Fragment=="Domain" & interface == "not_interface" & RSA>0.36 ~ "d_surface",
                                       Fragment=="Domain" & interface == "not_interface" & RSA<=0.36 ~ "d_buried",
                                       Fragment=="Motif" & interface == "interface" ~ "m_interface",
                                       Fragment=="Motif" & interface == "not_interface" ~ "m_n_interface",
                                       TRUE ~ "other" ) )

BRET_mutants_update$structural_region <- ordered(BRET_mutants_update$structural_region, levels = c("d_interface", "d_buried","d_surface", "m_interface","m_n_interface"))  
BRET_mutants_update$AlphaMissense <- ordered(BRET_mutants_update$AlphaMissense, levels = c("Benign", "Ambiguos","Pathogenic")) 
BRET_mutants_update_motif <-  BRET_mutants_update %>% filter(mutation_location!="domain") %>%
  mutate(mutation_location=case_when(mutation_location=="core motif" ~ "Core", mutation_location=="outside motif" ~ "Flank"))

BRET_mutants_update %>% filter(mCit_symbol=="CALM1")
```


```{r Figure5_data1}
DMI_AF_metrics <- read_csv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/HuRI_DMI_AF_modelling/Score_files/DMI_AF_model_metrics.csv")
#Q08209_SM00156_O_10_373.Q96KS9_ELME000237_D_3_9 DOC
#Q08999_PF01857_O_830_1030.Q99708_ELME000007_D_152_169 LIG
DMI_AF_metrics <- DMI_AF_metrics %>% mutate(ELM_type=case_when(DMI_model=="Q08209_SM00156_O_10_373.Q96KS9_ELME000237_D_3_9" ~ "DOC",
                                                               DMI_model=="Q08999_PF01857_O_830_1030.Q99708_ELME000007_D_152_169" ~ "LIG",
                                                               TRUE ~ ELM_type))

DMI_AF_class_table <- DMI_AF_metrics  %>% filter(model_score>=0.6) %>% select(intx_ID, ELM_type) %>% unique()

DMI_AF_metrics %>% filter(model_score>=0.6) %>% nrow()
DMIs_hq <- DMI_AF_metrics %>% filter(model_score>=0.6) %>% select(DMI_model) %>% unlist()
DMI_AF_mutations <- read_tsv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/HuRI_DMI_AF_modelling/Score_files/DMI_AF_model_metrics_mappedMutations.tsv")
DMI_AF_mut_hq <- DMI_AF_mutations %>% filter(DMI %in% DMIs_hq)
DMI_AF_mut_hq <- DMI_AF_mut_hq %>% mutate(interface = case_when(pLDDT >= 80 & Acc_change<(-0.05)  ~ "interface",
                                                                TRUE ~ "not_interface"))  %>% 
  mutate(structural_region = case_when(Fragment=="Domain" & interface == "interface"  ~ "d_interface",
                                       Fragment=="Domain" & interface == "not_interface" & Norm_Acc>0.36 ~ "d_surface",
                                       Fragment=="Domain" & interface == "not_interface" & Norm_Acc<=0.36 ~ "d_buried",
                                       Fragment=="Motif"  & interface == "interface"  ~ "m_interface",
                                       Fragment=="Motif"  & interface == "not_interface" ~ "m_non_interface",
                                       TRUE ~ "other" ) )

structural_region <- c("d_interface", "d_buried", "d_surface","m_interface", "m_non_interface")
DMI_AF_mut_hq$structural_region <- ordered(DMI_AF_mut_hq$structural_region, levels = structural_region)
DMI_AF_mut_hq$AM_pathogenicity <- ordered(DMI_AF_mut_hq$AM_pathogenicity, levels = c("AM_benign","AM_ambiguous","AM_pathogenic"))

DMI_AF_mut_hq$Pathogenicity <- toTitleCase(DMI_AF_mut_hq$Pathogenicity)

DMI_AF_mut_hq_table <- DMI_AF_mut_hq %>% filter(structural_region!="other") %>% count(Pathogenicity, structural_region) %>% pivot_wider(names_from = structural_region, values_from = n, values_fill=0 )

DMI_AF_mut_hq_dt <- as.table(as.matrix(DMI_AF_mut_hq_table[,2:6]))
rownames(DMI_AF_mut_hq_dt) <-DMI_AF_mut_hq_table$Pathogenicity
DMI_AF_mut_hq_chisq <- chisq.test(DMI_AF_mut_hq_dt)
DMI_AF_mut_hq_chisq

DMI_AF_mut_hq_PU_dt <- DMI_AF_mut_hq_dt[3:4,]
DMI_AF_mut_hq_PU_chisq <- chisq.test(DMI_AF_mut_hq_PU_dt)
DMI_AF_mut_hq_PU_chisq


DMI_AF_mut_interfacial_dt <- as.table(as.matrix(DMI_AF_mut_hq_table[,c(2,5)]))
rownames(DMI_AF_mut_interfacial_dt) <- as.vector(DMI_AF_mut_hq_table$Pathogenicity)
DMI_AF_mut_interfacial_chisq <- chisq.test(DMI_AF_mut_interfacial_dt)
DMI_AF_mut_interfacial_chisq

```

```{r Figure5_data2}
Model_metrics  <- read_tsv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/HuRI_DMI_AF_modelling/All_model_metrics.tsv")

Model_metrics_fragments <- read_tsv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/HuRI_DMI_AF_modelling/All_model_metrics_byFragments.tsv")
Model_metrics_fragments <- Model_metrics_fragments %>% filter(model_score>=0.6)

Model_metrics_motif <- Model_metrics %>% filter(Fragment=="Motif" & model_score>=0.6) %>% select(DMI, aa_code, residue_n, motif_start, pLDDT, Contact_3A, Contact_5A, SC_total_Acc, SC_norm_Acc, Acc_change)

Model_metrics_motif <-  Model_metrics_motif %>% rename(fragment_start=motif_start)
Model_metrics_motif$motif_start <- as.double(gsub("_.*?$","",gsub("^.*?_D_","",Model_metrics_motif$DMI)))
Model_metrics_motif$motif_end <- as.double(gsub("^.*?_","",gsub("^.*?_D_","",Model_metrics_motif$DMI)))
Model_metrics_motif$motif_len <- Model_metrics_motif$motif_end - Model_metrics_motif$motif_start
Model_metrics_motif <- Model_metrics_motif %>% mutate(Motif_region = case_when( residue_n < motif_start~ "Up",
                                                                                residue_n > motif_end~ "Down",
                                                                                TRUE  ~ "Core"))

Model_metrics_motif$Motif_region <- ordered(Model_metrics_motif$Motif_region, levels = c("Up", "Core","Down"))  
Model_metrics_motif$Motif_accession <- gsub("_D_.*?$","",gsub("^.*?_ELM","ELM",Model_metrics_motif$DMI))

ELM_classes <- HuRI_DMIs %>% select(Accession, ELM_class, ELM_type) %>% unique() %>% rename(Motif_accession=Accession)

Model_metrics_motif <- left_join(Model_metrics_motif, ELM_classes, by="Motif_accession")

Model_metrics_motif <- Model_metrics_motif %>% mutate(Motif_flanks = case_when( residue_n < motif_start ~ residue_n - motif_start,
                                                                                residue_n > motif_end ~ residue_n - motif_end,
                                                                                TRUE  ~ 0 ))

Model_metrics_motif %>% filter(Motif_flanks < -5 | Motif_flanks >= 5) %>% select(DMI) %>% unique()


Model_metrics[grep("Q8WWZ3", Model_metrics$DMI),] %>% select(DMI, ) %>% unique()
HuRi_DMI %>% filter(SLiMProtein=="P40763" & Accession=="ELME000278")
```

```{r Figure5_data3}
Scores_json_raw <- read_tsv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/HuRI_DMI_AF_modelling/Score_files/DMI_model_confidence_all_2.tab", col_names = T, na = "")
Scores_json <- Scores_json_raw %>% filter(rank==1) %>% select(DMI,model, score)
Scores_json$Model <- gsub("_multimer_v3_pred_0","",gsub("model_","",Scores_json$model))
Scores_json <- Scores_json %>% select(-model)
Scores_json$PPI <- gsub("_.*?\\.","-",  gsub("_ELM.*?$","", Scores_json$DMI))

scores_count <- tibble(bin=seq(from = 0, to = 1, by = 0.05), count_dmi=0, count_ppi=0)
cc_v <- c()
pp_v <- c() ## PPI count

for(i in seq(from = 0, to = 1, by = 0.05)){
  cc <- Scores_json %>% filter(score >= i) %>% nrow()
  pp <- Scores_json %>% filter(score >= i) %>% select(PPI) %>% unique() %>% nrow()
  cc_v <- c(cc_v, cc)
  pp_v <- c(pp_v, pp)
}

scores_count$count_dmi <-cc_v 
scores_count$count_ppi <-pp_v 

Scores_json %>% filter(score >= 0.6) %>% nrow()
Scores_json %>% filter(score >= 0.6) %>% select(PPI) %>% unique() %>% nrow()
(Scores_json %>% filter(score >= 0.6) %>% nrow()) /( Scores_json %>% nrow())

Scores_json %>% filter(DMI=="Q9Y4K3_SM00061_O_292_507.Q8WWZ3_ELME000133_D_25_33")
```


```{r Figure6_sankeys_data}
DMI_AF_mut_mapped <- DMI_AF_mut_hq %>% count(structural_region, Mapped_pathogenicity)

DMI_AF_mut_mapped <- DMI_AF_mut_mapped %>% separate(Mapped_pathogenicity, c("ClinVar", "AlphaMissence"), sep="_") %>% rename(value=n)
DMI_AF_mut_mapped$AlphaMissence <-  paste("AM_", DMI_AF_mut_mapped$AlphaMissence, sep="")

nodes <- data.frame(
  name=c(as.character(DMI_AF_mut_mapped$ClinVar), 
  as.character(DMI_AF_mut_mapped$AlphaMissence)) %>% unique()
)

nodes$name <- ordered(nodes$name, levels = c("B","C","P", "U", "AM_B","AM_A","AM_P"))
nodes$group <- ordered(nodes$name, levels = c("B","C","P", "U", "AM_B","AM_A","AM_P"))

DMI_AF_mut_mapped$ClinVar <- match(DMI_AF_mut_mapped$ClinVar, nodes$name)-1 
DMI_AF_mut_mapped$AlphaMissence <- match(DMI_AF_mut_mapped$AlphaMissence, nodes$name)-1

my_color <- 'd3.scaleOrdinal() .domain(["B", "C", "P", "U", "AM_B", "AM_A", "AM_P"]) .range(["#56B4E9", "#E69F00", "#d52221", "#999999",   "#56B4E9", "#E69F00","#d52221"])'
my_color2 <- 'd3.scaleOrdinal() .domain(["B", "C", "P", "U", "AM_B", "AM_P", "AM_A"]) .range(["#56B4E9", "#E69F00", "#d52221", "#999999",   "#56B4E9", "#d52221", "#E69F00"])'
c("#56B4E9", "#E69F00", "#d52221", "#999999",   "#56B4E9", "#E69F00","#d52221")
regions <- DMI_AF_mut_hq %>% select(structural_region) %>% unique() %>% unlist() %>% as.vector()


```

```{r Figure6_Data_GeneSymbols}
GeneSymbols <- read.csv(file="/Volumes/imb-luckgr/imb-luckgr2/projects/AlphaFold/AFDB_HumanReferenceProteome/GeneCode_012025/GeneSymbols_Uniprot.csv")

Symbol_map <- as.vector(GeneSymbols$Symbol)
names(Symbol_map) <- as.vector(GeneSymbols$UniProt)
```

```{r Figure6_Data_Larger_network}
source("/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Networks/visNetwork_DMI_functions.R")


Domain_UB_proteins <- DMI_AF_mut_hq %>% filter( structural_region=="d_interface" & Mapped_pathogenicity=="U_B") %>% select(Chain_protein) %>% unique()
Motif_UB_proteins <- DMI_AF_mut_hq %>% filter(structural_region=="m_interface" & Mapped_pathogenicity=="U_B") %>% select(Chain_protein) %>% unique()

table(Domain_UB_proteins$Chain_protein %in% Motif_UB_proteins$Chain_protein)

UB_proteins <- unique(c(Domain_UB_proteins$Chain_protein, Motif_UB_proteins$Chain_protein)) 

AF_graph <- mk_PPI_network(DMI_AF_metrics)
BU_graph <- slice_PPI_network(as.vector(UB_proteins),AF_graph)

BU_graph[[1]]$id <- Symbol_map[BU_graph[[1]]$id ]
BU_graph[[2]]$from <- Symbol_map[BU_graph[[2]]$from]
BU_graph[[2]]$to <- Symbol_map[BU_graph[[2]]$to]
BU_graph2 <- mk_igraph_PPI_network(BU_graph)


BU_graph[[1]] %>% count(group) #188 extra proteins
# 311 in total
```

```{r Figure6_data_DO_parents}
Protein_disorders_child_parents <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/AF_model_tables/DO_disease_parent_terms_filled.csv")
disease_parent_map <- c("Nervous system disease" = "skyblue", "Disease of metabolism" = "limegreen", "X-Linked monogenic disease"= "orange", "Chromosomal disease" ="yellow","Specific developmental disorder"="pink", "Musculoskeletal system disease"="red", "Other"="darkgrey")

#UB_disorder_net <- HuRi_DMI %>% select(DomainProtein, SLiMProtein) %>% filter(DomainProtein %in% UB_proteins & SLiMProtein %in% UB_proteins) %>% unique()
UB_disorder_net <- HuRi_DMI %>% filter(DMIMatchScore>=0.7) %>% select(DomainProtein, SLiMProtein) %>% filter(SLiMProtein %in% UB_proteins) %>% unique() #From all HuRI interactions which ones have slims proteins with interfacial uncertain variants

UB_disorder_net$DomainSymbol <- Symbol_map[UB_disorder_net$DomainProtein] # Use the symbol map to name the domain proteins
UB_disorder_net$MotifSymbol <- Symbol_map[UB_disorder_net$SLiMProtein] # Use the symbol map to name the motif proteins

UB_pairs_edges <- c(rbind(UB_disorder_net$DomainSymbol , UB_disorder_net$MotifSymbol)) # Interlace the domain and motif pairs in a single vector to create an igraph network
UB_pairs_ig <- make_graph(edges = UB_pairs_edges, directed = T) # Create igraph network

UB_pairs_vertices <- V(UB_pairs_ig)$name %>% as.tibble() %>% rename(Symbol=value) # Get the order of vertices in the igraph network (it is new every time it is created)
UB_pairs_vertices <- left_join(UB_pairs_vertices, GeneSymbols, by="Symbol") %>% rename(GeneSymbol=Symbol) %>% unique() # reAdd UniProt IDs




UB_pairs_vertices <- left_join(UB_pairs_vertices, Protein_disorders_child_parents, by="GeneSymbol", multiple = "first") # Assign the parent terms of the protein disease associations
UB_pairs_vertices <- UB_pairs_vertices %>% select(-SwissProtReference) # Remove Double UniProt columns
UB_pairs_vertices$color <- disease_parent_map[UB_pairs_vertices$parent_term] # Assign node/vertex color depeding on the parent term
UB_pairs_vertices <- UB_pairs_vertices %>% mutate(color=case_when(is.na(color) ~ "lightgrey", TRUE ~ color)) # When there is no assignment, then assign a different color 

UB_pairs_vertices <- UB_pairs_vertices %>% 
  mutate(DomainProtein=case_when(UniProt %in% HuRi_DMI$DomainProtein ~ 1, TRUE ~ 0), MotifProtein=case_when(UniProt %in% HuRi_DMI$SLiMProtein ~ 1, TRUE ~ 0)) %>%
  mutate(Fragment=case_when(DomainProtein==1 & MotifProtein==1 ~ "both", DomainProtein==1 & MotifProtein==0 ~ "domain",DomainProtein==0 & MotifProtein==1 ~ "motif")) %>%
  select(-DomainProtein,-MotifProtein) # Assign fragment type depending on whether or not the protein has DMI with domain, motif or both
UB_pairs_vertices <- UB_pairs_vertices %>% mutate(shape=case_when(Fragment=="both" ~ "circle", Fragment=="domain" ~ "square", Fragment=="motif" ~ "rectangle" )) # Assign shape based on Fragment type

V(UB_pairs_ig)$color <- UB_pairs_vertices$color
V(UB_pairs_ig)$shape <- UB_pairs_vertices$shape

## Create network to export
UB_disorder_network <-  UB_disorder_net %>% rename(UniProt=DomainProtein) %>% rename(GeneSymbol=DomainSymbol)

# UB_disorder_network <- left_join(UB_disorder_network, UB_pairs_edge, by=c("GeneSymbol", "UniProt"))
# UB_disorder_network <-  UB_disorder_network %>% rename(DomainProtein=UniProt) %>% rename(DomainSymbol=GeneSymbol) %>% rename(parent_term_domain=parent_term) %>% rename(GeneSymbol=MotifSymbol) %>% rename(UniProt=SLiMProtein )
# UB_disorder_network <- left_join(UB_disorder_network, UB_pairs_edge, by=c("GeneSymbol", "UniProt"))
# UB_disorder_network <-  UB_disorder_network %>% rename(SLiMProtein=UniProt) %>% rename(MotifSymbol=GeneSymbol) %>% rename(parent_term_motif=parent_term)
# 
# UB_disorder_network <- UB_disorder_network%>% mutate(parent_term = coalesce(parent_term_motif, parent_term_domain))
# UB_disorder_network <- UB_disorder_network %>% mutate(parent_term=case_when(is.na(parent_term) ~ "None", TRUE ~ parent_term))
# UB_disorder_network <- UB_disorder_network %>% select(-parent_term_motif, -parent_term_domain) %>% mutate(parent_term_2 = parent_term)

write_tsv(UB_disorder_network, file="/Users/jesusav/Desktop/UB_disorder_net_edges.tsv")

# UB_disorder_ver <- rbind(UB_disorder_network %>% select(DomainSymbol, parent_term)  %>% unique() %>% rename(name=DomainSymbol), 
#       UB_disorder_network %>% select(MotifSymbol, parent_term)  %>% unique() %>% rename(name=MotifSymbol)) 
# UB_disorder_ver <- UB_disorder_ver %>% group_by(name) %>% slice(1) %>%ungroup()

UB_disorder_ver <- UB_pairs_vertices %>% mutate(DisorderName=case_when(is.na(DisorderName) ~ "None", TRUE ~ DisorderName), parent_term=case_when(is.na(parent_term) ~ "None", TRUE ~ parent_term)) %>% select(-color, -shape)

write_tsv(UB_disorder_ver, file="/Users/jesusav/Desktop/UB_disorder_net_vertices.tsv")


```

```{r Figure6_data_GO_terms}
HuRI_UPID <- rbind(HuRi_DMI %>% select(SLiMProtein) %>% unique() %>% rename(ID=SLiMProtein), HuRi_DMI %>% select(DomainProtein) %>% unique() %>% rename(ID=DomainProtein)) %>% unique()
#write_tsv(HuRI_UPID, file="/Users/jesusav/Desktop/HuRI_UPID.txt", col_names = F)

HuRI_GO <- read_tsv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/AF_model_tables/HuRI_proteins_gene_GOterms.tsv")
HuRI_GO <- HuRI_GO %>% select(-From, -Organism, -Reviewed, -`Protein names`)
colnames(HuRI_GO) <- c("UniProt", "Name", "Symbol", "Length", "BP", "MF", "CC")

HuRI_GO_BP <- HuRI_GO %>% select(UniProt, BP) %>% separate_rows(BP, sep=";") %>% separate(BP, c("GO_term", "GO_code"), sep=" \\[GO:") %>% 
  mutate(GO_term=gsub("^ ","",GO_term)) %>% mutate(GO_code=gsub("\\]","",GO_code)) %>% filter(!is.na(GO_term))
HuRI_GO_MF <- HuRI_GO %>% select(UniProt, MF) %>% separate_rows(MF, sep=";") %>% separate(MF, c("GO_term", "GO_code"), sep=" \\[GO:") %>% 
  mutate(GO_term=gsub("^ ","",GO_term)) %>% mutate(GO_code=gsub("\\]","",GO_code)) %>% filter(!is.na(GO_term))
HuRI_GO_CC <- HuRI_GO %>% select(UniProt, CC) %>% separate_rows(CC, sep=";") %>% separate(CC, c("GO_term", "GO_code"), sep=" \\[GO:") %>% 
  mutate(GO_term=gsub("^ ","",GO_term)) %>% mutate(GO_code=gsub("\\]","",GO_code)) %>% filter(!is.na(GO_term))

HuRI_GO_BP %>% count(GO_term) %>% arrange(-n)  # regulation of transcription by RNA polymerase II
HuRI_GO_BP %>% filter(UniProt %in% UB_pairs_vertices$UniProt) %>% count(GO_term) %>% arrange(-n)  # signal transduction
HuRI_GO_MF %>% count(GO_term) %>% arrange(-n) # identical protein binding
HuRI_GO_CC %>% count(GO_term) %>% arrange(-n) # cytosol and nucleus


UB_GO_BP <- HuRI_GO_BP %>% filter(UniProt %in% UB_pairs_vertices$UniProt) %>% mutate(GO_group=gsub("(negative )|(positive )","",GO_term)) %>% mutate(GO_group=gsub("regulation of ","",GO_group)) %>% mutate(GO_group=gsub("intracellular ","",GO_group)) %>% mutate(GO_group=case_when(GO_group=="cell population proliferation" ~ "cell proliferation", GO_group=="cell cycle" ~ "cell proliferation", GO_group=="cell division" ~ "cell proliferation" , GO_group=="protein autophosphorylation" ~ "protein phosphorylation" , TRUE ~ GO_group)) 

GO_map <- c("transcription by RNA polymerase II" = "skyblue", "signal transduction" = "limegreen", "protein phosphorylation" = "red", "apoptotic process" ="yellow", "cell proliferation"= "orange", "DNA-templated transcription"="pink",  "Other"="darkgrey")

HuRI_GO_BP_sel <- left_join(UB_GO_BP, UB_GO_BP %>% count(GO_group)) %>% arrange(-n) %>% group_by(UniProt) %>% slice(1) %>% ungroup() %>%  
                              mutate(Group=case_when(n > 40 ~  GO_group, TRUE ~ "Other"))  %>% select(UniProt, GO_term, Group)

#UB_pairs_ig <- make_graph(edges = UB_pairs_edges, directed = T)

UB_pairs_vertices <- left_join(UB_pairs_vertices, HuRI_GO_BP_sel, by="UniProt", multiple = "first")

UB_pairs_vertices$color2 <- GO_map[UB_pairs_vertices$Group]
UB_pairs_vertices <- UB_pairs_vertices %>% mutate(color2=case_when(is.na(color2) ~ "darkgrey", TRUE ~ color2)) %>% mutate(color2=case_when(is.na(Group) ~ "lightgrey", TRUE ~ color2))


UB_GO_ver <- UB_pairs_vertices %>% mutate(DisorderName=case_when(is.na(DisorderName) ~ "None", TRUE ~ DisorderName), parent_term=case_when(is.na(parent_term) ~ "None", TRUE ~ parent_term)) %>% select(GeneSymbol, UniProt, Fragment, GO_term, Group, color2)

write_tsv(UB_GO_ver, file="/Users/jesusav/Desktop/UB_GO_net_vertices.tsv")
```

```{r Figure6_data_Uncertain_HuRI_essentiality}
HuRI_essentiality <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_paper_tables/HuRI_Essentiality.csv")
HuRI_essentiality <- HuRI_essentiality %>% select(Gene_symbol, Fitness_effect) %>% rename(Symbol=Gene_symbol)

HuRI_age <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_paper_tables/HuRI_Age.csv")
HuRI_age <- HuRI_age %>% select(Gene_symbol, Age) %>% rename(Symbol=Gene_symbol)

HuRI_exp <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_paper_tables/HuRI_Exp_GTEx.csv")
HuRI_exp <- HuRI_exp %>% select(Gene_symbol, Median_expression) %>% rename(Symbol=Gene_symbol)

HuRI_LoFint <- read_csv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_paper_tables/HuRI_LoFint.csv")
HuRI_LoFint <- HuRI_LoFint %>% select(Gene_symbol, pLI) %>% rename(Symbol=Gene_symbol)

UB_proteins_sym <- GeneSymbols %>% filter(UniProt %in% UB_proteins)
UB_proteins_sym <-left_join(UB_proteins_sym, HuRI_essentiality)
UB_proteins_sym <-left_join(UB_proteins_sym, HuRI_age)
UB_proteins_sym <-left_join(UB_proteins_sym, HuRI_exp)
UB_proteins_sym <-left_join(UB_proteins_sym, HuRI_LoFint)
UB_proteins_sym <- UB_proteins_sym %>% arrange(-Fitness_effect)

Protein_mutant <- DMI_AF_mut_hq %>% select(Chain_protein, residue_n, Pathogenicity, M_number)%>% unique() %>% 
  count(Chain_protein, Pathogenicity) %>% rename(UniProt=Chain_protein) %>% pivot_wider(names_from = Pathogenicity, values_from = n) %>% select(-Benign, -Conflicting)
UB_proteins_sym <-left_join(UB_proteins_sym, Protein_mutant)
UB_proteins_sym <- UB_proteins_sym %>% mutate(Motif=case_when(UniProt %in% HuRi_DMI$SLiMProtein ~ 1, TRUE ~ 0)) %>% mutate(Domain=case_when(UniProt %in% HuRi_DMI$DomainProtein ~ 1, TRUE ~ 0))
UB_proteins_sym <- UB_proteins_sym %>% unique()

UB_proteins_sym %>% filter(Fitness_effect>0.15 & Median_expression>10 & pLI>0.9) %>% arrange(-Uncertain) # 6 proteins
#ggplot(UB_proteins_sym %>% filter(!is.na(Fitness_effect)), aes(y=Fitness_effect)) + geom_boxplot()

#Q9C0A6 SETD5 N-methyltransferase
acc <- HuRi_DMI %>% filter(SLiMProtein=="Q9C0A6") %>% select(Accession) %>% unique() %>% unlist()
HURI_ClinVar_mutation %>% filter(fragment_protein=="Q9C0A6" & pathogenicity=="Uncertain") %>% filter(fragment_id %in% acc) 

#Q9C0A6 SETD5 N-methyltransferase
acc <- HuRi_DMI %>% filter(SLiMProtein=="Q6UB99") %>% select(Accession) %>% unique() %>% unlist()
HURI_ClinVar_mutation %>% filter(fragment_protein=="Q6UB99" & pathogenicity=="Uncertain") %>% filter(fragment_id %in% acc) #%>% select(aa_change) %>% unique()


HuRi_DMI %>% filter(ELM_type=="DEG" & SLiMProtein %in% UB_proteins_sym$UniProt) 
HuRi_DMI %>% filter(ELM_type=="DEG" & SLiMProtein %in% UB_proteins_sym$UniProt) %>% select(SLiMProtein) %>% unique()

UB_motif_degrons <- DMI_AF_mut_hq %>% filter(structural_region=="m_interface" & Mapped_pathogenicity=="U_B") %>% 
  mutate(Motif_accession=gsub("_D.*?$", "", gsub("*.*?_ELM", "ELM", DMI)))
UB_motif_degrons <- left_join(UB_motif_degrons, ELM_classes) %>% filter(ELM_type=="DEG")

UB_motif_degrons %>% select(DMI, Chain_protein, residue_n, M_number, ELM_class)
UB_motif_degrons %>% separate(DMI, c("Domain", "Motif"), sep="\\.") %>% select(Chain_protein) %>% unique()
UB_motif_degrons %>% separate(DMI, c("Domain", "Motif"), sep="\\.") %>% select(Motif) %>% unique()
# 15 DMIs, 10 AF models, 2 degron classes, 9 motif instances in 6 proteins
```

```{r Figure6_data_Motif_REGEX_change}
DMI_AF_mut_hq_motif <- DMI_AF_mut_hq %>% filter(Fragment=="Motif")

HURI_ClinVar_mut_motif_2 <- read_tsv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_DMIs-ClinVar_mutations/DMI_human_HuRI_overlap_clinvar_2023_release_MotifDisruption.tsv")

HURI_ClinVar_mut_motif_pLDDT <- read_tsv(file="/Volumes/imb-luckgr/Luck_lab_members/Jesus/DMIs_ClinVar/Datasets/HuRI_DMIs-ClinVar_mutations/DMI_human_HuRI_overlap_clinvar_2023_release_MotifpLDDT.tsv")

HURI_ClinVar_mut_motif_summ <- HURI_ClinVar_mut_motif_2 %>% mutate(residue_n=as.double(gsub("[A-z]","",aa_change))) %>% count(fragment_protein, Accession,ELMIdentifier, fragment_pos, pathogenicity, residue_n,variant_match)
DMI_AF_mut_hq_temp <- DMI_AF_mut_hq %>% select(DMI, pLDDT, Norm_Acc, Acc_change,residue_n, Chain_protein,Fragment_id,structural_region, Pathogenicity,AM_pathogenicity)
HURI_ClinVar_mut_motif_summ <-  HURI_ClinVar_mut_motif_summ %>% rename(Chain_protein=fragment_protein, Fragment_id=Accession) %>% mutate(ELM_type=gsub("_.*?$","", ELMIdentifier))
HURI_ClinVar_mut_motif_summ <- left_join(HURI_ClinVar_mut_motif_summ, DMI_AF_mut_hq_temp)
HURI_ClinVar_mut_motif_summ <- HURI_ClinVar_mut_motif_summ %>% filter(!is.na(DMI))

HURI_ClinVar_mut_motif_summ <- left_join(HURI_ClinVar_mut_motif_summ, AF_model_scores)

UB_motif_degrons <- HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface" & ELM_type=="DEG" & pathogenicity=="Uncertain" & variant_match==0 & fragment_pLDDT<70) %>% unique()
UB_motif_degrons
UB_motif_degrons %>% select(Chain_protein ) %>% unique()
# 14 DMIs, 10 AF models, 3 degron classes, 9 motif instances in 7 proteins
# PALB2 DEG_Kelch_Keap1_1 degron  various VUS do not have disease association, other parts are associated with breast cancer
HURI_ClinVar_mut_motif_summ %>% filter(Chain_protein=="Q13501" & structural_region=="m_interface" & variant_match==0 & fragment_pLDDT<70) %>% select(DMI, ELMIdentifier) %>% unique()
# Q14145_SM00612_O_317_621.Q13501_ELME000390_D_347_352	DEG_Kelch_Keap1_1
# Paget Disease of Bone 3
HURI_ClinVar_mutation %>% filter(fragment_protein=="Q13501" & fragment_id=="ELME000390") %>% select(-fragment_protein, fragment_id) %>% 
  select(aa_change) %>% unlist()

HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface" & pathogenicity=="Uncertain" & variant_match==0 & fragment_pLDDT<70 & AM_pathogenicity=="AM_benign")
# Q99816_SM00212_O_1_311.Q6UWE0_ELME000099_D_648_653, Q99816_SM00212_O_1_311.Q6UWE0_ELME000099_D_660_665 LIG_PTAP_UEV_1 
# Charcot-Marie-Tooth disease axonal type 2P
HURI_ClinVar_mutation %>% filter(fragment_protein=="Q6UWE0" & fragment_id=="ELME000099") %>% select(-fragment_protein, fragment_id) %>% 
  select(aa_change) %>% unlist()


UB_motif_degrons <- HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface" & ELM_type=="LIG" & pathogenicity=="Uncertain" & variant_match==0 & fragment_pLDDT<70) %>% unique()
UB_motif_degrons %>% select(Chain_protein ) %>% unique()

ClinVar_mut_dis_short <- ClinVar_mut_dis %>% filter(!is.na(AM_pathogenicity)) %>% select(UniProt_ID, aa_change, pathogenicity,avg_pLDDT,avg_IUPred, dis_pLDDT, dis_IUPred, AM_value,AM_pathogenicity) %>% rename(fragment_protein=UniProt_ID )
HURI_ClinVar_mut_motif_2 <- left_join(HURI_ClinVar_mut_motif_2, ClinVar_mut_dis_short)

AF_model_scores %>% filter()
# 	Q14145_SM00612_O_317_621.Q86YC2_ELME000390_D_89_94
```




```{r Figure1_Graphic_A}
# Number of ClinVar variants
p1 <- ggplot(ClinVar_mutations, aes(y=pathogenicity , fill=pathogenicity)) + geom_bar(width = 0.6) +
  ggtitle("ClinVar variant pathogenicity") +  ylab("Clinical relevance") + xlab("Variant number") + xlim(0, 850000) +
  theme_classic() + scale_fill_manual(values = colors_Var) + labs(fill = "Pathogenicity") +
  theme(text = element_text(size = 15), axis.text = element_text(color = "black"), legend.position = "none",  
        axis.ticks = element_line(linewidth = 1), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1)) + 
  scale_x_continuous(labels = scales::comma) + scale_y_discrete(limits=rev) 
p1
# with labels
p1 +geom_text(aes(label = ..count..), stat = "count", vjust = -2, position = position_dodge(0.9), size = 3.5) +
  geom_text(stat='prop', vjust=-0.5, , position = position_dodge(0.9), size=3)

#SAVE TIFF
tiff("Figures/Figure1_graphics/ClinVar_mutations.tiff", units="cm", width=10, height=7, res=800)
p1
dev.off()
#SAVE PDF
pdf("Figures/Figure1_graphics/ClinVar_mutations.pdf", width=5, height=3)
p1
dev.off()

```

```{r Figure1_Graphic_B}
p1 <- ggplot(ClinVar_mut_dis, aes(x=avg_pLDDT, y=pathogenicity , color=as.factor(pathogenicity))) + 
  geom_quasirandom(size=0.5, alpha=0.4) + geom_boxplot(color="black", width=0.25, outlier.alpha = 0.2, alpha=0.5) +
  geom_vline(xintercept = 70 , size=0.5, color="red") + 
  ylab("Clinical relevance") + xlab("Residue pLDDT") + ggtitle("") + 
  theme_classic()  +  scale_color_manual(values=colors_Var) +
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), legend.position = "none", 
        axis.ticks = element_line(linewidth = 1)) + scale_y_discrete(limits=rev) 

p1

#SAVE PNG
png("Figures/Figure1_graphics/ClinVar_pLDDT.png", units="cm", width=12, height=8, res=800)
p1
dev.off()
# #SAVE PDF
# pdf("Figures/Figure1_graphics/ClinVar_pLDDT.pdf", width=4, height=4)
# p1
# dev.off()


p2 <- ggplot(ClinVar_mut_dis, aes(x=pathogenicity, y=avg_IUPred, color=as.factor(pathogenicity))) + 
  geom_quasirandom(size=0.5, alpha=0.4) + geom_boxplot(color="black", width=0.25, outlier.alpha = 0.2, alpha=0.5) +
  ggtitle("ClinVar variant IUPred") + xlab("Clinical relevance") + ylab("Residue IUPred") + 
  theme_classic()  +  scale_color_manual(values=colors_Var) +
  theme(text = element_text(size = 20), legend.position = "none", legend.title = element_text(size=8), 
        legend.text = element_text(size=8), legend.key.size = unit(0.3, 'cm'), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1),
        axis.line = element_line(size = 1) , axis.ticks = element_line(linewidth = 1)) 

p2 

#SAVE PNG
# png("Figures/Figure1_graphics/ClinVar_IUPred.png", units="cm", width=10, height=10, res=800)
# p2
# dev.off()
# #SAVE PDF
# pdf("Figures/Figure1_graphics/ClinVar_IUPred.pdf", width=4, height=4)
# p2
# dev.off()

#rm (p1,p2)
```

```{r Figure1_Graphic_C}
p1 <- ggplot(All_ELM_types, aes(x="", y=n, fill=ELM_type)) +
  geom_bar(stat="identity", width=1) +
  scale_fill_manual(values=colors_ELMtypes) +
  coord_polar("y", start=0) + theme_void() + theme(legend.position="none")
p1

#SAVE TIFF
tiff("Figures/Figure1_graphics/ELM_types_charts.tiff", units="cm", width=8, height=8, res=800)
p1
dev.off()

#SAVE PDF
pdf("Figures/Figure1_graphics/ELM_types_charts.pdf", width=4, height=4)
p1
dev.off()
```

```{r Figure1_Graphic_D}
p1 <- ggplot(RRS_ROC, aes(x=mean_fpr, y=mean_tpr, fill=RRS)) +  geom_line(color="grey30") + 
  geom_ribbon(aes(y = mean_tpr, ymin = mean_tpr - std_tpr, ymax = mean_tpr + std_tpr), alpha = 0.3) +
  #annotate(geom="text", label="mean AUC = 0.93", x=0.7, y=0.1, size=3, color="red")+
  ggtitle("RRS ROC") + xlab("1 - Specificity") + ylab("Sensitivity") + 
  theme_classic() + 
  theme(text = element_text(size = 30), axis.text = element_text(color = "black"), legend.position = "none",
        axis.line = element_line(size = 1), axis.ticks = element_line(linewidth = 1)) + 
  scale_x_continuous(breaks=c(0,0.5,1)) + scale_y_continuous(breaks=c(0,0.5,1)) 

p1
p1 + geom_point() + geom_text(aes(label = mean_tpr), vjust = -1, hjust = -0.2, size = 3) 

#SAVE TIFF
tiff("Figures/Figure1_graphics/DMI_predictor_ROC.tiff", units="cm", width=6, height=6, res=800)
p1
dev.off()
#SAVE PDF
pdf("Figures/Figure1_graphics/DMI_predictor_ROC.pdf", width=4, height=4)
p1
dev.off()

p2 <- ggplot(RRS_PRC, aes(x=mean_recall, y=mean_precision, fill=RRS)) +  geom_line(color="grey30") +
  geom_ribbon(aes(y = mean_precision, ymin = mean_precision - std_precision, ymax = mean_precision + std_precision), alpha = 0.3) +
  #annotate(geom="text", label="mean AP = 0.93", x=0.3, y=0.5, size=3, color="red")+
  ggtitle("RRS PRC") + xlab("Recall") + ylab("Precision") + 
  theme_classic() + 
  theme(text = element_text(size = 30), axis.text = element_text(color = "black"),legend.position = "none",
        axis.line = element_line(size = 1), axis.ticks = element_line(linewidth = 1)) + 
  scale_x_continuous(breaks=c(0,0.5,1)) 

p2
p2 + geom_point() + geom_text(aes(label = mean_precision), vjust = -1, hjust = -0.2, size = 3) 

#SAVE TIFF
tiff("Figures/Figure1_graphics/DMI_predictor_PRC.tiff", units="cm", width=6, height=6, res=800)
p2
dev.off()
#SAVE PDF
pdf("Figures/Figure1_graphics/DMI_predictor_PRC.pdf", width=4, height=4)
p2
dev.off()


```

```{r Figure1_Graphic_F}
p1 <- ggplot(HuRI_DMIs %>% filter(DMIMatchScore >= 0.7), aes(y = ELM_type, fill = ELM_type)) + geom_bar(width = 0.7) + 
  ylab("ELM category") + xlab("DMI number") + 
  scale_fill_manual(values=colors_ELMtypes) + 
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", axis.text = element_text(color = "black"),
                  axis.ticks = element_line(linewidth = 1), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1)) + 
  scale_x_continuous(labels = scales::comma, limits = c(0,7000)) + 
  scale_y_discrete(limits=rev, labels = rev(elm_categories)) 
p1

p1 + geom_text(stat='count', aes(label=format(after_stat(count), big.mark=",")), vjust=-2, size=3) +
  geom_text(stat='prop', vjust=-0.25, size=3) 

# SAVE TIFF
tiff("Figures/Figure1_graphics/HuRI_DMI_higScore.tiff", units="cm", width=12, height=8, res=800)
p1
dev.off()

# SAVE PDF
pdf("Figures/Figure1_graphics/HuRI_DMI_higScore.pdf", width=4, height=3)
p1
dev.off()

p2 <- ggplot(HuRI_DMIs %>% filter(DMIMatchScore >= 0.7) %>% select(intx_ID, ELM_type) %>% unique(), aes(y = ELM_type, fill = ELM_type)) + 
  geom_bar(width = 0.7) + 
  ylab("ELM category") + xlab("PPI number") + 
  scale_fill_manual(values=colors_ELMtypes) +  
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", axis.text = element_text(color = "black"), 
                 axis.ticks = element_line(linewidth = 1), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1)) + 
  scale_x_continuous(labels = scales::comma, limits = c(0,2000)) + 
  scale_y_discrete(limits=rev, labels = rev(elm_categories)) 
p2

p2 + geom_text(stat='count', aes(label=format(after_stat(count), big.mark=",")), vjust=-2, size=3) +
  geom_text(stat='prop', vjust=-0.25, size=3) 

# SAVE TIFF
tiff("Figures/Figure1_graphics/HuRI_DMI_higScore_PPI.tiff", units="cm", width=12, height=8, res=800)
p2
dev.off()

# SAVE PDF
pdf("Figures/Figure1_graphics/HuRI_DMI_higScore_PPI.pdf", width=4, height=3)
p2
dev.off()

```


```{r Figure1_Supp_Graphic_ChiSq1}
### Correlations plots - pLDDT classification

Pathogenicity_pLDDT_chisq$residuals
Pathogenicity_pLDDT_chisq$observed
#SAVE TIFF
tiff("Figures/Figure1_graphics/ClinVar_pLDDT_disorder_chisq.tiff", units="cm", width=10, height=20, res=800)
corrplot(Pathogenicity_pLDDT_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')), 
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, col.lim= c(-80, 80))
dev.off()
#SAVE PDF
pdf("Figures/Figure1_graphics/ClinVar_pLDDT_disorder_chisq.pdf", width=4, height=8)
corrplot(Pathogenicity_pLDDT_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')), 
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, col.lim= c(-80, 80))
dev.off()

#corrplot(Pathogenicity_pLDDT_chisq$observed,  method = 'color', is.corr   = FALSE, col = rev(COL2('RdBu')), cl.pos = 'b', addCoef.col = 'grey50')

Pathogenicity_pLDDT_PU_chisq$residuals
Pathogenicity_pLDDT_PU_chisq$observed
#SAVE TIFF
tiff("Figures/Figure1_graphics/ClinVar_pLDDT_disorder_PU_chisq.tiff", units="cm", width=10, height=10, res=800)
corrplot(Pathogenicity_pLDDT_PU_chisq$residuals, method = 'color', is.corr = FALSE, col = rev(COL2('RdBu')), 
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, col.lim= c(-80, 80))
dev.off()
#SAVE PDF
pdf("Figures/Figure1_graphics/ClinVar_pLDDT_disorder_PU_chisq.pdf", width=4, height=4)
corrplot(Pathogenicity_pLDDT_PU_chisq$residuals, method = 'color', is.corr = FALSE, col = rev(COL2('RdBu')), 
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, col.lim= c(-80, 80))
dev.off()
```

```{r Figure1_Supp_Graphic_ChiSq2}
### Correlations plots - IUPred classification

Pathogenicity_IUPred_chisq$residuals
Pathogenicity_IUPred_chisq$observed
#SAVE TIFF
tiff("Figures/Figure1_graphics/ClinVar_IUPred_disorder_chisq.tiff", units="cm", width=10, height=20, res=800)
corrplot(Pathogenicity_IUPred_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')),  
         col.lim= c(-70, 70), cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")
dev.off()
#SAVE PDF
pdf("Figures/Figure1_graphics/ClinVar_IUPred_disorder_chisq.pdf", width=4, height=8)
corrplot(Pathogenicity_IUPred_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')),  
         col.lim= c(-70, 70), cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")
dev.off()
#corrplot(Pathogenicity_IUPred_chisq$observed,  method = 'color', is.corr   = FALSE, col = rev(COL2('RdBu')), cl.pos = 'b', addCoef.col = 'grey50', outline=T, addgrid.col="white")

Pathogenicity_IUPred_PU_chisq$residuals
Pathogenicity_IUPred_PU_chisq$observed
#SAVE TIFF
tiff("Figures/Figure1_graphics/ClinVar_IUPred_disorder_PU_chisq.tiff", units="cm", width=10, height=20, res=800)
corrplot(Pathogenicity_IUPred_PU_chisq$residuals, method = 'color', is.corr = FALSE, col = rev(COL2('RdBu')),  
         col.lim= c(-70, 70), cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")
dev.off()
#SAVE PDF
pdf("Figures/Figure1_graphics/ClinVar_IUPred_disorder_PU_chisq.pdf", width=4, height=4)
corrplot(Pathogenicity_IUPred_PU_chisq$residuals, method = 'color', is.corr = FALSE, col = rev(COL2('RdBu')),  
         col.lim= c(-70, 70), cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")
dev.off()
```

```{r Figure1_Supp_Graphic_AM}
## Needs to be updated to include all ClinVar and not only the DMI proteins mutations
p1 <- ggplot(ClinVar_mut_dis %>% filter(!is.na(AM_pathogenicity)), aes(x=AM_pathogenicity, y=avg_pLDDT, color=as.factor(AM_pathogenicity))) + 
  geom_quasirandom(size=0.5, alpha=0.4) + geom_boxplot(color="black", width=0.25, outlier.alpha = 0.2, alpha=0.5) +
  ggtitle("AM ClinVar variants pLDDT") + xlab("AM pathogenicity") + ylab("Residue pLDDT") + 
  theme_classic()  +  scale_color_manual(values=colors_Var) +
  theme(text = element_text(size = 15), axis.text = element_text(color = "black"), legend.position = "none", 
        legend.title = element_text(size=8), legend.text = element_text(size=8), legend.key.size = unit(0.3, 'cm'), 
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), axis.line = element_line(size = 1) , 
        axis.ticks = element_line(linewidth = 1)) 
p1

# #SAVE TIFF
# tiff("Figures/Figure1_graphics/Other/Supp_AM_pLDDT.tiff", units="cm", width=10, height=10, res=800)
# p1
# dev.off()
# #SAVE PDF
# pdf("Figures/Figure1_graphics/Other/Supp_AM_pLDDT.pdf", width=4, height=4)
# p1
# dev.off()


p2 <- ggplot(ClinVar_mut_dis%>% filter(!is.na(AM_pathogenicity)), aes(x=AM_pathogenicity, y=avg_IUPred, color=as.factor(AM_pathogenicity))) + 
  geom_quasirandom(size=0.5, alpha=0.4) + geom_boxplot(color="black", width=0.25, outlier.alpha = 0.2, alpha=0.5) +
  ggtitle("AM ClinVar variant pLDDT") + xlab("AM pathogenicity") + ylab("Residue IUPred") + 
  theme_classic()  +  scale_color_manual(values=colors_Var) +
  theme(text = element_text(size = 15), legend.position = "none", axis.text = element_text(color = "black"), 
        legend.title = element_text(size=8), 
        legend.text = element_text(size=8), legend.key.size = unit(0.3, 'cm'), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1),
        axis.line = element_line(size = 1) , axis.ticks = element_line(linewidth = 1)) 
p2
# 
# #SAVE TIFF
# tiff("Figures/Figure1_graphics/Other/Supp_AM_IUPred.tiff", units="cm", width=10, height=10, res=800)
# p2
# dev.off()
# #SAVE PDF
# pdf("Figures/Figure1_graphics/Other/Supp_AM_IUPred.pdf", width=4, height=4)
# p2
# dev.off()


```

```{r Figure1_Supp_DMI_Graphic_B_DMIscores}
p1 <- ggplot(DMIscores_count, aes(x=bin, y=count_dmi)) +  geom_line() + 
  annotate(geom="text", label="0.7", x=0.78, y=140000, size=5, color="red")+ 
  geom_vline(xintercept = 0.7, size=0.5, color="red") + 
  ggtitle("HuRI DMI scores") + xlab("RF score") + ylab("DMI count") + 
  theme_classic() + theme(text = element_text(size = 20), axis.line = element_line(size = 1), axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1)) + 
  scale_y_continuous(labels = scales::comma, limits = c(0,150000)) 

p1 + geom_point() +
  annotate(geom="text", label="Step = 0.05", x=0.2, y=4000, size=3.5)+
  geom_text(aes(label = count_dmi), vjust = -1, hjust = -0.2, size = 3) 


p2 <- ggplot(DMIscores_count, aes(x=bin, y=count_ppi)) +  geom_line() + 
  annotate(geom="text", label="0.7", x=0.78, y=8000, size=5, color="red")+ 
  geom_vline(xintercept = 0.7, size=0.5, color="red") + 
  ggtitle("HuRI DMI scores") + xlab("RF score") + ylab("PPI count") + 
  theme_classic() + theme(text = element_text(size = 20), axis.line = element_line(size = 1), axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))+ 
  scale_y_continuous(labels = scales::comma) 

p2 + geom_point() +
  annotate(geom="text", label="Step = 0.05", x=0.2, y=4000, size=3.5)+
  geom_text(aes(label = count_ppi), vjust = -1, hjust = -0.2, size = 3) 


coeff <-  max(DMIscores_count$count_ppi) /max(DMIscores_count$count_dmi)

p3 <- ggplot(DMIscores_count, aes(x=bin)) +  geom_line( aes(y=count_dmi), size=1) + geom_line( aes(y=count_ppi / coeff)) + 
  annotate(geom="text", label=c("0.7","DMI","PPI"), x=c(0.78,0.25,0.45), y=c(140000,60000,120000), size=5, color=c("red","black","black") )+ 
  geom_vline(xintercept = 0.7, size=0.5, color="red") + 
  ggtitle("HuRI DMI scores") + xlab("RF DMI score") +
  theme_classic() + 
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))+ 
  scale_y_continuous(labels = scales::comma, , limits = c(0,150000), name = "DMI count", sec.axis = sec_axis(~.*coeff, name="PPI count"))


# SAVE TIFF
tiff("Figures/SuppFig_DMI_HuRI_graphics/DMIScore_cummulative_DMI_PPI.tiff", units="cm", width=12, height=8, res=800)
p3
dev.off()

# SAVE PDF
pdf("Figures/SuppFig_DMI_HuRI_graphics/DMIScore_cummulative_DMI_PPI.pdf", width=5, height=3)
p3
dev.off()

```

```{r Figure1_Supp_DMI_Graphic_C_DMIscores_ELM}
p1 <- ggplot(HuRI_DMIs, aes(x=DMIMatchScore, y = ELM_type, color = ELM_type)) + geom_quasirandom(size=0.1, alpha=0.5) + #geom_beeswarm(corral = "wrap", alpha=0.2, size=0.1) + 
  geom_boxplot(color="black", width=0.15, lwd=0.4, alpha=0.75, outlier.size = 1, outlier.alpha = 0.5) + 
  geom_vline(xintercept = 0.7, color="black") +
  ylab("ELM category") + xlab("DMI score") + 
  scale_color_manual(values=colors_ELMtypes) +
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", 
                          axis.text = element_text(color = "black"),axis.ticks = element_line(linewidth = 1))+ 
  scale_y_discrete(limits=rev, labels = rev(elm_categories)) 
p1

#SAVE png
png("Figures/SuppFig_DMI_HuRI_graphics/HuRI_DMI_Scores.png", units="cm", width=12, height=8, res=800)
p1
dev.off()
# SAVE PDF
# pdf("Figures/SuppFig_DMI_HuRI_graphics/HuRI_DMI_Scores.pdf", width=4, height=3)
# p1
# dev.off()
```

```{r Figure1_Supp_DMI_Graphic_D_ELMclasses}
p1 <- ggplot(HuRI_DMIs %>% filter(DMIMatchScore >= 0.7) %>% select(Accession, ELM_class, ELM_type) %>% unique(), aes(y = ELM_type, fill = ELM_type)) + 
  geom_bar( width = 0.5) + 
  ylab("ELM category") + xlab("Class number") + 
  scale_fill_manual(values=colors_ELMtypes) + xlim(0, 102) +  
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", axis.text = element_text(color = "black"),
                          axis.ticks = element_line(linewidth = 1))+ 
  scale_y_discrete(limits=rev, labels = rev(elm_categories)) 
p1

p1 + geom_text(stat='count', aes(label=format(after_stat(count), big.mark=",")), vjust=-2, size=3) +
  geom_text(stat='prop', vjust=-0.25, size=3) 

# # SAVE TIFF
# tiff("Figures/SuppFig_DMI_HuRI_graphics/HuRI_DMI_highScore_Classes.tiff", units="cm", width=10, height=8, res=800)
# p1
# dev.off()
# 
# # SAVE PDF
# pdf("Figures/SuppFig_DMI_HuRI_graphics/HuRI_DMI_highScore_Classes.pdf", width=4, height=3)
# p1
# dev.off()

HuRI_DMI_classes <- HuRI_DMIs %>% select(Accession, ELM_class, ELM_type) %>% unique()
HuRI_DMI_classes_high <- HuRI_DMIs %>% filter(DMIMatchScore >= 0.7) %>% select(Accession, ELM_class, ELM_type) %>% unique()
HuRI_DMI_classes <- HuRI_DMI_classes %>% mutate(HighScored=case_when(Accession %in% HuRI_DMI_classes_high$Accession ~ "1", TRUE ~ "0"))

colors_ELMtypes_ln <- c(rbind(rep("lightgrey",6), colors_ELMtypes))

p2 <- ggplot(HuRI_DMI_classes, aes(y=ELM_type , fill=interaction(HighScored, ELM_type))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ylab("ELM category") + xlab("ELM class fraction") + 
  theme_minimal() + scale_fill_manual(values=colors_ELMtypes_ln) + labs(fill = "High Scored\nDMI classes") +
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(linewidth = 1.5, size = 0.5), legend.position="none")+ 
  scale_y_discrete(limits=rev, labels = rev(elm_categories)) 
p2
p2 + geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

# SAVE TIFF
tiff("Figures/SuppFig_DMI_HuRI_graphics/HuRI_DMI_highScore_Classes_fraction.tiff", units="cm", width=12, height=8, res=800)
p2
dev.off()

# SAVE PDF
pdf("Figures/SuppFig_DMI_HuRI_graphics/HuRI_DMI_highScore_Classes_fraction.pdf", width=5, height=3)
p2
dev.off()

```

```{r Figure1_Supp_DMI_Graphic_E_ELMinstances}
p1 <- ggplot(DMI_ELM_instances, aes(y=DMIMatchScore, x = ELM_instance, color = ELM_instance)) + geom_quasirandom(size=1, alpha=0.75) + 
  geom_boxplot(color = "black", width = 0.15, lwd = 0.4, alpha = 0.75, outlier.size = 1, outlier.alpha = 0.5) + 
  ggtitle("HuRi DMIs - ELM") +  xlab("ELM instance") + ylab("DMI score") + 
  scale_color_manual(values = colors_BIN) +
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", 
                          axis.text = element_text(color = "black"),axis.ticks = element_line(linewidth = 1))
p1

# SAVE png
png("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_DMI_scores_instances.png", units="cm", width=8, height=8, res=800)
p1
dev.off()

p1 <- ggplot(DMI_ELM_instances, aes(y=DMIMatchScore, x = ELM_motif, color = ELM_motif)) + geom_quasirandom(size=0.5, alpha=0.5) + 
  geom_boxplot(color="black", width=0.15, lwd=0.4, outlier.size = 1, outlier.alpha = 0.5) + 
  ggtitle("HuRi DMIs - ELM") +  xlab("ELM instance") + ylab("DMI score") + 
  scale_color_manual(values=colors_BIN) +
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", 
                          axis.line = element_line(size = 1), axis.ticks = element_line(linewidth = 1))
p1
```

```{r Figure1_Supp_DMI_Graphic_F_ELMinstances}
p1 <- ggplot(DMI_ELM_instances %>% filter(ELM_instance== "1"), aes(y = ELM_type, fill = ELM_type)) + geom_bar(width = 0.5) + 
  ylab("ELM category") + xlab("DMI number") +  ylim(0, 30) +  
  scale_fill_manual(values=colors_ELMtypes[3:5]) + 
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", 
                          axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1)) + 
  scale_y_continuous(labels = scales::comma, limits = c(0, 30)) + 
  scale_y_discrete(limits=rev, labels = rev(elm_categories[3:5])) 
p1

p1 + geom_text(stat='count', aes(label=format(after_stat(count), big.mark=",")), vjust=-2, size=3) +
  geom_text(stat='prop', vjust=-0.25, size=3) 

# SAVE TIFF
tiff("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_DMI_high_instances.tiff", units="cm", width=8, height=6, res=800)
p1
dev.off()
# SAVE PDF
pdf("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_DMI_high_instances.pdf", width=4, height=3)
p1
dev.off()


p2 <- ggplot(DMI_ELM_instances %>% filter(ELM_instance== "1") %>% select(intx_ID, ELM_type) %>% unique(), aes(y = ELM_type, fill = ELM_type)) + 
  geom_bar(width = 0.5) + 
  ylab("ELM category") + xlab("PPI number") +  ylim(0, 30) + 
  scale_fill_manual(values=colors_ELMtypes[3:5]) +  
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", 
                          axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1)) + 
  scale_y_discrete(limits=rev, labels = rev(elm_categories[3:5])) 
p2

p2 + geom_text(stat='count', aes(label=format(after_stat(count), big.mark=",")), vjust=-2, size=3) +
  geom_text(stat='prop', vjust=-0.25, size=3) 

# SAVE TIFF
tiff("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_DMI_high_instances_PPI.tiff", units="cm", width=8, height=6, res=800)
p2
dev.off()
# SAVE PDF
pdf("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_DMI_high_instances_PPI.pdf", width=4, height=3)
p2
dev.off()


p3 <- ggplot(DMI_ELM_instances %>% filter(ELM_instance== "1") %>% select(Accession, ELM_class, ELM_type) %>% unique(), aes(y = ELM_type, fill = ELM_type)) + 
  geom_bar( width = 0.5) + 
  ylab("ELM category") + xlab("Class number") + 
  scale_fill_manual(values=colors_ELMtypes[3:5]) + ylim(0, 15) +  
  theme_classic() + theme(text = element_text(size = 18), legend.position="none", 
                          axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))+ 
  scale_y_discrete(limits=rev, labels = rev(elm_categories[3:5])) 
p3

p3 + geom_text(stat='count', aes(label=format(after_stat(count), big.mark=",")), vjust=-2, size=3) +
  geom_text(stat='prop', vjust=-0.25, size=3) 

# SAVE TIFF
tiff("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_DMI_high_instances_classes.tiff", units="cm", width=8, height=6, res=800)
p3
dev.off()
# SAVE PDF
pdf("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_DMI_high_instances_classes.pdf", width=4, height=3)
p3
dev.off()


```

```{r Figure1_Graphic_Supp_Venn}

ggvenn(HURI_ELM_DMI_PPI_list, fill_color = colors_ELMtypes, stroke_size = 0.4, set_name_size = 8, text_size = 5)
ggvenn(HURI_ELM_DMI_PPI_list, c("DOC","LIG","MOD", "TRG"), fill_color = colors_ELMtypes, stroke_size = 0.4, set_name_size = 8, text_size = 5)

p1 <- ggvenn(HURI_ELM_DMI_PPI_list, fill_color = colors_ELMtypes, fill_alpha = 0.6, stroke_size = 0.05, set_name_size = 6, text_size = 0.01)


p2 <- plot_upset(Venn(HURI_ELM_DMI_PPI_list), nintersects = 17, order.set.by = "name", sets.bar.color = colors_ELMtypes, top.bar.numbers.size = 2, relative_height = 1.5)
p2

# # SAVE TIFF
# tiff("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_PPIs_ELM_upset.tiff", units="cm", width=15, height=7, res=800)
# p2
# dev.off()
# 
# # SAVE PDF
# pdf("Figures/SuppFig_DMI_HuRI_graphics/Supp_HuRI_PPIs_ELM_upset.pdf", width=7, height=4)
# p2
# dev.off()
```



```{r Figure2_Graphic_A1}
p1 <- ggplot(HURI_ClinVar_DMI_mut_count_DMI_long, aes(x=pathogenicity, y=value, color=pathogenicity)) + geom_boxplot(width=0.25)+
  xlab( "Clinical relevance") + ylab("Variant number per DMI") + 
  theme_classic() + scale_color_manual(values=colors_Var) + labs(color = "Pathogenicity") +
  theme(text = element_text(size = 20), axis.title.x=element_blank(), axis.text = element_text(color = "black"),
                          axis.line = element_line(size = 1), axis.ticks = element_line(linewidth = 1))
p1
p1 + ylim(0,25)


p2 <- ggplot(HURI_ClinVar_DMI_mut_count_DMI_long, aes(y=pathogenicity, fill=pathogenicity)) + geom_bar( width = 0.5) +
  xlab("DMI number") + ylab("Clinical relevance") + 
  theme_classic() + scale_fill_manual(values=colors_Var) + labs(fill = "Clinical relevance") +
  theme(text = element_text(size = 20), legend.position = "none", axis.text = element_text(color = "black"),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), axis.ticks = element_line(linewidth = 1)) + 
  scale_x_continuous(labels = scales::comma, limits = c(0,11000)) + scale_y_discrete(limits=rev) 

p2
p2 +  geom_text(aes(label = ..count..), stat = "count", vjust = -1, position = position_dodge(0.9), size = 3.5)

# SAVE TIFF
tiff("Figures/Figure2_graphics/HuRI_DMIs_OneMutation.tiff", units="cm", width=12, height=8, res=800)
p2
dev.off()

# SAVE PDF
pdf("Figures/Figure2_graphics/HuRI_DMIs_OneMutation.pdf", width=4, height=4)
p2
dev.off()

HURI_ClinVar_DMI_mut_count_DMI_long %>% filter(pathogenicity=="Uncertain") 
HURI_ClinVar_DMI_mut_count_DMI_long %>% filter(pathogenicity=="Uncertain" | pathogenicity=="Pathogenic") %>% select(-pathogenicity, -value) %>% unique()

```

```{r Figure2_Graphic_A2}
#PPI - Pathogenicity variants
p1 <- ggplot(HURI_ClinVar_DMI_mut_count_long, aes(x=pathogenicity, y=value, color=pathogenicity)) + geom_boxplot(width=0.25)+
  xlab("Clinical relevance") + ylab("Variant number per PPI") + 
  theme_classic() + scale_color_manual(values=colors_Var) + labs(color = "Clinical relevance") +
  theme(text = element_text(size = 20), axis.title.x=element_blank(), axis.text = element_text(color = "black"),
                          axis.line = element_line(size = 1), axis.ticks = element_line(linewidth = 1))
p1
p1 + ylim(0,25)

# PPI with at least 1 variant
p2 <- ggplot(HURI_ClinVar_DMI_mut_count_long, aes(y=pathogenicity, fill=pathogenicity)) + geom_bar( width = 0.5) +
  xlab("PPI number") + ylab("Clinical relevance") + 
  theme_classic() + scale_fill_manual(values=colors_Var) + labs(fill = "Clinical relevance") +
  theme(text = element_text(size = 20), legend.position = "none", axis.text = element_text(color = "black"),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), axis.ticks = element_line(linewidth = 1)) + 
  scale_x_continuous(labels = scales::comma, limits = c(0,3400)) + scale_y_discrete(limits=rev) 
p2
p2 +  geom_text(aes(label = ..count..), stat = "count", vjust = -1, position = position_dodge(0.9), size = 3.5)

# SAVE TIFF
tiff("Figures/Figure2_graphics/HuRI_PPIs_OneMutation.tiff", units="cm", width=12, height=8, res=800)
p2
dev.off()

# SAVE PDF
pdf("Figures/Figure2_graphics/HuRI_PPIs_OneMutation.pdf", width=4, height=4)
p2
dev.off()



```

```{r Figure2_Graphic_B}
plots_title <- "DMI - Region variants"
plots_xlab <- "Protein region"
plots_ylab <- "Variant fraction"

p1 <- ggplot(HURI_ClinVar_mutation, aes(x=fragment_type , fill=pathogenicity)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle(plots_title) +  xlab(plots_xlab) + ylab(plots_ylab) + 
  theme_minimal() + scale_fill_manual(values=colors_Var) + labs(fill = "Pathogenicity") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), axis.text = element_text(color = "black"),
        panel.grid.minor = element_blank(), axis.ticks.y = element_line(size = 1), axis.title.x=element_blank())
p1

# SAVE TIFF
tiff("Figures/Figure2_graphics/HuRI_human_ClinVar_mutations_fragment.tiff", units="cm", width=12, height=8, res=800)
p1
dev.off()

# SAVE PDF
pdf("Figures/Figure2_graphics/HuRI_human_ClinVar_mutations_fragment.pdf", width=6, height=4)
p1
dev.off()

p1 +geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 


plots_title <- "PPI - Pathogenicity variants"
plots_xlab <- "Pathogenicity"
plots_ylab <- "Variant fraction"

p2 <- ggplot(HURI_ClinVar_mutation, aes(x=pathogenicity , fill=fragment_type)) + geom_bar(position = "fill", width = 0.5) +
  ggtitle(plots_title) +  xlab(plots_xlab) + ylab(plots_ylab) + 
  theme_classic() + scale_fill_manual(values=colors_DMI) + labs(fill = "Pathogenicity") +
  theme(text = element_text(size = 20), axis.title.x=element_blank(), 
        axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))
p2
p2 + geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 
```

```{r Figure2_Graphic_D}
#DMI - Motif variants

p1 <- ggplot(HURI_ClinVar_mut_motif, aes(y=ELM_type , fill=pathogenicity)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  xlab("Variant fraction") + ylab("Motif category") + 
  theme_minimal() + scale_fill_manual(values=colors_Var) + labs(fill = "Clinical relevance") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  axis.text = element_text(color = "black"), 
        legend.position = "none", panel.grid.minor = element_blank(),  axis.ticks = element_line(size = 0.5))+ 
  scale_y_discrete(limits=rev, labels = rev(elm_categories)) 

p1
p1 +geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

# SAVE TIFF
tiff("Figures/Figure2_graphics/HuRI_human_ClinVar_mutations_ELM.tiff", units="cm", width=12, height=8, res=800)
p1
dev.off()

# SAVE PDF
pdf("Figures/Figure2_graphics/HuRI_human_ClinVar_mutations_ELM.pdf", width=8, height=4)
p1
dev.off()


```

```{r Figure2_Graphic_C}
corrplot(Fragment_contingency_chisq$residuals, method = 'color', is.cor = FALSE,col = rev(COL2('RdBu')), col.lim= c(-35,35), cl.pos = 'b', addCoef.col = 'grey50', outline=T, addgrid.col="white")

# SAVE TIFF
tiff("Figures/Figure3_graphics/DMI_fragment_mutation_enrichment.tiff", units="cm", width=10, height=20, res=800)
corrplot(Fragment_contingency_chisq$residuals, method = 'color', is.cor = FALSE,col = rev(COL2('RdBu')), col.lim= c(-35, 35), cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")
dev.off()

# SAVE PDF
pdf("Figures/Figure3_graphics/DMI_fragment_mutation_enrichment.pdf", width=4, height=8)
corrplot(Fragment_contingency_chisq$residuals, method = 'color', is.cor = FALSE, col = rev(COL2('RdBu')), col.lim= c(-35, 35), cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")
dev.off()

round(Fragment_contingency_chisq$observed,2)
```

```{r Figure2_Supp_PPIvenn}
ggvenn(HURI_ClinVar_DMI_PPI_list, fill_color = colors_Var, stroke_size = 0.5, set_name_size = 8, text_size = 6)
p <- ggvenn(HURI_ClinVar_DMI_PPI_list, fill_color = colors_Var, fill_alpha = 0.6, stroke_size = 0.05, set_name_size = 6, text_size = 0.01)

# SAVE TIFF
tiff("Figures/Figure2_graphics/HuRI_PPIs_OneMutation_Venn.tiff", units="cm", width=10, height=8, res=800)
p
dev.off()

# SAVE PDF
pdf("Figures/Figure2_graphics/HuRI_PPIs_OneMutation_Venn.pdf", width=4, height=3)
p
dev.off()

p2 <- plot_upset(Venn(HURI_ClinVar_DMI_PPI_list), nintersects = 10, order.set.by = "name", sets.bar.color = colors_Var, top.bar.numbers.size = 3, relative_height = 4)
p2

# SAVE TIFF
tiff("Figures/Figure2_graphics/HuRI_PPIs_OneMutation_Upset.tiff", units="cm", width=12, height=8, res=800)
p2
dev.off()

# SAVE PDF
pdf("Figures/Figure2_graphics/HuRI_PPIs_OneMutation_Upset.pdf", width=6, height=4)
p2
dev.off()

```



```{r Figure3_Graph_domains}
# Heatmap
heatmap_pltt <- 'RdBu'
# p1 <- ggplot(domain_BRET %>% filter(NL_symbol=="CTBP1") %>% mutate(mutant=gsub("CTBP1_", "", mutant)) , 
#              aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_bret50  )) + 
#   geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
#   scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
#   xlab("CTBP1 domain mutants") +
#   theme_minimal() + labs(fill = "Effect\nsize") +
#   theme(text = element_text(size = 10), panel.grid.major = element_blank(), 
#         axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
#         axis.ticks = element_line(size = 1), axis.title.y=element_blank(), axis.title.x=element_blank(),
#         axis.text.x = element_text(angle = 30, vjust = 1, hjust=1),
#         legend.title = element_text(size=15), legend.key.height= unit(0.6, 'cm'), 
#         legend.key.width= unit(0.3, 'cm'), legend.text = element_text(size=10, vjust=1)) +
#   scale_y_discrete(limits=rev)


p1 <- ggplot(domain_BRET %>% filter(NL_symbol=="CTBP1") %>% mutate(mutant=gsub("CTBP1_", "", mutant)) , 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  xlab("CTBP1 domain mutants") +
  theme_minimal() + labs(fill = "Effect\nsize") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)  + 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET )))

p2 <- ggplot(domain_BRET %>% filter(NL_symbol=="CTBP1") %>% mutate(mutant=gsub("CTBP1_", "", mutant)) , 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_cBRET )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") +  xlab("CTBP1 domain mutants") + 
  theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)  + 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET ))) 

grid.arrange(p1, p2,nrow=2, heights=c(9,11))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/DomainValidation_CTBP1.tiff", units="cm", width=8, height=13, res=800)
grid.arrange(p1, p2, heights=c(4,5), nrow=2, bottom=textGrob("CTBP1 domain mutants", gp = gpar(fontsize = 20)) )
dev.off()

# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/DomainValidation_CTBP1.tiff", units="cm", width=11, height=12, res=800)
# grid.arrange(p1, p2,nrow=2, bottom=textGrob("CTBP1 domain mutants", gp = gpar(fontsize = 20)) )
# dev.off()


p1 <- ggplot(domain_BRET %>% filter(NL_symbol=="SPOP") %>% mutate(mutant=gsub("SPOP_", "", mutant)) , 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET[2:3] ))) 

p2 <- ggplot(domain_BRET %>% filter(NL_symbol=="SPOP") %>% mutate(mutant=gsub("SPOP_", "", mutant)) , 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET[2:3] ))) 

grid.arrange(p1, p2,nrow=2, bottom=textGrob("SPOP domain mutants", gp = gpar(fontsize = 20)))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/DomainValidation_SPOP.tiff", units="cm", width=7.5, height=13, res=800)
grid.arrange(p1, p2, heights=c(2.2,3), nrow=2, bottom=textGrob("SPOP domain mutants", gp = gpar(fontsize = 20)) )
dev.off()

# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/DomainValidation_SPOP.tiff", units="cm", width=11, height=11, res=800)
# grid.arrange(p1, p2,nrow=2, bottom=textGrob("SPOP domain mutants", gp = gpar(fontsize = 20)) )
# dev.off()



p1 <- ggplot(domain_BRET %>% filter(NL_symbol=="WWOX") %>% mutate(mutant=gsub("WWOX_", "", mutant)) , 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none", 
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=18, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(category~., scales = "free_y", switch = "y", space = "free_y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET )))

p2 <- ggplot(domain_BRET %>% filter(NL_symbol=="WWOX") %>% mutate(mutant=gsub("WWOX_", "", mutant)) , 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none", 
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), plot.title=element_text(size=18, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y",space = "free_y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET )))

grid.arrange(p1, p2, nrow=2, bottom=textGrob("WWOX domain mutants", gp = gpar(fontsize = 20)))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/DomainValidation_WWOX.tiff", units="cm", width=8, height=21, res=800)
grid.arrange(p1, p2, heights=c(4,5), nrow=2, bottom=textGrob("WWOX domain mutants", gp = gpar(fontsize = 20)))
dev.off()

# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/DomainValidation_WWOX.tiff", units="cm", width=10, height=18, res=800)
# grid.arrange(p1, p2, nrow=2, bottom=textGrob("WWOX domain mutants", gp = gpar(fontsize = 20)))
# dev.off()




p1 <- ggplot(domain_BRET %>% filter(NL_symbol=="PPP3CA") %>% mutate(mutant=gsub("PPP3CA_", "", mutant)) , 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_bret50)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  legend.position = "none", 
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET[c(1,3)] )))

p2 <- ggplot(domain_BRET %>% filter(NL_symbol=="PPP3CA") %>% mutate(mutant=gsub("PPP3CA_", "", mutant)) , 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_cBRET)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  legend.position = "none", 
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET[c(1,3)] )))

grid.arrange(p1, p2, nrow=2, bottom=textGrob("PPP3CA domain mutants", gp = gpar(fontsize = 20)))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/DomainValidation_PPP3CA.tiff", units="cm", width=8, height=13, res=800)
grid.arrange(p1, p2, heights=c(2.2,3), nrow=2, bottom=textGrob("PPP3CA domain mutants", gp = gpar(fontsize = 20)))
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/DomainValidation_PPP3CA.tiff", units="cm", width=9, height=11, res=800)
# grid.arrange(p1, p2, nrow=2, bottom=textGrob("PPP3CA domain mutants", gp = gpar(fontsize = 18)))
# dev.off()


p1 <- ggplot(domain_BRET %>% filter(NL_symbol=="IQCB1") %>% 
               mutate(mCit_plasmid=gsub("_", "\n", gsub("^.*?His3C-", "", mCit_plasmid))) , 
             aes(mCit_plasmid, NL_symbol, fill=tstat_bret50)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(.~mCit_symbol, scales = "free_x", switch = "x",
                 strip = strip_themed(text_x = element_text(size=0)))

p2 <- ggplot(domain_BRET %>% filter(NL_symbol=="IQCB1") %>% mutate(mCit_plasmid=gsub("_", "\n", gsub("^.*?His3C-", "", mCit_plasmid))) , aes(mCit_plasmid, NL_symbol, fill=tstat_cBRET)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(.~mCit_symbol, scales = "free_x", switch = "x",
                 strip = strip_themed(text_x = element_text(size=0),
                                     background_x = elem_list_rect( fill =colors_BRET[2:3] )))
grid.arrange(p1, p2, nrow=2, left=textGrob("IQCB1 domain mutants", gp = gpar(fontsize = 18), rot=90))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/DomainValidation_IQCB1.tiff", units="cm", width=7, height=8, res=800)
grid.arrange(p1, p2, heights=c(2.1,3), nrow=2, bottom=textGrob("Domain mutants", gp = gpar(fontsize = 20)))
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/DomainValidation_IQCB1.tiff", units="cm", width=10, height=9, res=800)
# grid.arrange(p1, p2, nrow=2, bottom=textGrob("Domain mutants", gp = gpar(fontsize = 18)))
# dev.off()
```

```{r Figure3_Graph_motifs}
# Heatmaps 
heatmap_pltt <- 'RdBu'
p1 <- ggplot(motif_BRET %>% filter(NL_symbol=="CTBP1") %>% mutate(mutant=gsub("_([0-9]{1,3})_", "\n\\1-", mutant)), 
             aes(NL_symbol, fct_reorder2(mutant, as.numeric(residue), desc(mCit_symbol)), fill=tstat_bret50)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=12, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET[2:3] ))) 

p2 <- ggplot(motif_BRET %>% filter(NL_symbol=="CTBP1") %>% mutate(mutant=gsub("_([0-9]{1,3})_", "\n\\1-", mutant)), 
             aes(NL_symbol, fct_reorder2(mutant, as.numeric(residue), desc(mCit_symbol)), fill=tstat_cBRET)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.y=element_blank(), plot.title=element_text(size=12, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0))) 

grid.arrange(p1, p2, nrow=1, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90 ))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/MotifValidation_CTBP1.tiff", units="cm", width=8.5, height=6, res=800)
grid.arrange(p1, p2, widths=c(3.8,2), nrow=1, left=textGrob("Motif deletion", gp = gpar(fontsize = 20), rot=90) )
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/MotifValidation_CTBP1.tiff", units="cm", width=9, height=11, res=800)
# grid.arrange(p1, p2, nrow=2, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()


p1 <- ggplot(motif_BRET %>% filter(NL_symbol=="SPOP") %>% mutate(mutant=gsub("_([0-9]{1,3})_", "\n\\1-", mutant)), 
             aes(NL_symbol, fct_reorder(mutant, residue), fill=tstat_bret50)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=12, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y", space = "free_y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET[2:3] )))

p2 <- ggplot(motif_BRET %>% filter(NL_symbol=="SPOP") %>% mutate(mutant=gsub("_([0-9]{1,3})_", "\n\\1-", mutant)), 
             aes( NL_symbol, fct_reorder(mutant, residue), fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(),  legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.y=element_blank(), plot.title=element_text(size=12, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y", space = "free_y",
                 strip = strip_themed(text_y = element_text(size=0))) 

grid.arrange(p1, p2, nrow=1, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90 ))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/MotifValidation_SPOP.tiff", units="cm", width=9, height=11, res=800)
grid.arrange(p1, p2, widths=c(4.3,2), nrow=1, left=textGrob("Motif deletion", gp = gpar(fontsize = 20), rot=90) )
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/MotifValidation_SPOP.tiff", units="cm", width=16, height=9, res=800)
# grid.arrange(p1, p2, nrow=1, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()


p1 <- ggplot(motif_BRET %>% filter(NL_symbol=="WWOX") %>% mutate(mutant=gsub("_", " ", gsub("_([0-9]{1,3})_", "\n\\1-", mutant))), 
             aes( NL_symbol, fct_reorder(mutant, residue), fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none", 
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=12, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y", space = "free_y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET[2:3] )))

p2 <- ggplot(motif_BRET %>% filter(NL_symbol=="WWOX") %>% mutate(mutant=gsub("_", " ",gsub("_([0-9]{1,3})_", "\n\\1-", mutant))), 
             aes( NL_symbol, fct_reorder(mutant, residue), fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none", 
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.y=element_blank(), plot.title=element_text(size=12, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(category~., scales = "free_y", switch = "y", space = "free",
                 strip = strip_themed(text_y = element_text(size=0))) 

grid.arrange(p1, p2, nrow=1, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90 ))


# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/MotifValidation_WWOX.tiff", units="cm", width=9, height=15, res=800)
grid.arrange(p1, p2, widths=c(4.4,2), nrow=1, left=textGrob("Motif deletion", gp = gpar(fontsize = 20), rot=90) )
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/MotifValidation_WWOX.tiff", units="cm", width=16, height=13, res=800)
# grid.arrange(p1, p2, nrow=1, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()
# #p2+ scale_fill_gradient2(guide = guide_legend(), limits = c(-20, 20), high="red", mid = "white",  low="blue")


p1 <- ggplot(motif_BRET %>% filter(NL_symbol=="PPP3CA") %>% 
               mutate(mutant=gsub("_", " ", gsub("_([0-9]{1,3})_", "\n\\1-", mutant))),
             aes( NL_symbol, fct_reorder2(mutant, residue, mCit_symbol), fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none", 
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0),
                                     background_y = elem_list_rect( fill =colors_BRET[3] )))

p2 <- ggplot(motif_BRET %>% filter(NL_symbol=="PPP3CA") %>% 
               mutate(mutant=gsub("_([0-9]{1,3})_", "\n\\1-", mutant)), 
             aes(NL_symbol, fct_reorder2(mutant, residue, mCit_symbol), fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none", 
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.y=element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(category~., scales = "free_y", switch = "y",
                 strip = strip_themed(text_y = element_text(size=0))) 

grid.arrange(p1, p2, widths=c(4.2,2), nrow=1, bottom=textGrob("Motif deletion", gp = gpar(fontsize = 20)) )

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/MotifValidation_PPP3CA.tiff", units="cm", width=11, height=5, res=800)
grid.arrange(p1, p2, widths=c(4,2.4), nrow=1, left=textGrob("Motif deletion", gp = gpar(fontsize = 20), rot=90) )
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/MotifValidation_PPP3CA.tiff", units="cm", width=9, height=8, res=800)
# grid.arrange(p1, p2, nrow=2, left=textGrob("Motif deletion", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()


p1 <- ggplot(motif_BRET %>% filter(NL_symbol=="IQCB1") %>% 
               mutate(NL_plasmid=gsub("_([0-9]{1,3})_", "\n\\1-", gsub("^.*?NL-", "", NL_plasmid))), 
             aes(fct_reorder(fct_reorder2(mutant, residue, mCit_symbol), residue), NL_plasmid, fill=tstat_bret50)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(.~category, scales = "free_x", switch = "x",
                 strip = strip_themed(text_x = element_text(size=0)))

p2 <- ggplot(motif_BRET %>% filter(NL_symbol=="IQCB1") %>% 
               mutate(NL_plasmid=gsub("_([0-9]{1,3})_", "\n\\1-", gsub("^.*?NL-", "", NL_plasmid))),
             aes(fct_reorder2(mutant, residue, mCit_symbol), NL_plasmid, fill=tstat_cBRET)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_grid2(.~category, scales = "free_x", switch = "x",
                 strip = strip_themed(text_x = element_text(size=0),
                                     background_x = elem_list_rect( fill =colors_BRET )))

grid.arrange(p1, p2, nrow=2, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90 ))
# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/MotifValidation_IQCB1.tiff", units="cm", width=10, height=8, res=800)
grid.arrange(p1, p2, heights=c(2, 3), nrow=2, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90) )
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/MotifValidation_IQCB1.tiff", units="cm", width=13, height=10, res=800)
# grid.arrange(p1, p2, nrow=2, left=textGrob("Motif deletions", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()

#p1+ scale_fill_gradient2(guide = guide_legend(), limits = c(-20, 20), high="red", mid = "white",  low="blue")
motif_BRET %>% filter(NL_symbol=="SPOP")
```

```{r Figure3_Graph_variants}
# Heatmap 
heatmap_pltt <- 'RdBu'

p1 <- ggplot(mutants_BRET %>% filter(NL_symbol=="CTBP1" & motif_variant==TRUE) %>% mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(NL_symbol, fct_reorder2(mutant, residue, mCit_symbol), fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_nested(category + ClinVar ~ . , scales = "free_y",switch = "y",space = "free_y",
                  strip = strip_nested(text_y = element_text(size=0),
                    background_y = elem_list_rect(fill = c("green","magenta", "#999999","#999999"))
                 ))

p2 <- ggplot(mutants_BRET %>% filter(NL_symbol=="CTBP1" & motif_variant==TRUE) %>% mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(NL_symbol, fct_reorder2(mutant, residue, mCit_symbol), fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.y=element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_nested(category + ClinVar ~ . , scales = "free_y",switch = "y",space = "free_y",
                  strip = strip_nested(text_y = element_text(size=0)
                 ))
                  
grid.arrange(p1, p2,nrow=1, left=textGrob("Motif variants", gp = gpar(fontsize = 20), rot=90 ))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/VariantTest_CTBP1_motif.tiff", units="cm", width=9, height=9, res=800)
grid.arrange(p1, p2, widths=c(4,2.2), nrow=1, left=textGrob("Motif variants", gp = gpar(fontsize = 20), rot=90) )
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/VariantTest_CTBP1_motif.tiff", units="cm", width=15, height=8, res=800)
# grid.arrange(p1, p2,nrow=1, left=textGrob("Motif variants", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()


p1 <- ggplot(mutants_BRET %>% filter(NL_symbol=="SPOP" & domain_variant==TRUE) %>% mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(category ~  ClinVar , scales = "free", switch = "both", space = "free",
                  strip = strip_nested(text_x = element_text(size=0), text_y = element_text(size=0),
                                       background_y = elem_list_rect( fill =colors_BRET )) )

p2 <- ggplot(mutants_BRET %>% filter(NL_symbol=="SPOP" & domain_variant==TRUE) %>% mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(category ~  ClinVar , scales = "free", switch = "both",space = "free",
                  strip = strip_nested(text_x = element_text(size=0), text_y = element_text(size=0),
                    background_x = elem_list_rect( fill =colors_Var[3:4] ),
                    background_y = elem_list_rect( fill =colors_BRET )) )

grid.arrange(p1, p2, nrow=2, bottom=textGrob("Domain variant", gp = gpar(fontsize = 20)))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/VariantTest_SPOP_domain.tiff", units="cm", width=5.5, height=12, res=800)
grid.arrange(p1, p2, heights=c(2,2.4), nrow=2, bottom=textGrob("Domain variant", gp = gpar(fontsize = 20) ))
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/VariantTest_SPOP_domain.tiff", units="cm", width=9, height=12, res=800)
# grid.arrange(p1, p2, nrow=2, bottom=textGrob("SPOP domain variant", gp = gpar(fontsize = 20)))
# dev.off()


p1 <- ggplot(mutants_BRET %>% filter(NL_symbol=="WWOX" & domain_variant==TRUE) %>% 
               mutate(mutant=gsub("WWOX ", "", mutant)), aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(category ~  ClinVar , scales = "free", switch = "both", space = "free",
                  strip = strip_nested(text_x = element_text(size=0), text_y = element_text(size=0),
                                       background_y = elem_list_rect( fill =colors_BRET )) )
                 
p2 <- ggplot(mutants_BRET %>% filter(NL_symbol=="WWOX" & domain_variant==TRUE) %>% 
               mutate(mutant=gsub("WWOX ", "", mutant)) , aes(fct_reorder(mutant, residue), mCit_symbol, fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(category ~  ClinVar , scales = "free", switch = "both", space = "free",
                  strip = strip_nested(text_x = element_text(size=0), text_y = element_text(size=0),
                    background_x = elem_list_rect( fill =colors_Var[3:4] ),
                    background_y = elem_list_rect( fill =colors_BRET )) )

grid.arrange(p1, p2, nrow=2, bottom=textGrob("WWOX domain variants", gp = gpar(fontsize = 20)))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/VariantTest_WWOX_domain.tiff", units="cm", width=8, height=17, res=800)
grid.arrange(p1, p2, nrow=2, heights=c(2,2.5), bottom=textGrob("WWOX domain variants", gp = gpar(fontsize = 20)))
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/VariantTest_WWOX_domain.tiff", units="cm", width=10, height=18, res=800)
# grid.arrange(p1, p2, nrow=2, bottom=textGrob("WWOX domain variants", gp = gpar(fontsize = 20)))
# dev.off()


p1 <- ggplot(mutants_BRET %>% filter(NL_symbol=="WWOX" & motif_variant==TRUE) %>% mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(NL_symbol, fct_reorder2(mutant, residue, mCit_symbol), fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_nested(category + ClinVar ~ . , scales = "free_y",switch = "y",space = "free_y",
                  strip = strip_nested(text_y = element_text(size=0),
                    background_y = elem_list_rect(fill = c("green","magenta", "#999999","#999999"))
                 ))

p2 <- ggplot(mutants_BRET %>% filter(NL_symbol=="WWOX" & motif_variant==TRUE) %>% mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(NL_symbol, fct_reorder2(mutant, residue, mCit_symbol), fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.y=element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev)+ 
  facet_nested(category + ClinVar ~ . , scales = "free_y",switch = "y", space = "free_y",
                  strip = strip_nested(text_y = element_text(size=0))
                 )
grid.arrange(p1, p2, nrow=1, left=textGrob("Motif variants", gp = gpar(fontsize = 20), rot=90 ))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/VariantTest_WWOX_motif.tiff", units="cm", width=9, height=14, res=800)
grid.arrange(p1, p2, nrow=1, widths=c(3,1.7), left=textGrob("Motif variants", gp = gpar(fontsize = 20), rot=90 ))
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/VariantTest_WWOX_motif.tiff", units="cm", width=14, height=11, res=800)
# grid.arrange(p1, p2, nrow=1, left=textGrob("Motif variants", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()

p1 <- ggplot(mutants_BRET %>% filter(NL_symbol=="PPP3CA" & motif_variant==TRUE) %>% mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(NL_symbol, fct_reorder(mutant, residue), fill=tstat_bret50  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x=element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_nested(category + ClinVar ~ . , scales = "free_y",switch = "y",space = "free_y",
                  strip = strip_nested(text_y = element_text(size=0),
                    background_y = elem_list_rect(fill = c("magenta", "#999999"))
                 ))

p2 <- ggplot(mutants_BRET %>% filter(NL_symbol=="PPP3CA" & motif_variant==TRUE) %>% mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(NL_symbol, fct_reorder(mutant, residue), fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(), 
        plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_nested(category + ClinVar ~ . , scales = "free_y",switch = "y",space = "free_y",
                  strip = strip_nested(text_y = element_text(size=0),
                    background_y = elem_list_rect(fill = c("magenta", "#999999"))
                 ))

grid.arrange(p1, p2, nrow=2, left=textGrob("Motif variant", gp = gpar(fontsize = 20), rot=90 ))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/VariantTest_PPP3CA_motif.tiff", units="cm", width=7, height=7, res=800)
grid.arrange(p1, p2, nrow=2, heights=c(2,2.4), left=textGrob("Motif variant", gp = gpar(fontsize = 20), rot=90 ))
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/VariantTest_PPP3CA_motif.tiff", units="cm", width=9, height=8, res=800)
# grid.arrange(p1, p2, nrow=2, left=textGrob("Motif variant", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()



p1 <- ggplot(mutants_BRET %>% filter(NL_symbol=="IQCB1" & motif_variant==TRUE) %>% 
               mutate(mutant=gsub(" ", "\n", mutant)), 
             aes(fct_reorder(mutant,  mCit_symbol),  NL_symbol, fill=tstat_bret50)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x=element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  facet_grid2(.~ClinVar, scales = "free_x", switch = "x",space = "free_x",
                 strip = strip_themed(text_x = element_text(size=0)))
  #facet_nested(. ~ category + ClinVar , scales = "free_x",switch = "x",space = "free_x",strip = strip_nested(text_x = element_text(size=0)))

p2 <- ggplot(mutants_BRET %>% filter(NL_symbol=="IQCB1" & motif_variant==TRUE) %>% 
               mutate(mutant=gsub(" ", "\n", mutant))
             , aes(fct_reorder(mutant,  mCit_symbol), NL_symbol, fill=tstat_cBRET  )) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), plot.title=element_text(size=15, hjust=0.95)) +
  facet_grid2(.~ClinVar, scales = "free_x", switch = "x",space = "free_x",
                 strip = strip_themed(text_x = element_text(size=0),
                                     background_x = elem_list_rect( fill =colors_Var[3:4] )))
  #facet_nested(. ~ category + ClinVar , scales = "free_x",switch = "x",space = "free_x",  strip = strip_nested(text_x = element_text(size=0),  background_x = elem_list_rect(fill = c("green", "magenta", "#d52221", "#999999", "#999999"))  ))
grid.arrange(p1, p2, nrow=2, bottom=textGrob("Domain variants", gp = gpar(fontsize = 20) ))

# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/VariantTest_IQCB1_domain.tiff", units="cm", width=13, height=10, res=800)
grid.arrange(p1, p2, nrow=2, heights=c(2.4,4), bottom=textGrob("Domain variants", gp = gpar(fontsize = 20) ))
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/VariantTest_IQCB1_domain.tiff", units="cm", width=18, height=9, res=800)
# grid.arrange(p1, p2, nrow=2, bottom=textGrob("Domain variants", gp = gpar(fontsize = 20) ))
# dev.off()

p1 <- ggplot(mutants_BRET %>% filter(NL_symbol=="IQCB1" & domain_variant==TRUE) %>% 
               mutate(mutant=gsub("IQCB1 ", "", mutant), 
                      residue=as.numeric(gsub("[A-Z]", "", gsub("IQCB1 ", "", mutant)))), 
             aes(mCit_symbol, fct_reorder(mutant, residue), fill=tstat_bret50)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_BRET50), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("BRET50") +  theme_minimal() + labs(fill = "BRET50\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_blank(), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(ClinVar~category , scales = "free", switch = "both",space = "free",
                  strip = strip_nested(text_x = element_text(size=0), text_y = element_text(size=0),
                    background_y = elem_list_rect( fill = colors_Var[4])) )

p2 <- ggplot(mutants_BRET %>% filter(NL_symbol=="IQCB1" & domain_variant==TRUE) %>% 
               mutate(mutant=gsub("IQCB1 ", "", mutant), 
                      residue=as.numeric(gsub("[A-Z]", "", gsub("IQCB1 ", "", mutant)))) , 
             aes(mCit_symbol, fct_reorder(mutant, residue), fill=tstat_cBRET)) + 
  geom_tile(color = "white", lwd = 1.2) + geom_text(aes(label=sig_maxBRET), size=7) +
  scale_fill_continuous_divergingx(palette = heatmap_pltt, rev = TRUE, mid = 0, limits = c(-20, 20), na.value = "grey80")+
  ggtitle("maxBRET") + theme_minimal() + labs(fill = "maxBRET\neffect size") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position = "none",
        axis.text = element_text(color = "black"), panel.grid.minor = element_blank(), 
        axis.ticks = element_line(size = 0), axis.title.y=element_blank(), axis.title.x=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1),
        legend.title = element_text(size=10), legend.key.height= unit(0.4, 'cm'), 
        legend.key.width= unit(0.2, 'cm'), legend.text = element_text(size=7, vjust=1), plot.title=element_text(size=15, hjust=0.95)) +
  scale_y_discrete(limits=rev) + 
  facet_grid2(ClinVar~category , scales = "free", switch = "both",space = "free",
                  strip = strip_nested(text_x = element_text(size=0), text_y = element_text(size=0),
                    background_x = elem_list_rect( fill = colors_BRET ),
                    background_y = elem_list_rect( fill = colors_Var[4])) )

grid.arrange(p1, p2, nrow=2, left=textGrob("Motif variants", gp = gpar(fontsize = 20), rot=90 ))
# SAVE TIFF
tiff("/Users/jesusav/Desktop/Heatmaps/VariantTest_IQCB1_motif.tiff", units="cm", width=9.5, height=10, res=800)
grid.arrange(p1, p2, nrow=2,  heights=c(2.7,4), left=textGrob("IQCB1 motif variants", gp = gpar(fontsize = 20), rot=90 ))
dev.off()
# # SAVE TIFF
# tiff("Figures/Figure3_graphics/Heatmaps/VariantTest_IQCB1_motif.tiff", units="cm", width=11, height=12, res=800)
# grid.arrange(p1, p2, nrow=2,left=textGrob("IQCB1 motif variants", gp = gpar(fontsize = 20), rot=90 ))
# dev.off()

  
```

```{r Figure3_Graph_ComplexHeatmap_draft, include=F}
### https://jokergoo.github.io/ComplexHeatmap-reference/book/a-single-heatmap.html
library(ComplexHeatmap)
library(circlize)

mat <- temp %>% select(mutant, mCit_symbol, tstat_bret50) %>% pivot_wider(names_from = mutant, values_from = tstat_bret50)
mat_1 <- as.matrix(mat[,2:4])
rownames(mat_1) <- mat$mCit_symbol

ha <- HeatmapAnnotation(
  Interaction = mat$mCit_symbol,
  col = list(Interaction = c("CTBP2" = "blue", "DMRTB1" = "red", "IKZF1" = "green"))
)
Heatmap(mat_1, 
        name = "maxBRET\neffect size",
        col = colorRamp2(c(-20, 0, 20), c("blue", "white", "red")),
        right_annotation = rowAnnotation(Interaction = mat$mCit_symbol),
        row_names_side = "left",
        show_column_names = TRUE,
        cluster_rows = FALSE, cluster_columns = FALSE)
       
```

```{r Figure3_Graph_titrationPlots}
for(i in 1:nrow(BRET_mutant_pairs)){
  pair_info <- BRET_mutant_pairs[i,] %>% unlist() %>% as.vector()
  
  project <- pair_info[1]
  NL <- pair_info[2]
  mCIT <- pair_info[3]
  
  plot_data <- BRET_values_info %>% filter(project_id==project & NL_plasmid_id==NL & mCit_plasmid_id==mCIT & test=="mutant")
  WT_data <-  BRET_values_info %>% filter(project_id==project & NL_protein==pair_info[7] & mCit_protein==pair_info[8] & test=="WT")
  plot_data <- rbind(plot_data, WT_data)
  
  plot_title <- paste(gsub(" ([0-9]{1,3}del)", "-\\1", gsub("_", " ", pair_info[5])), 
                      gsub(" ([0-9]{1,3}del)", "-\\1", gsub("_", " ", pair_info[6])), sep=" - ")
  if(pair_info[4]=="BRET50"){
    test_label <- "BRET50"
    
    plot_fit <- BRET_fits %>% filter(project_id==project & NL_plasmid_id==NL & mCit_plasmid_id==mCIT)%>% mutate(test="mutant")
    WT_fit <-  BRET_fits %>% filter(project_id==project & NL_plasmid_id==unlist(WT_data$NL_plasmid_id[1]) &
                                      mCit_plasmid_id==unlist(WT_data$mCit_plasmid_id[1])) %>% mutate(test="WT")
    plot_fit <- rbind(plot_fit, WT_fit)
    
    plot_fit_data <- tibble(repl_id=character(), cBRET=double(), ratio=double(), test=character()) 
    # cBRET = bret_max * expr_ratio / (bret50 + expr_ratio)
    for(i in 1:nrow(plot_fit)){
      line_info <- plot_fit[i,] %>% unlist() %>% as.vector()
      
      line_test <- line_info[10]
      replicate_id <- line_info[4]
      bret50 <- as.numeric(line_info[5])
      bretmax <-  as.numeric(line_info[7])
      
      expr_ratio_max <- plot_data %>% filter(repl_id==replicate_id & test==line_test) %>% select(ratio) %>% max()
      
      expr_ratio_line <- seq(0, expr_ratio_max, by = expr_ratio_max/100)
      cBRET_line <- bretmax * expr_ratio_line / (bret50 + expr_ratio_line)
      
      plot_fit_line <- tibble(repl_id=replicate_id, cBRET=cBRET_line, ratio=expr_ratio_line, test=line_test) 
      plot_fit_data <- rbind(plot_fit_data, plot_fit_line)
    }

  p <- ggplot() + geom_point(data=plot_data, size=2.2, aes(x=ratio, y = cBRET, color=test) ) + 
      geom_line(data=plot_fit_data, size=0.8, aes(x=ratio, y = cBRET, group=interaction(test, repl_id), color=test))+
      ggtitle(plot_title) +  ylab("BRET") + xlab("Protein Acc/Don") +
      #annotate(geom="text", label=test_label, x=max(plot_data$ratio)*0.9, y=max(plot_data$cBRET)/3, size=3, color="red")+
      theme_classic() + labs(fill = "Pathogenicity") + scale_color_manual(values=c("#d52221","black")) +
      theme(text = element_text(size = 18), axis.text = element_text(color = "black"), legend.position="none", 
            plot.title= element_text(size = 14), axis.ticks = element_line(linewidth = 1)) +
      scale_x_continuous(breaks = scales::pretty_breaks(n = 2)) + 
      scale_y_continuous(labels = function(x) number(x, accuracy = 0.1), 
                         breaks = scales::pretty_breaks(n = 3), limits=c(0,max((plot_data$cBRET))*1.1))
  
  file_name <- paste("/Users/jesusav/Desktop/Titrations/",pair_info[5],"_",pair_info[6],".tiff",sep="")
    tiff(file_name, units="cm", width=8, height=6, res=800)
    print(p)
    dev.off()
    
    
  } 
  else if(pair_info[4]=="maxBRET"){
    test_label <- "maxBRET"
    
    p <- ggplot(plot_data, aes(x=ratio, y = cBRET, color=test) ) + 
      geom_point(size=2.2) +
      ggtitle(plot_title) +  ylab("BRET") + xlab("Protein Acc/Don") +
      #annotate(geom="text", label=test_label, x=max(plot_data$ratio)*0.9, y=max(plot_data$cBRET)/3, size=3, color="red")+
      theme_classic() + labs(fill = "Pathogenicity") + scale_color_manual(values=c("#d52221","black")) +
      theme(text = element_text(size = 18), axis.text = element_text(color = "black"), legend.position="none", 
            plot.title= element_text(size = 14), axis.ticks = element_line(linewidth = 1)) +
      scale_x_continuous(breaks = scales::pretty_breaks(n = 2)) + 
      scale_y_continuous(labels = function(x) number(x, accuracy = 0.1), 
                         breaks = scales::pretty_breaks(n = 3), limits=c(0,max((plot_data$cBRET))*1.1)) 
    
  file_name <- paste("/Users/jesusav/Desktop/Titrations/",pair_info[5],"_",pair_info[6],".tiff",sep="")
    tiff(file_name, units="cm", width=8, height=6, res=800)
    print(p)
    dev.off()
    
    
  } 
  
  
  
  
}

#BRET_results %>% filter(NL_symbol=="CTBP1" & mCit_symbol=="IKZF1") %>% arrange(project_id) %>% select(NL_plasmid, mCit_plasmid, tstat_bret50, tstat_cBRET, sig_BRET50, sig_maxBRET)
```



```{r Figure4_Graph_E}
p1 <- ggplot(BRET_mutants_update_motif, aes(x=mutation_location, fill=as.factor(exp_conclusion))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  xlab("Motif region") + ylab("Fraction DMIs perturbed") + 
  theme_minimal() + scale_fill_manual(values=c("darkgrey","#006AB4")) + labs(fill = "Interaction") +
  theme(text = element_text(size = 20), axis.title.x=element_blank() , panel.grid.minor = element_blank(), legend.position="none",
        axis.ticks.y = element_line(size = 0.5), panel.grid.major = element_blank(), axis.text = element_text(color = "black"))  

p1 +geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 


# SAVE TIFF
tiff("Figures/Figure4_graphics/BRET_mutants_results_2.tiff", units="cm", width=8, height=8, res=800)
p1
dev.off()


p1 <- ggplot(BRET_mutants_update_motif %>% select(aa_change, mutation_location, exp_conclusion) %>% unique(), aes(x=mutation_location, fill=as.factor(exp_conclusion))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  xlab("Motif region") + ylab("Fraction DMIs perturbed") + 
  theme_minimal() + scale_fill_manual(values=c("darkgrey","#006AB4")) + labs(fill = "Interaction") +
  theme(text = element_text(size = 20), axis.title.x=element_blank() , panel.grid.minor = element_blank(), legend.position="none",
        axis.ticks.y = element_line(size = 0.5), panel.grid.major = element_blank(), axis.text = element_text(color = "black"))  

p1 +geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 




p1 <- ggplot(BRET_mutants_update_motif, aes(x=interaction(mutation_location,exp_conclusion), y=tstat_bret50)) + 
  geom_quasirandom(size=2, alpha=1, aes(color=mutation_location)) + #geom_beeswarm(corral = "wrap", alpha=0.2, size=0.1) + 
  geom_boxplot(color="black", alpha=0.1, width=0.5, lwd=0.4, outlier.size = 1, outlier.alpha = 1) + 
  scale_color_manual(values=c("forestgreen","limegreen")) +
  ggtitle("BRET50") +  xlab("Group") + ylab("Effect size") +  ylim(-25,5) +
  theme_classic() + 
  theme(text = element_text(size = 15), legend.position="none", axis.text = element_text(color = "black"), 
         axis.title.x=element_blank(), plot.title=element_text(size=12), 
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1))  +
  scale_x_discrete(labels = c("Core","Flank","Core","Flank")) 

p2 <- ggplot(BRET_mutants_update_motif, aes(x=interaction(mutation_location,exp_conclusion), y=tstat_cBRET)) + 
  geom_quasirandom(size=2, alpha=1, aes(color=mutation_location)) + #geom_beeswarm(corral = "wrap", alpha=0.2, size=0.1) + 
  geom_boxplot(color="black", alpha=0.1, width=0.5, lwd=0.4, outlier.size = 1, outlier.alpha = 1) + 
  scale_color_manual(values=c("forestgreen","limegreen")) +
  ggtitle("maxBRET") +  xlab("Group") + ylab("Effect size") + ylim(-25,5) +
  theme_classic() + 
  theme(text = element_text(size = 15), legend.position="none", axis.text = element_text(color = "black"), 
        axis.text.y = element_blank(), axis.title=element_blank(), plot.title=element_text(size=12), 
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1) ) +
  scale_x_discrete(labels = c("Core","Flank","Core","Flank")) 

grid.arrange(p1, p2,  widths=c(6,5), nrow=1)

# SAVE TIFF
tiff("Figures/Figure4_graphics/BRET_mutants_scores_BRET.tiff", units="cm", width=11, height=8, res=800)
grid.arrange(p1, p2, widths=c(6,5),  nrow=1)
dev.off()


```

```{r Figure4_Graph_G}
p1 <- ggplot(BRET_mutants_update , aes(x=structural_region, y=tstat_bret50)) + 
  geom_quasirandom(size=2, alpha=1, aes(color=exp_conclusion)) + #geom_beeswarm(corral = "wrap", alpha=0.2, size=0.1) + 
  geom_boxplot(color="black", alpha=0.1, width=0.5, lwd=0.4, outlier.size = 1, outlier.alpha = 1) + 
  scale_color_manual(values=c("darkgrey","#006AB4")) +
  ggtitle("BRET50") +  xlab("Group") + ylab("Effect size") +  ylim(-25,5) +
  theme_classic() + 
  theme(text = element_text(size = 18), legend.position="none", axis.text = element_text(color = "black"), 
         axis.title.x=element_blank(), plot.title=element_text(size=15), 
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1)) +
  scale_x_discrete(labels = strc_categories) 

p2 <- ggplot(BRET_mutants_update , aes(x=structural_region, y=tstat_cBRET)) + 
  geom_quasirandom(size=2, alpha=1, aes(color=exp_conclusion)) + #geom_beeswarm(corral = "wrap", alpha=0.2, size=0.1) + 
  geom_boxplot(color="black", alpha=0.1, width=0.5, lwd=0.4, outlier.size = 1, outlier.alpha = 1) + 
  scale_color_manual(values=c("darkgrey","#006AB4")) +
  ggtitle("maxBRET") +  xlab("Group") + ylab("Effect size") +  ylim(-25,5) +
  theme_classic() + 
  theme(text = element_text(size = 18), legend.position="none", axis.text = element_text(color = "black"), 
        axis.text.y = element_blank(), axis.title=element_blank(), plot.title=element_text(size=15), 
        axis.text.x = element_text(angle = 30, vjust = 1, hjust=1) ) +
  scale_x_discrete(labels = strc_categories) 
grid.arrange(p1, p2, widths=c(7,6), nrow=1)


# SAVE TIFF
tiff("Figures/Figure4_graphics/BRET_mutants_regions_BRET.tiff", units="cm", width=15, height=10, res=800)
grid.arrange(p1, p2, widths=c(7,6), nrow=1)
dev.off()
```

```{r Figure4_Graph_H}
ggplot(BRET_mutants_update, aes(y=mutation_location, fill=as.factor(exp_conclusion))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ylab("Fraction DMIs perturbed") + xlab("Mutation location") + 
  theme_minimal() + scale_fill_manual(values=c("darkgrey","#006AB4")) + labs(fill = "Interaction") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), legend.position="none", 
        axis.ticks = element_line(size = 0.5), panel.grid.minor = element_blank(), axis.text = element_text(color = "black"))  + 
  scale_y_discrete(limits=rev, labels = rev(strc_categories)) 

p1 <- ggplot(BRET_mutants_update, aes(y=structural_region, fill=as.factor(exp_conclusion))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ylab("Structural region") + xlab("Fraction DMIs perturbed") + 
  theme_minimal() + scale_fill_manual(values=c("darkgrey","#006AB4")) + labs(fill = "Interaction") +
  theme(text = element_text(size = 20),  panel.grid.minor = element_blank(), legend.position="none",
        axis.ticks = element_line(size = 0.5), panel.grid.major = element_blank(), 
        axis.ticks.y = element_line(size = 0), axis.text = element_text(color = "black"))  + 
  scale_y_discrete(limits=rev, labels = rev(strc_categories)) 

p1 +  geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

# SAVE TIFF
tiff("Figures/Figure4_graphics/BRET_mutants_regions_2.tiff", units="cm", width=12, height=8, res=800)
p1
dev.off()

# SAVE PDF
pdf("Figures/Figure4_graphics/BRET_mutants_regions_2.pdf", width=7, height=4)
p1
dev.off()


p2 <- ggplot(BRET_mutants_update %>% select(aa_change, structural_region, exp_conclusion) %>% unique(), 
             aes(y=structural_region, fill=as.factor(exp_conclusion))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  xlab("Structural region") + ylab("Fraction DMIs perturbed") + 
  theme_minimal() + scale_fill_manual(values=c("darkgrey","#006AB4")) + labs(fill = "Disruption") +
  theme(text = element_text(size = 20),  panel.grid.minor = element_blank(), legend.position="none",
        axis.ticks = element_line(size = 0.5), panel.grid.major = element_blank(), 
        axis.ticks.y = element_line(size = 0), axis.text = element_text(color = "black"))   + 
  scale_y_discrete(limits=rev, labels = rev(strc_categories)) 

p2 +  geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 


p3 <- ggplot(BRET_mutants_update %>% select(aa_change, structural_region, exp_conclusion,AlphaMissense) %>% unique(), 
             aes(y=structural_region, fill=AlphaMissense)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ylab("Structural region") + xlab("Variant fraction") + 
  theme_minimal() + scale_fill_manual(values=colors_Var) + labs(fill = "AlphaMissense") +
  theme(text = element_text(size = 20),  panel.grid.minor = element_blank(), legend.position="none",
        axis.ticks = element_line(size = 0.5), panel.grid.major = element_blank(), 
        axis.ticks.y = element_line(size = 0), axis.text = element_text(color = "black"))   + 
  scale_y_discrete(limits=rev, labels = rev(strc_categories)) 

p3 +  geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

# SAVE TIFF
tiff("Figures/Figure4_graphics/BRET_mutants_AM_2.tiff", units="cm", width=12, height=8, res=800)
p3
dev.off()

# SAVE PDF
pdf("Figures/Figure4_graphics/BRET_mutants_AM_2.pdf", width=7, height=4)
p3
dev.off()

p4 <- ggplot(BRET_mutants_update %>% select(aa_change, structural_region, exp_conclusion, ClinVar) %>% unique(), 
             aes(y=structural_region, fill=ClinVar)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ylab("Structural region") + xlab("Variant fraction") + 
  theme_minimal() + scale_fill_manual(values=colors_Var[3:4]) + labs(fill = "ClinVar") +
  theme(text = element_text(size = 20),  panel.grid.minor = element_blank(), legend.position="none",
        axis.ticks = element_line(size = 0.5), panel.grid.major = element_blank(), 
        axis.ticks.y = element_line(size = 0), axis.text = element_text(color = "black"))   + 
  scale_y_discrete(limits=rev, labels = rev(strc_categories)) 

p4 +  geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 


# SAVE TIFF
tiff("Figures/Figure4_graphics/BRET_mutants_CV_2.tiff", units="cm", width=12, height=8, res=800)
p4
dev.off()

# SAVE PDF
pdf("Figures/Figure4_graphics/BRET_mutants_CV_2.pdf", width=7, height=4)
p4
dev.off()



BRET_mutants_update %>% filter(structural_region=="m_interface")
ggplot(BRET_mutants_update %>% select(aa_change, structural_region, exp_conclusion,AlphaMissense) %>% unique(), 
             aes(y=structural_region, fill=AlphaMissense)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ylab("Structural region") + xlab("Variant fraction") + 
  theme_minimal() + scale_fill_manual(values=colors_Var, guide = guide_legend(theme = theme(
      legend.direction = "horizontal", legend.title = element_text(size=20), legend.text = element_text(size=20)))) + labs(fill = "AlphaMissense") + guides(col = guide_legend(ncol = 3))
```

```{r Figure4_Graph_Supp_Scores}
p1 <- ggplot(BRET_mutants_update_motif, aes(x=mutation_location, y=pLDDT)) + 
  geom_quasirandom(size=2, alpha=1, aes(color=mutation_location)) + #geom_beeswarm(corral = "wrap", alpha=0.2, size=0.1) + 
  geom_boxplot(color="black", alpha=0.1, width=0.5, lwd=0.4, outlier.size = 1, outlier.alpha = 1) + 
  geom_hline(yintercept = 80, color="red") + ylim(25,100) + scale_color_manual(values=c("forestgreen","limegreen")) +
  ggtitle("BRET motif mut") +  xlab("Motif region") + ylab("Residue pLDDT") + 
  theme_classic() + 
  theme(text = element_text(size = 18), legend.position="none", axis.text = element_text(color = "black")) 

p2 <- ggplot(BRET_mutants_update_motif, aes(x=mutation_location, y=RSA)) + 
  geom_quasirandom(size=2, alpha=1, aes(color=mutation_location)) + #geom_beeswarm(corral = "wrap", alpha=0.2, size=0.1) + 
  geom_boxplot(color="black", alpha=0.1, width=0.5, lwd=0.4, outlier.size = 1, outlier.alpha = 1) + 
  geom_hline(yintercept = 0.36, color="red") + ylim(0,1) + scale_color_manual(values=c("forestgreen","limegreen")) +
  ggtitle("BRET motif mut") +  xlab("Motif region") + ylab("Residue RSA") + 
  theme_classic() + 
  theme(text = element_text(size = 18), legend.position="none", axis.text = element_text(color = "black")) 

p3 <- ggplot(BRET_mutants_update_motif, aes(x=mutation_location, y=deltaRSA)) + 
  geom_quasirandom(size=2, alpha=1, aes(color=mutation_location)) + #geom_beeswarm(corral = "wrap", alpha=0.2, size=0.1) + 
  geom_boxplot(color="black", alpha=0.1, width=0.5, lwd=0.4, outlier.size = 1, outlier.alpha = 1) + 
  geom_hline(yintercept = -0.05, color="red") + ylim(-1,0) + scale_color_manual(values=c("forestgreen","limegreen")) +
  ggtitle("BRET motif mut") +  xlab("Motif region") + ylab("Residue \u0394 RSA") + 
  theme_classic() + 
  theme(text = element_text(size = 18), legend.position="none", axis.text = element_text(color = "black")) 

grid.arrange(p1, p2, p3,  nrow=1)

# SAVE TIFF
tiff("Figures/Figure4_graphics/BRET_mutants_scores.tiff", units="cm", width=20, height=8, res=800)
grid.arrange(p1, p2, p3,  nrow=1)
dev.off()

# SAVE PDF
pdf("Figures/Figure4_graphics/BRET_mutants_scores.pdf", width=8, height=4)
grid.arrange(p1, p2, p3,  nrow=1)
dev.off()
```



```{r Figure5_Graph_A}
p1 <- ggplot(scores_count, aes(x=bin, y=count_dmi)) +  geom_line() + 
  annotate(geom="text", label="0.6", x=0.68, y=6000, size=5, color="red")+ 
  geom_vline(xintercept = 0.6, size=0.5, color="red") + 
  ggtitle("DMI AF models scores") + xlab("Model confidence") + ylab("Model count") + 
  theme_classic() + 
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1)) + 
  scale_y_continuous(labels = scales::comma, limits = c(0,8000)) 

p1 + geom_point() +
  annotate(geom="text", label="Step = 0.05", x=0.2, y=4000, size=3.5)+
  geom_text(aes(label = count_dmi), vjust = -1, hjust = -0.2, size = 3) 

# SAVE TIFF
tiff("Figures/Figure5_graphics/ModelScore_cummulative.tiff", units="cm", width=10, height=8, res=800)
p1
dev.off()

# SAVE PDF
pdf("Figures/Figure5_graphics/ModelScore_cummulative.pdf", width=4, height=3)
p1
dev.off()

p2 <- ggplot(scores_count, aes(x=bin, y=count_ppi)) +  geom_line() + 
  annotate(geom="text", label="0.6", x=0.68, y=1900, size=5, color="red")+ 
  geom_vline(xintercept = 0.6, size=0.5, color="red") + 
  ggtitle("DMI AF models scores") + xlab("Model confidence") + ylab("PPI count") + 
  theme_classic() + 
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1)) + 
  scale_y_continuous(labels = scales::comma, limits = c(0,2100)) 

p2 + geom_point() +
  annotate(geom="text", label="Step = 0.05", x=0.2, y=4000, size=3.5)+
  geom_text(aes(label = count_ppi), vjust = -1, hjust = -0.2, size = 3) 
  
# SAVE TIFF
tiff("Figures/SuppFig_AFmodels_graphics/ModelScore_cummulative_PPI.tiff", units="cm", width=10, height=8, res=800)
p2
dev.off()

# SAVE PDF
pdf("Figures/SuppFig_AFmodels_graphics/ModelScore_cummulative_PPI.pdf", width=4, height=3)
p2
dev.off()

coeff <-  max(scores_count$count_ppi) /max(scores_count$count_dmi)

p3 <- ggplot(scores_count, aes(x=bin)) +  geom_line( aes(y=count_dmi), size=1) + geom_line( aes(y=count_ppi / coeff)) + 
  annotate(geom="text", label=c("0.6","DMI","PPI"), x=c(0.68,0.3,0.45), y=c(7700,6400,7600), size=5, color=c("red","black","black") )+ 
  geom_vline(xintercept = 0.6, size=0.5, color="red") + 
  ggtitle("HuRI DMI scores") + xlab("RF DMI score") +
  theme_classic() + 
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))+ 
  scale_y_continuous(labels = scales::comma, , limits = c(0,8000), name = "DMI count", sec.axis = sec_axis(~.*coeff, name="PPI count"))
p3

# SAVE TIFF
tiff("Figures/SuppFig_AFmodels_graphics/DMIScore_cummulative_DMI_PPI.tiff", units="cm", width=12, height=8, res=800)
p3
dev.off()

# SAVE PDF
pdf("Figures/SuppFig_AFmodels_graphics/DMIScore_cummulative_DMI_PPI.pdf", width=5, height=3)
p3
dev.off()

```

```{r Figure5_Graph_B}
# Highscored DMI by category
p1 <- ggplot(DMI_AF_metrics  %>% filter(model_score>=0.6), aes(y=ELM_type, fill=ELM_type)) + geom_bar(width = 0.5) +
  xlab("DMI number") + ylab("ELM category") + 
  theme_classic() + scale_fill_manual(values=colors_ELMtypes[c(1:4,6)]) + labs(fill = "ELM category") +
  theme(text = element_text(size = 20), legend.position = "none", axis.text = element_text(color = "black"), 
        axis.ticks = element_line(linewidth = 1), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1))  + 
  scale_x_continuous(labels = scales::comma, limits = c(0,2600)) + 
  scale_y_discrete(limits=rev, labels = rev(elm_categories[c(1:4,6)])) 
p1 +  geom_text(aes(label = ..count..), stat = "count", vjust = -1, position = position_dodge(0.9), size = 3.5)

# SAVE TIFF
tiff("Figures/Figure5_graphics/AF_high_DMIs.tiff", units="cm", width=10, height=7, res=800)
p1
dev.off()

# SAVE PDF
pdf("Figures/Figure5_graphics/AF_high_DMIs.pdf", width=4, height=3)
p1
dev.off()

#"PPI with at least 1 DMI category"
p2 <- ggplot(DMI_AF_class_table, aes(y=ELM_type, fill=ELM_type)) + geom_bar(width = 0.5) +
  xlab("PPI number") + ylab("ELM category") + 
  theme_classic() + scale_fill_manual(values=colors_ELMtypes[c(1:4,6)]) + labs(fill = "ELM category") +
  theme(text = element_text(size = 20), legend.position = "none",  axis.text = element_text(color = "black"), 
        axis.ticks = element_line(linewidth = 1), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1))  + 
  scale_x_continuous(labels = scales::comma, limits = c(0,1000))  +
  scale_y_discrete(limits=rev, labels = rev(elm_categories[c(1:4,6)])) 
p2 +  geom_text(aes(label = ..count..), stat = "count", vjust = -1, position = position_dodge(0.9), size = 3.5)

# SAVE TIFF
tiff("Figures/Figure5_graphics/AF_high_PPIs.tiff", units="cm", width=10, height=7, res=800)
p2
dev.off()

# SAVE PDF
pdf("Figures/Figure5_graphics/AF_high_PPIs.pdf", width=4, height=3)
p2
dev.off()


```

```{r Figure5_Graph_C}
graph_x <- "Fragment"
graph_y <-  "Fragment\naverage pLDDT"

p1 <- ggplot(Model_metrics_fragments %>% filter(Fragment!="C"), aes(x=Fragment, y=avg_pLDDT, color=Fragment)) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.5) +
  ggtitle("Fragment confidence") + xlab(graph_x) + ylab(graph_y) + 
  scale_color_manual(values=colors_DMI) + 
  theme_classic() +  
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), 
        legend.position = "none", axis.ticks = element_line(linewidth = 1))

graph_y <-  "Fragment\naverage RSA"
p2 <-ggplot(Model_metrics_fragments %>% filter(Fragment!="C"), aes(x=Fragment, y=avg_Acc, color=Fragment)) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.5) +
  ggtitle("Fragment accessibility") + xlab(graph_x) + ylab(graph_y) + ylim(0.15,0.7) +
  scale_color_manual(values=colors_DMI) + 
  theme_classic() +  
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), 
        legend.position = "none", axis.ticks = element_line(linewidth = 1))

# SAVE PNG
png("Figures/Figure5_graphics/Fragments_pLDDT_RSA.png", units="cm", width=18, height=10, res=800)
grid.arrange(p1, p2,nrow=1)
dev.off()

# SAVE PDF
# pdf("Figures/Figure5_graphics/AF_high_PPIs.pdf", width=7, height=4)
# grid.arrange(p1, p2,nrow=1)
# dev.off()

grid.arrange(p1, p2,nrow=1)
```

```{r Figure5_Graph_D}
graph_x <- "Motif region"
graph_y <-  "Motif\nresidue pLDDT"

p1 <- ggplot(Model_metrics_motif, aes(x=Motif_region, y=pLDDT, color=Motif_region)) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.5) +
  ggtitle("Motif confidence") + xlab(graph_x) + ylab(graph_y) + 
  scale_color_manual(values=c("limegreen","forestgreen","limegreen")) + 
  theme_classic() +  
  theme(text = element_text(size = 20), legend.position = "none", 
        axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))
p1 

# SAVE PNG
png("Figures/Figure5_graphics/Motif_regions_pLDDT.png", units="cm", width=10, height=8, res=800)
p1
dev.off()

# SAVE PDF
# pdf("Figures/Figure5_graphics/Motif_regions_pLDDT.pdf", width=4, height=3)
# p1
# dev.off()


graph_y <-  "Motif residue RSA"
p2 <-ggplot(Model_metrics_motif, aes(x=Motif_region, y=SC_norm_Acc, color=Motif_region)) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.5) +
  ggtitle("Motif accessibility") + xlab(graph_x) + ylab(graph_y) + 
  scale_color_manual(values=c("limegreen","forestgreen","limegreen")) + 
  theme_classic() +  
  theme(text = element_text(size = 20), legend.position = "none", 
        axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))
p2 

# SAVE PNG
png("Figures/Figure5_graphics/Motif_regions_RSA.png", units="cm", width=10, height=8, res=800)
p2
dev.off()

# SAVE PDF
# pdf("Figures/Figure5_graphics/Motif_regions_RSA.pdf", width=4, height=3)
# p2
# dev.off()

graph_y <-  "Motif\nresidue \u0394 RSA"
p3 <-ggplot(Model_metrics_motif, aes(x=Motif_region, y=Acc_change, color=Motif_region)) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.2) +
  ggtitle("Motif \u0394 RSA") + xlab(graph_x) + ylab(graph_y) + 
  scale_color_manual(values=c("limegreen","forestgreen","limegreen")) + 
  theme_classic() +  
  theme(text = element_text(size = 20), legend.position = "none", 
        axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))
p3

# SAVE PNG
png("Figures/Figure5_graphics/Motif_regions_deltaRSA.png", units="cm", width=10, height=8, res=800)
p3
dev.off()

grid.arrange(p1, p2, p3, nrow=1)

graph_x <-  "Motif residue \u0394 RSA"
p3c <- ggplot(Model_metrics_motif %>% filter(Motif_region=="Core"), aes(x=Acc_change)) + geom_density() +
  ggtitle("Motif Core \u0394 RSA") + xlab(graph_x) + ylab("Density") + 
  #scale_color_manual(values=c("limegreen","forestgreen","limegreen")) + 
  theme_classic() +  
  theme(text = element_text(size = 20), legend.position = "none", 
        axis.text = element_text(color = "black"), axis.ticks = element_line(linewidth = 1))
p3c


```

```{r Figure5_Graph_D2}
graph_x <- "Motif region"
graph_y <-  "Motif residue pLDDT"

p1 <- ggplot(Model_metrics_motif, aes(x=Motif_region, y=pLDDT, fill=ELM_type)) + 
  geom_boxplot(na.rm = TRUE, width=0.25, outlier.alpha = 0.3) +
  ggtitle("Motif confidence") + xlab(graph_x) + ylab(graph_y) + 
  scale_fill_manual(values=colors_ELMtypes) + 
  theme_classic() +  theme(text = element_text(size = 15), legend.position = "none")

graph_y <-  "Motif residue RSA"
p2 <-ggplot(Model_metrics_motif, aes(x=Motif_region, y=SC_norm_Acc, fill=ELM_type)) + 
  geom_boxplot(na.rm = TRUE, width=0.25, outlier.alpha = 0.3) +
  ggtitle("Motif accessibility") + xlab(graph_x) + ylab(graph_y) + 
  scale_fill_manual(values=colors_ELMtypes) + 
  theme_classic() +  theme(text = element_text(size = 15), legend.position = "none")

p3 <-ggplot(Model_metrics_motif, aes(x=Motif_region, y=Acc_change, fill=ELM_type)) + 
  geom_boxplot(na.rm = TRUE, width=0.25, outlier.alpha = 0.3) +
  ggtitle("Motif accessibility") + xlab(graph_x) + ylab(graph_y) + 
  scale_fill_manual(values=colors_ELMtypes) + 
  theme_classic() +  theme(text = element_text(size = 15), legend.position = "none")

p1
p2
p3
```

```{r Figure5_Graph_E}
# Variants mapped
p1 <- ggplot(DMI_AF_mut_hq %>% filter(structural_region!="other"), aes(y=structural_region, fill=Pathogenicity)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  xlab("Variant fraction") + ylab("Structural region") + 
  theme_minimal() + scale_fill_manual(values=colors_Var) + labs(fill = "Pathogenicity") +
  theme(text = element_text(size = 20), legend.position = "none", axis.text = element_text(color = "black"),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  axis.ticks = element_line(size = 0.5))  + 
  scale_y_discrete(limits=rev, labels = rev(strc_categories)) 
p1 +  geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

# SAVE TIFF
tiff("Figures/Figure5_graphics/ClinVar_modelRegion.tiff", units="cm", width=12, height=8, res=800)
p1
dev.off()

# SAVE PDF
pdf("Figures/Figure5_graphics/ClinVar_modelRegion.pdf", width=7, height=4)
p1
dev.off()



DMI_AF_mut_hq %>% count(structural_region, Pathogenicity)
```

```{r Figure5_Graph_F}
DMI_AF_mut_hq_chisq$residuals
DMI_AF_mut_hq_chisq$observed

# SAVE TIFF
tiff("Figures/Figure5_graphics/ClinVar_modelRegion_chisq.tiff", units="cm", width=8, height=8, res=800)
corrplot(DMI_AF_mut_hq_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')),  col.lim = c(-15,15),
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")
dev.off()

# SAVE PDF
pdf("Figures/Figure5_graphics/ClinVar_modelRegion_chisq.pdf", width=4, height=4)
corrplot(DMI_AF_mut_hq_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')),  col.lim = c(-15,15),
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")
dev.off()

corrplot(DMI_AF_mut_hq_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')),  col.lim = c(-15,15),
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 1, outline=T, addgrid.col="white")

DMI_AF_mut_hq_PU_chisq$residuals
DMI_AF_mut_hq_PU_chisq$observed
corrplot(DMI_AF_mut_hq_PU_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')), col.lim = c(-15,15),
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")


DMI_AF_mut_interfacial_chisq$residuals
DMI_AF_mut_interfacial_chisq$observed
corrplot(DMI_AF_mut_interfacial_chisq$residuals, method = 'color', is.corr  = FALSE, col = rev(COL2('RdBu')), col.lim = c(-9,9),
         cl.pos = 'b', addCoef.col = 'grey50', number.cex = 0.01, outline=T, addgrid.col="white")

```

```{r Figure5_Graph_Supp_flanks}
graph_x <- "Motif flanks"
graph_y <-  "Motif\nresidue pLDDT"
colors_flanks <- c(rep("limegreen",4),"forestgreen",rep("limegreen",5))

p1 <- ggplot(Model_metrics_motif %>% filter(Motif_flanks> -5 & Motif_flanks < 5), aes(x=as.factor(Motif_flanks), y=pLDDT, color=as.factor(Motif_flanks))) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.5) +
  ggtitle("Motif confidence") + xlab(graph_x) + ylab(graph_y) + 
  theme_classic() + scale_color_manual(values=colors_flanks) +
  theme(text = element_text(size = 20), legend.position = "none", axis.line = element_line(size = 1) , axis.ticks = element_line(linewidth = 1))
p1

# SAVE PNG
png("Figures/Figure5_graphics/Motif_flanks_pLDDT.png", units="cm", width=12, height=8, res=800)
p1
dev.off()

graph_y <-  "Motif residue RSA"
p2 <- ggplot(Model_metrics_motif %>% filter(Motif_flanks> -5 & Motif_flanks < 5), aes(x=as.factor(Motif_flanks), y=SC_norm_Acc, color=as.factor(Motif_flanks))) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.5) +
  ggtitle("Motif flank RSA") + xlab(graph_x) + ylab(graph_y) + 
  theme_classic() + scale_color_manual(values=colors_flanks) +
  theme(text = element_text(size = 20), legend.position = "none", axis.line = element_line(size = 1) , axis.ticks = element_line(linewidth = 1))
p2

# SAVE PNG
png("Figures/Figure5_graphics/Motif_flanks_RSA.png", units="cm", width=12, height=8, res=800)
p2
dev.off()

graph_y <-  "Motif\nresidue \u0394 RSA"
p3 <-ggplot(Model_metrics_motif %>% filter(Motif_flanks> -5 & Motif_flanks < 5), aes(x=as.factor(Motif_flanks), y=Acc_change, color=as.factor(Motif_flanks))) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.5) +
  ggtitle("Motif flank \u0394 RSA") + xlab(graph_x) + ylab(graph_y) + 
  theme_classic() + scale_color_manual(values=colors_flanks) +
  theme(text = element_text(size = 20), legend.position = "none", axis.line = element_line(size = 1) , axis.ticks = element_line(linewidth = 1))
p3

# SAVE PNG
png("Figures/Figure5_graphics/Motif_flanks_deltaRSA.png", units="cm", width=12, height=8, res=800)
p3
dev.off()


ggplot(Model_metrics_motif %>% filter(Motif_flanks> -5 & Motif_flanks < 5 & ELM_class=="TRG_NLS_MonoCore_2" ), aes(x=as.factor(Motif_flanks), y=Acc_change, color=as.factor(Motif_flanks))) + 
  geom_quasirandom(size=0.5, alpha=0.5) + geom_boxplot(color="black", na.rm = TRUE, width=0.25, alpha=0.5, outlier.alpha = 0.5) +
  ggtitle("Motif flank \u0394 RSA") + xlab(graph_x) + ylab(graph_y) + 
  theme_classic() + scale_color_manual(values=colors_flanks) +
  theme(text = element_text(size = 20), legend.position = "none", axis.line = element_line(size = 1) , axis.ticks = element_line(linewidth = 1))

Model_metrics_motif %>% filter(Motif_flanks> -5 & Motif_flanks < 5 & ELM_type=="TRG") %>% select(ELM_class) %>% unique() %>% unlist()
```



```{r Figure6_Graph_B}
DMI_AF_mut_hq$AM_pathogenicity <- toTitleCase(gsub("AM_","", DMI_AF_mut_hq$AM_pathogenicity))
DMI_AF_mut_hq$AM_pathogenicity <- ordered(DMI_AF_mut_hq$AM_pathogenicity, levels = c("Benign","Ambiguous","Pathogenic"))

# Alpha Missence variants

p2 <- ggplot(DMI_AF_mut_hq %>% filter(structural_region!="other"), aes(y=structural_region, fill=AM_pathogenicity)) + 
  geom_bar(position="fill", width = 0.3) +
  xlab("Variant fraction") + ylab("Structural region") + 
  theme_classic() + scale_fill_manual(values=colors_Var) + labs(fill = "Alpha Missense") +
  theme(text = element_text(size = 15), axis.title.x=element_blank(), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1))  
p2 +  geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 
  
# Alpha Missence - Uncertain variants
p2a <- ggplot(DMI_AF_mut_hq %>% filter(structural_region!="Other" & Pathogenicity=="Uncertain"), aes(y=structural_region, fill=AM_pathogenicity)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4)  +
  xlab("Uncertain variant fraction") + ylab("Structural region") + 
  theme_minimal() + scale_fill_manual(values=colors_Var) + labs(fill = "Alpha Missense") +
  theme(text = element_text(size = 20), axis.text = element_text(color = "black"), panel.grid.major = element_blank(), legend.position="none", panel.grid.minor = element_blank())  + 
  scale_y_discrete(limits=rev, labels = rev(strc_categories)) 

p2a +  geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3)

# SAVE TIFF
tiff("Figures/Figure6_graphics/AM_modelRegion_VUS.tiff", units="cm", width=12, height=8, res=800)
p2a
dev.off()

# SAVE PDF
pdf("Figures/Figure6_graphics/AM_modelRegion_VUS_chisq.pdf", width=7, height=4)
p2a
dev.off()
 

plots_title <- "Alpha Missence - Pathogenic variants"
p2b <- ggplot(DMI_AF_mut_hq %>% filter(structural_region!="Other" & Pathogenicity=="Pathogenic"), aes(x=structural_region, fill=AM_pathogenicity)) + 
  geom_bar(position="fill", width = 0.3) +
  ggtitle(plots_title) +  xlab(plots_xlab) + ylab(plots_ylab) + 
  theme_classic() + scale_fill_manual(values=colors_Var) + labs(fill = "Alpha Missense") +
  theme(text = element_text(size = 15), axis.title.x=element_blank(), axis.text.x = element_text(angle = 30, vjust = 1, hjust=1))  
p2b +  geom_text(aes(label = ..count..), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

```

```{r Figure6_sankeys_Graph_A}
#https://r-graph-gallery.com/322-custom-colours-in-sankey-diagram.html
i <- 1
SK1 <- DMI_AF_mut_mapped %>% filter(structural_region==regions[[i]]) %>% select(-structural_region)
p1 <- sankeyNetwork(Links = SK1, Nodes = nodes, colourScale=my_color,  NodeGroup="group",
              Source = "ClinVar", Target = "AlphaMissence",
              Value = "value", NodeID = "name", sinksRight=FALSE, 
              nodeWidth=50, nodePadding=30, fontSize=0.01, fontFamily = 'Helvetica',
              height = 500, width = 500)
p1

i <- 2
SK2 <- DMI_AF_mut_mapped %>% filter(structural_region==regions[[i]]) %>% select(-structural_region)
p2 <- sankeyNetwork(Links = SK2, Nodes = nodes, colourScale=my_color, NodeGroup="group",
              Source = "ClinVar", Target = "AlphaMissence",
              Value = "value", NodeID = "name", sinksRight=FALSE, 
              nodeWidth=50, nodePadding=30, fontSize=0.01, fontFamily = 'Helvetica',
              height = 500, width = 500)
p2

i <- 3
SK3 <- DMI_AF_mut_mapped %>% filter(structural_region==regions[[i]])# %>% select(-structural_region)
p3 <- sankeyNetwork(Links = SK3, Nodes = nodes, colourScale=my_color, NodeGroup="group",
              Source = "ClinVar", Target = "AlphaMissence",
              Value = "value", NodeID = "name", sinksRight=FALSE, 
              nodeWidth=50, nodePadding=30, fontSize=0.01, fontFamily = 'Helvetica',
              height = 500, width = 500)

saveNetwork(p3, file = 'Figures/Figure6_graphics/Region_domain_interface.html')
p3

i <- 4
SK4 <- DMI_AF_mut_mapped %>% filter(structural_region==regions[[i]]) %>% select(-structural_region)
p4 <- sankeyNetwork(Links = SK4, Nodes = nodes, colourScale=my_color, NodeGroup="group",
              Source = "ClinVar", Target = "AlphaMissence",
              Value = "value", NodeID = "name", sinksRight=FALSE, 
              nodeWidth=50, nodePadding=30, fontSize=0.01, fontFamily = 'Helvetica',
              height = 500, width = 500)
p4

i <- 5
SK5 <- DMI_AF_mut_mapped %>% filter(structural_region==regions[[i]]) %>% select(-structural_region)
p5 <- sankeyNetwork(Links = SK5, Nodes = nodes, colourScale=my_color, NodeGroup="group",
              Source = "ClinVar", Target = "AlphaMissence",
              Value = "value", NodeID = "name", sinksRight=FALSE, 
              nodeWidth=50, nodePadding=30, fontSize=0.01, fontFamily = 'Helvetica',
              height = 500, width = 500)

p5
saveNetwork(p5, file = 'Figures/Figure6_graphics/Region_motif_interface.html')


```

```{r Figure6_Graph_D}
p1 <- ggplot(DMI_AF_mut_hq %>% filter(structural_region=="m_interface" & Mapped_pathogenicity=="U_B"), aes(x=Norm_Acc, y=pLDDT)) + 
  geom_point(alpha=0.3) + geom_hline(yintercept=80, color="red") + geom_vline(xintercept=0.36, color="red") +
  ggtitle("Interfacial motif variants") +  xlab("RSA") + ylab("Residue pLDDT") + 
  theme_classic() + 
  theme(text = element_text(size = 20))  
p1 

p2 <- ggplot(DMI_AF_mut_hq %>% filter((structural_region=="m_interface" | structural_region=="d_interface") & Mapped_pathogenicity=="U_B"), aes(x=Norm_Acc, y=pLDDT, color=Fragment)) + geom_point(size=0.5, alpha=0.5) + 
  ggtitle("Interfacial motif variants") +  xlab("RSA") + ylab("Residue pLDDT") + 
  theme_classic() + scale_color_manual(values=colors_DMI) + 
  theme(text = element_text(size = 20))  
p2

p3 <- ggplot(DMI_AF_mut_hq %>% filter((structural_region=="m_interface" | structural_region=="d_interface") & Mapped_pathogenicity=="U_B"), aes(x=Fragment, y=pLDDT, color=Fragment)) + geom_quasirandom(size=1, alpha=0.75) + geom_boxplot(color="black", width=0.2) + 
  ggtitle("Interfacial motif UB variants") +  xlab("Fragment") + ylab("Residue pLDDT") + 
  theme_classic() + scale_color_manual(values=colors_DMI) + 
  theme(text = element_text(size = 20), legend.position = "none")  
p3

# SAVE TIFF
tiff("Figures/Figure6_graphics/Interfacial_UB_pLDDT_Fragment.tiff", units="cm", width=10, height=10, res=800)
p3
dev.off()

# SAVE PDF
pdf("Figures/Figure6_graphics/Interfacial_UB_pLDDT_Fragment.pdf", width=5, height=3.5)
p3
dev.off()

DMI_AF_mut_hq %>% filter(structural_region=="m_interface" & Mapped_pathogenicity=="U_B" & pLDDT>=80 & Norm_Acc<=0.36) %>% nrow()
```

```{r Figure6_Graph_VariantMatch}

p1 <- ggplot(HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface"), aes(x=pathogenicity, fill=as.factor(variant_match))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Motif interface residue variant REGEX match") +  xlab("Pathogenicity") + ylab("Variant number") + 
  theme_minimal() + scale_fill_manual(values=colors_BIN) + labs(fill = "REGEX match") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p1 +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

p2 <- ggplot(HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface" & pathogenicity=="Uncertain"), aes(x=as.factor(variant_match), fill=AM_pathogenicity)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Motif interface residue VUS REGEX match") +  xlab("AM pathogenicity") + ylab("Variant number") + 
  theme_minimal() + scale_fill_manual(values=colors_Var) + labs(fill = "REGEX match") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p2 +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

p2 <- ggplot(HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface" & pathogenicity=="Uncertain"), aes(x=AM_pathogenicity, fill=as.factor(variant_match))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Motif interface residue VUS REGEX match") +  xlab("AM pathogenicity") + ylab("Variant number") + 
  theme_minimal() + scale_fill_manual(values=colors_BIN) + labs(fill = "REGEX match") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p2 +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

p3 <- ggplot(HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface" & pathogenicity=="Uncertain"), aes(x=as.factor(variant_match) , fill=ELM_type )) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Motif interface VUS REGEX match") +  xlab("ELM type") + ylab("Variant number") + 
  theme_minimal() + scale_fill_manual(values=colors_ELMtypes[c(1:4,6)]) + labs(fill = "REGEX match") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p3 +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

p3 <- ggplot(HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface" & pathogenicity=="Uncertain"), aes(x=ELM_type, fill=as.factor(variant_match))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Motif interface VUS REGEX match") +  xlab("ELM type") + ylab("Variant number") + 
  theme_minimal() + scale_fill_manual(values=colors_BIN) + labs(fill = "REGEX match") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p3 +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 


```

```{r Figure6_Graph_Larger_network}
plot(BU_graph2, layout = layout_nicely(BU_graph2), 
     vertex.size = 4 , vertex.label.cex=0.4, vertex.label.color="black", #vertex.label.dist=1, 
     edge.size = 60, edge.label.cex=0.4,  edge.label.color="black", #edge.label.dist=10,
     edge.arrow.size=0.1, edge.width=1.5, main = "PPI network -")


```

```{r Figure6_Graph_DO_network}

plot(UB_pairs_ig, layout = layout_nicely(UB_pairs_ig), 
     vertex.size = 12, vertex.label.cex = 0.4, vertex.label.color = "black", vertex.frame.color=NA,
     edge.size = 15, edge.label.cex = 0.01, edge.color = "dimgrey", 
     edge.arrow.size = 0.15, edge.width = 1, main = "UB network - ALL")

# SAVE TIFF
tiff("Figures/Figure6_graphics/UB_network_MotifSide.tiff", units="cm", width=20, height=20, res=800, bg = "transparent")
plot(UB_pairs_ig, layout = layout_nicely(UB_pairs_ig), 
     vertex.size = 8, vertex.label.cex = 0.4, vertex.label.color = "black", vertex.frame.color=NA,
     edge.size = 12, edge.label.cex = 0.01, edge.color = "dimgrey", 
     edge.arrow.size = 0.15, edge.width = 1, main = "UB network - ALL")
dev.off()

# SAVE PDF
pdf("Figures/Figure6_graphics/UB_network.pdf", width=5, height=5)
plot(UB_pairs_ig, layout = layout_nicely(UB_pairs_ig), 
     vertex.size = 8, vertex.label.cex = 0.4, vertex.label.color = "black",  vertex.frame.color=NA,
     edge.size = 30, edge.label.cex = 0.01, edge.color = "dimgrey", 
     edge.arrow.size = 0.15, edge.width = 1, main = "UB network - ALL")
dev.off()

```

```{r Figure6_Graph_GO_terms}
V(UB_pairs_ig)$color <- UB_pairs_vertices$color2

plot(UB_pairs_ig, layout = layout_nicely(UB_pairs_ig), 
     vertex.size = 12, vertex.label.cex = 0.4, vertex.label.color = "black", vertex.frame.color=NA,
     edge.size = 15, edge.label.cex = 0.01, edge.color = "dimgrey", 
     edge.arrow.size = 0.15, edge.width = 1, main = "UB network - ALL")

# SAVE PDF
pdf("Figures/Figure6_graphics/UB_network_GO.pdf", width=5, height=5)
plot(UB_pairs_ig, layout = layout_nicely(UB_pairs_ig), 
     vertex.size = 8, vertex.label.cex = 0.2, vertex.label.color = "black",  vertex.frame.color=NA,
     edge.size = 30, edge.label.cex = 0.01, edge.color = "dimgrey", 
     edge.arrow.size = 0.15, edge.width = 1, main = "UB network - ALL")
dev.off()


```

```{r Figure6_Graph_REGEXchange}

p1a <- ggplot(HURI_ClinVar_mut_motif_2, aes(x=as.factor(variant_match), fill=pathogenicity)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Variant and REGEX match") +  xlab("REGEX match") + ylab("Variant number") + 
  theme_minimal() + scale_fill_manual(values=colors_Var) + labs(fill = "Pathogenicity") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p1a +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

p1b <- ggplot(HURI_ClinVar_mut_motif_2, aes(x=as.factor(variant_match), fill=AM_pathogenicity)) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Variant and REGEX match") +  xlab("REGEX match") + ylab("Variant number") + 
  theme_minimal() + scale_fill_manual(values=colors_Var) + labs(fill = "Pathogenicity") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p1b +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

p2 <- ggplot(HURI_ClinVar_mut_motif_summ, aes(x=structural_region, fill=as.factor(variant_match))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Variant and REGEX match") +  xlab("Structural region") + ylab("Variant number") + 
  theme_minimal() + scale_fill_manual(values=colors_BIN) + labs(fill = "REGEX match") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p2 +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

p3 <- ggplot(HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface"), aes(x=Pathogenicity, fill=as.factor(variant_match))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Variant and REGEX match") +  xlab("ClinVar athogenicity") + ylab("Motif interface\nvariant number") + 
  theme_minimal() + scale_fill_manual(values=colors_BIN) + labs(fill = "REGEX match") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p3 +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

p4 <- ggplot(HURI_ClinVar_mut_motif_summ %>% filter(structural_region=="m_interface"), aes(x=AM_pathogenicity, fill=as.factor(variant_match))) + 
  geom_bar(position="fill", width = 0.6, color="black", size=0.4) +
  ggtitle("Variant and REGEX match") +  xlab("AM pathogenicity") + ylab("Motif interface\nvariant number") + 
  theme_minimal() + scale_fill_manual(values=colors_BIN) + labs(fill = "REGEX match") +
  theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.ticks.y = element_line(size = 1))
p4 +geom_text(aes(label = after_stat(count)), stat = "count", vjust = -0.1,hjust= 1, position = position_fill(0.25), size = 3) +
  geom_text(stat='prop', vjust=-0.1, hjust= -0.1, position = position_fill(0.25), size=3) 

```

```{r examples}
UB_proteins
#Q9NZC7_Q99732
#Q08209_Q96KS9
DMI_AF_metrics %>% filter(intx_ID=="O43791_Q99836")
#O43791_Q99836
#All_DMI %>% filter(intx_ID=="O43791_Q99836")
ClinVar_mutations %>% filter(uniprot_id=="P42768") %>%  count(pathogenicity)
```


